From be3ced3ff26f7929b18902ee6c4e63b551eef42f Mon Sep 17 00:00:00 2001
From: Robert Ancell <robert.ancell@canonical.com>
Date: Tue, 12 Mar 2019 17:05:34 +1300
Subject: [PATCH] applications: Add missing whitespace

---
 panels/applications/cc-applications-panel.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/panels/applications/cc-applications-panel.c b/panels/applications/cc-applications-panel.c
index 6d7be3398..a0ac02417 100644
--- a/panels/applications/cc-applications-panel.c
+++ b/panels/applications/cc-applications-panel.c
@@ -1346,6 +1346,7 @@ clear_cache_cb (CcApplicationsPanel *self)
   dir = get_flatpak_app_dir (self->current_app_id, "cache");
   file_remove_async (dir, self->cancellable, cache_cleared, self);
 }
+
 static void
 update_app_row (CcApplicationsPanel *self,
                 const gchar         *app_id)
-- 
2.24.0

