From d38617482d618734365091029c3f872a01fbb7e5 Mon Sep 17 00:00:00 2001
From: Robert Ancell <robert.ancell@canonical.com>
Date: Tue, 12 Mar 2019 17:05:34 +1300
Subject: [PATCH 2/7] applications: Add missing whitespace

---
 panels/applications/cc-applications-panel.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/panels/applications/cc-applications-panel.c b/panels/applications/cc-applications-panel.c
index 16b423292..edcf0738c 100644
--- a/panels/applications/cc-applications-panel.c
+++ b/panels/applications/cc-applications-panel.c
@@ -1292,6 +1292,7 @@ clear_cache_cb (CcApplicationsPanel *self)
   dir = get_flatpak_app_dir (self->current_app_id, "cache");
   file_remove_async (dir, cache_cleared, self);
 }
+
 static void
 update_app_row (CcApplicationsPanel *self,
                 const gchar         *app_id)
-- 
2.20.1

