From: Robert Ancell <robert.ancell@canonical.com>
Date: Tue, 12 Mar 2019 17:05:34 +1300
Subject: [PATCH 2/7] applications: Add missing whitespace

---
 panels/applications/cc-applications-panel.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/panels/applications/cc-applications-panel.c b/panels/applications/cc-applications-panel.c
index 1cc4760..204395c 100644
--- a/panels/applications/cc-applications-panel.c
+++ b/panels/applications/cc-applications-panel.c
@@ -1315,6 +1315,7 @@ clear_cache_cb (CcApplicationsPanel *self)
   dir = get_flatpak_app_dir (self->current_app_id, "cache");
   file_remove_async (dir, self->cancellable, cache_cleared, self);
 }
+
 static void
 update_app_row (CcApplicationsPanel *self,
                 const gchar         *app_id)
