From d38617482d618734365091029c3f872a01fbb7e5 Mon Sep 17 00:00:00 2001
From: Robert Ancell <robert.ancell@canonical.com>
Date: Tue, 12 Mar 2019 17:05:34 +1300
Subject: [PATCH 2/7] applications: Add missing whitespace

---
 panels/applications/cc-applications-panel.c | 1 +
 1 file changed, 1 insertion(+)

Index: gnome-control-center-3.32.1/panels/applications/cc-applications-panel.c
===================================================================
--- gnome-control-center-3.32.1.orig/panels/applications/cc-applications-panel.c
+++ gnome-control-center-3.32.1/panels/applications/cc-applications-panel.c
@@ -1312,6 +1312,7 @@ clear_cache_cb (CcApplicationsPanel *sel
   dir = get_flatpak_app_dir (self->current_app_id, "cache");
   file_remove_async (dir, self->cancellable, cache_cleared, self);
 }
+
 static void
 update_app_row (CcApplicationsPanel *self,
                 const gchar         *app_id)
