From d1435afe4c50d836d98076d290e9e49cf0cec4a9 Mon Sep 17 00:00:00 2001
From: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Tue, 31 Jul 2018 04:44:12 +0100
Subject: [PATCH 06/22] keyboard: Add launch terminal shortcut

---
 panels/keyboard/01-launchers.xml.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/panels/keyboard/01-launchers.xml.in b/panels/keyboard/01-launchers.xml.in
index 67c83254b..dd0292dfb 100644
--- a/panels/keyboard/01-launchers.xml.in
+++ b/panels/keyboard/01-launchers.xml.in
@@ -9,6 +9,8 @@
 
 	<KeyListEntry name="email" description="Launch email client"/>
 
+	<KeyListEntry name="terminal" description="Launch terminal"/>
+
 	<KeyListEntry name="www" description="Launch web browser"/>
 
 	<KeyListEntry name="home" description="Home folder"/>
-- 
2.25.1

