From: Robert Ancell <robert.ancell@canonical.com>
Date: Mon, 6 Apr 2020 10:30:27 +1200
Subject: [PATCH 26/26] applications: Launch snap-store if it is installed

---
 panels/applications/cc-applications-panel.c | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

Index: gnome-control-center/panels/applications/cc-applications-panel.c
===================================================================
--- gnome-control-center.orig/panels/applications/cc-applications-panel.c
+++ gnome-control-center/panels/applications/cc-applications-panel.c
@@ -161,6 +161,13 @@ gnome_software_is_installed (void)
   return path != NULL;
 }
 
+static gboolean
+snap_store_is_installed (void)
+{
+  g_autofree gchar *path = g_find_program_in_path ("snap-store");
+  return path != NULL;
+}
+
 /* Callbacks */
 
 static gboolean
@@ -180,6 +187,9 @@ open_software_cb (CcApplicationsPanel *s
 {
   const gchar *argv[] = { "gnome-software", "--details", "appid", NULL };
 
+  if (!gnome_software_is_installed ())
+    argv[0] = "snap-store";
+
   if (self->current_app_id == NULL)
     argv[1] = NULL;
   else
@@ -1678,7 +1688,7 @@ update_panel (CcApplicationsPanel *self,
 
   gtk_label_set_label (self->title_label, g_app_info_get_display_name (info));
   gtk_stack_set_visible_child (self->stack, GTK_WIDGET (self->settings_box));
-  gtk_widget_set_visible (GTK_WIDGET (self->header_button), gnome_software_is_installed ());
+  gtk_widget_set_visible (GTK_WIDGET (self->header_button), gnome_software_is_installed () || snap_store_is_installed ());
 
   g_clear_pointer (&self->current_app_id, g_free);
   g_clear_pointer (&self->current_portal_app_id, g_free);
@@ -2063,7 +2073,7 @@ cc_applications_panel_init (CcApplicatio
 
   gtk_widget_init_template (GTK_WIDGET (self));
 
-  gtk_widget_set_visible (GTK_WIDGET (self->install_button), gnome_software_is_installed ());
+  gtk_widget_set_visible (GTK_WIDGET (self->install_button), gnome_software_is_installed () || snap_store_is_installed ());
 
   provider = GTK_STYLE_PROVIDER (gtk_css_provider_new ());
   gtk_css_provider_load_from_resource (GTK_CSS_PROVIDER (provider),
