Index: gnome-control-center-2.91.92/configure.ac
===================================================================
--- gnome-control-center-2.91.92.orig/configure.ac	2011-03-21 16:55:17.000000000 +0100
+++ gnome-control-center-2.91.92/configure.ac	2011-03-21 20:56:18.160367002 +0100
@@ -123,12 +123,7 @@
 GDESKTOP_PREFIX=`$PKG_CONFIG --variable prefix gsettings-desktop-schemas`
 AC_SUBST(GDESKTOP_PREFIX)
 
-# Check for NetworkManager ~0.9
-PKG_CHECK_MODULES(NETWORK_MANAGER, libnm-glib >= $NETWORK_MANAGER_REQUIRED_VERSION,
-		  [have_networkmanager=yes], have_networkmanager=no)
-if test "x$have_networkmanager" = xno ; then
-        AC_MSG_WARN(*** Network panel will not be built (NetworkManager ~0.9 or newer not found) ***)
-fi
+have_networkmanager=yes
 AM_CONDITIONAL(BUILD_NETWORK, [test x$have_networkmanager = xyes])
 
 # Check for CUPS 1.4 or newer
Index: gnome-control-center-2.91.92/panels/network/Makefile.am
===================================================================
--- gnome-control-center-2.91.92.orig/panels/network/Makefile.am	2011-03-21 20:55:18.080367002 +0100
+++ gnome-control-center-2.91.92/panels/network/Makefile.am	2011-03-21 20:56:18.160367002 +0100
@@ -14,18 +14,6 @@
 
 libnetwork_la_SOURCES =					\
 	network-module.c				\
-	panel-common.c					\
-	panel-common.h					\
-	net-object.c					\
-	net-object.h					\
-	net-device.c					\
-	net-device.h					\
-	net-vpn.c					\
-	net-vpn.h					\
-	panel-cell-renderer-mode.c			\
-	panel-cell-renderer-mode.h			\
-	panel-cell-renderer-signal.c			\
-	panel-cell-renderer-signal.h			\
 	cc-network-panel.c				\
 	cc-network-panel.h
 
Index: gnome-control-center-2.91.92/panels/network/cc-network-panel.c
===================================================================
--- gnome-control-center-2.91.92.orig/panels/network/cc-network-panel.c	2011-03-21 20:55:18.072367002 +0100
+++ gnome-control-center-2.91.92/panels/network/cc-network-panel.c	2011-03-21 21:03:09.456367002 +0100
@@ -25,6 +25,7 @@
 
 #include "cc-network-panel.h"
 
+#if 0
 #include "nm-remote-settings.h"
 #include "nm-client.h"
 #include "nm-device.h"
@@ -47,6 +48,7 @@
 #include "panel-common.h"
 #include "panel-cell-renderer-mode.h"
 #include "panel-cell-renderer-signal.h"
+#endif
 
 G_DEFINE_DYNAMIC_TYPE (CcNetworkPanel, cc_network_panel, CC_TYPE_PANEL)
 
@@ -58,8 +60,6 @@
         GCancellable     *cancellable;
         GSettings        *proxy_settings;
         GtkBuilder       *builder;
-        NMClient         *client;
-        NMRemoteSettings *remote_settings;
         gboolean          updating_device;
 };
 
@@ -124,6 +124,7 @@
                 g_object_unref (priv->builder);
                 priv->builder = NULL;
         }
+#if 0
         if (priv->client != NULL) {
                 g_object_unref (priv->client);
                 priv->client = NULL;
@@ -132,6 +133,7 @@
                 g_object_unref (priv->remote_settings);
                 priv->remote_settings = NULL;
         }
+#endif
 
         G_OBJECT_CLASS (cc_network_panel_parent_class)->dispose (object);
 }
@@ -213,6 +215,7 @@
         check_wpad_warning (panel);
 }
 
+#if 0
 static NetObject *
 get_selected_object (CcNetworkPanel *panel)
 {
@@ -235,6 +238,7 @@
 
         return object;
 }
+#endif
 
 static void
 panel_proxy_mode_combo_setup_widgets (CcNetworkPanel *panel, guint value)
@@ -336,6 +340,7 @@
         gtk_tree_path_free (path);
 }
 
+#if 0
 static void
 panel_device_got_modem_manager_cb (GObject *source_object,
                                    GAsyncResult *res,
@@ -596,6 +601,7 @@
                 g_object_unref (object_tmp);
         } while (gtk_tree_model_iter_next (model, &iter));
 }
+#endif
 
 static void
 panel_add_devices_columns (CcNetworkPanel *panel, GtkTreeView *treeview)
@@ -631,6 +637,7 @@
         gtk_tree_view_column_set_expand (column, TRUE);
 }
 
+#if 0
 static void
 panel_set_widget_data (CcNetworkPanel *panel,
                        const gchar *sub_pane,
@@ -1460,6 +1467,7 @@
                                "group_password",
                                 net_vpn_get_password (vpn));
 }
+#endif
 
 static void
 refresh_ui (CcNetworkPanel *panel)
@@ -1468,7 +1476,7 @@
         GtkTreeIter iter;
         GtkTreeModel *model;
         GtkWidget *widget;
-        NetObject *object = NULL;
+        void *object = NULL;
         CcNetworkPanelPrivate *priv = panel->priv;
 
         widget = GTK_WIDGET (gtk_builder_get_object (panel->priv->builder,
@@ -1481,8 +1489,6 @@
                 goto out;
         }
 
-        object = get_selected_object (panel);
-
         /* this is the proxy settings device */
         if (object == NULL) {
 
@@ -1516,6 +1522,7 @@
                 goto out;
         }
 
+#if 0
         /* VPN */
         if (NET_IS_VPN (object)) {
                 nm_device_refresh_vpn_ui (panel, NET_VPN (object));
@@ -1538,6 +1545,7 @@
                 /* refresh device */
                 nm_device_refresh_device_ui (panel, NET_DEVICE (object));
         }
+#endif
 out:
         return;
 }
@@ -1571,6 +1579,7 @@
         g_free (title);
 }
 
+#if 0
 static void
 cc_network_panel_notify_enable_active_cb (GtkSwitch *sw,
                                           GParamSpec *pspec,
@@ -1919,25 +1928,17 @@
                 return;
         }
 }
+#endif
 
 static void
 on_toplevel_map (GtkWidget      *widget,
                  CcNetworkPanel *panel)
 {
-        gboolean ret;
-
-        /* is the user compiling against a new version, but running an
-         * old daemon version? */
-        ret = panel_check_network_manager_version (panel);
-        if (ret) {
-                manager_running (panel->priv->client, NULL, panel);
-        } else {
-                /* just select the proxy settings */
-                select_first_device (panel);
-        }
+        /* just select the proxy settings */
+        select_first_device (panel);
 }
 
-
+#if 0
 static void
 wireless_ap_changed_cb (GtkComboBox *combo_box, CcNetworkPanel *panel)
 {
@@ -2023,11 +2024,11 @@
         g_free (ssid_target);
         g_free (object_path);
 }
+#endif
 
 static void
 cc_network_panel_init (CcNetworkPanel *panel)
 {
-        DBusGConnection *bus = NULL;
         gboolean ret;
         GError *error = NULL;
         gint value;
@@ -2154,6 +2155,7 @@
         /* add the virtual proxy device */
         panel_add_proxy_device (panel);
 
+#if 0
         /* setup wireless combobox model */
         combobox = GTK_COMBO_BOX (gtk_builder_get_object (panel->priv->builder,
                                                           "combobox_network_name"));
@@ -2255,6 +2257,7 @@
                           G_CALLBACK (notify_connections_read_cb), panel);
         g_signal_connect (panel->priv->remote_settings, NM_REMOTE_SETTINGS_NEW_CONNECTION,
                           G_CALLBACK (notify_new_connection_cb), panel);
+#endif
 
         toplevel = gtk_widget_get_toplevel (GTK_WIDGET (panel));
         g_signal_connect_after (toplevel, "map", G_CALLBACK (on_toplevel_map), panel);
