diff --git a/shell/gnome-control-center.c b/shell/gnome-control-center.c
index a771282..fbca9c0 100644
--- a/shell/gnome-control-center.c
+++ b/shell/gnome-control-center.c
@@ -167,7 +167,23 @@ activate_panel (GnomeControlCenter *shell,
         }
       else
         {
-          g_warning ("Could not find the loadable module for panel '%s'", id);
+	  GKeyFile *key_file;
+
+	  /* It might be an external panel */
+	  key_file = g_key_file_new ();
+	  if (g_key_file_load_from_file (key_file, desktop_file, G_KEY_FILE_NONE, NULL))
+	    {
+	      gchar *command;
+
+	      command = g_key_file_get_string (key_file, G_KEY_FILE_DESKTOP_GROUP, G_KEY_FILE_DESKTOP_KEY_EXEC, NULL);
+	      if (command && command[0])
+	        {
+		  g_spawn_command_line_async (command, NULL);
+		  g_free (command);
+		}
+	    }
+
+	  g_key_file_free (key_file);
         }
     }
 }
diff --git a/configure.ac b/configure.ac
index 98ece7f..6210f5f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -303,6 +303,7 @@ AC_CONFIG_FILES([
 Makefile
 help/Makefile
 libgnome-control-center/Makefile
+libgnome-control-center/libgnome-control-center.pc
 panels/Makefile
 panels/common/Makefile
 panels/background/Makefile
diff --git a/libgnome-control-center/Makefile.am b/libgnome-control-center/Makefile.am
index eff6734..0fdfa3d 100644
--- a/libgnome-control-center/Makefile.am
+++ b/libgnome-control-center/Makefile.am
@@ -11,6 +11,11 @@ AM_CPPFLAGS =					\
 
 lib_LTLIBRARIES = libgnome-control-center.la
 
+libgnome_control_center_include_HEADERS =      \
+	cc-panel.h                              \
+	cc-shell.h                              \
+	$(NULL)
+
 libgnome_control_center_la_SOURCES =		\
 	cc-panel.c				\
 	cc-panel.h				\
@@ -31,4 +36,13 @@ libgnome_control_center_la_LIBADD = 		\
 
 libgnome_control_center_la_LIBTOOLFLAGS = --tag=disable-static
 
+libgnome_control_center_includedir = $(includedir)/gnome-control-center-1/libgn
+
+pkgconfigdir=$(libdir)/pkgconfig
+pkgconfig_DATA=libgnome-control-center.pc
+
+EXTRA_DIST =                                   \
+	libgnome-control-center.pc.in  \
+	$(NULL)
+
 -include $(top_srcdir)/git.mk
diff --git a/libgnome-control-center/libgnome-control-center.pc.in b/libgnome-control-center/libgnome-control-center.pc.in
new file mode 100644
index 0000000..54defcf
--- /dev/null
+++ b/libgnome-control-center/libgnome-control-center.pc.in
@@ -0,0 +1,12 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+extensiondir=@libdir@/control-center-1/panels
+
+Name: libgnome-control-center
+Description: A library to create GNOME Control Center extensions
+Version: @VERSION@
+Requires: glib-2.0 gio-2.0 gtk+-3.0
+Libs: -L${libdir} -lgnome-control-center
+Cflags: -I${includedir}/gnome-control-center-1
