Index: gnome-control-center-3.1.4/shell/gnome-control-center.c
===================================================================
--- gnome-control-center-3.1.4.orig/shell/gnome-control-center.c	2011-07-08 12:10:36.000000000 -0400
+++ gnome-control-center-3.1.4/shell/gnome-control-center.c	2011-07-27 11:11:05.678459013 -0400
@@ -167,7 +167,23 @@
         }
       else
         {
-          g_warning ("Could not find the loadable module for panel '%s'", id);
+	  GKeyFile *key_file;
+
+	  /* It might be an external panel */
+	  key_file = g_key_file_new ();
+	  if (g_key_file_load_from_file (key_file, desktop_file, G_KEY_FILE_NONE, NULL))
+	    {
+	      gchar *command;
+
+	      command = g_key_file_get_string (key_file, G_KEY_FILE_DESKTOP_GROUP, G_KEY_FILE_DESKTOP_KEY_EXEC, NULL);
+	      if (command && command[0])
+	        {
+		  g_spawn_command_line_async (command, NULL);
+		  g_free (command);
+		}
+	    }
+
+	  g_key_file_free (key_file);
         }
     }
 }
Index: gnome-control-center-3.1.4/configure.ac
===================================================================
--- gnome-control-center-3.1.4.orig/configure.ac	2011-07-25 14:50:52.000000000 -0400
+++ gnome-control-center-3.1.4/configure.ac	2011-07-27 11:11:05.682459013 -0400
@@ -303,6 +303,7 @@
 Makefile
 help/Makefile
 libgnome-control-center/Makefile
+libgnome-control-center/libgnome-control-center.pc
 panels/Makefile
 panels/common/Makefile
 panels/background/Makefile
Index: gnome-control-center-3.1.4/libgnome-control-center/Makefile.am
===================================================================
--- gnome-control-center-3.1.4.orig/libgnome-control-center/Makefile.am	2011-07-20 06:53:14.000000000 -0400
+++ gnome-control-center-3.1.4/libgnome-control-center/Makefile.am	2011-07-27 11:11:25.346459221 -0400
@@ -11,6 +11,11 @@
 
 lib_LTLIBRARIES = libgnome-control-center.la
 
+libgnome_control_center_include_HEADERS =      \
+	cc-panel.h                              \
+	cc-shell.h                              \
+	$(NULL)
+
 libgnome_control_center_la_SOURCES =		\
 	cc-panel.c				\
 	cc-panel.h				\
@@ -31,4 +36,13 @@
 
 libgnome_control_center_la_LIBTOOLFLAGS = --tag=disable-static
 
+libgnome_control_center_includedir = $(includedir)/gnome-control-center-1/libgnome-control-center
+
+pkgconfigdir=$(libdir)/pkgconfig
+pkgconfig_DATA=libgnome-control-center.pc
+
+EXTRA_DIST =                                   \
+	libgnome-control-center.pc.in  \
+	$(NULL)
+
 -include $(top_srcdir)/git.mk
Index: gnome-control-center-3.1.4/libgnome-control-center/libgnome-control-center.pc.in
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ gnome-control-center-3.1.4/libgnome-control-center/libgnome-control-center.pc.in	2011-07-27 11:11:05.682459013 -0400
@@ -0,0 +1,12 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+extensiondir=@libdir@/control-center-1/panels
+
+Name: libgnome-control-center
+Description: A library to create GNOME Control Center extensions
+Version: @VERSION@
+Requires: glib-2.0 gio-2.0 gtk+-3.0
+Libs: -L${libdir} -lgnome-control-center
+Cflags: -I${includedir}/gnome-control-center-1
