--- Makefile.am.old	2008-03-16 21:21:10.000000000 +0100
+++ Makefile.am	2008-03-16 21:21:20.000000000 +0100
@@ -1,4 +1,4 @@
-SUBDIRS = po libsounds libbackground libwindow-settings libslab capplets shell help vfs-methods $(TYPING_BREAK)
+SUBDIRS = po libsounds libbackground libwindow-settings capplets shell help vfs-methods $(TYPING_BREAK)
 
 ACLOCAL_AMFLAGS = -I m4
 
--- configure.in.old	2008-03-16 21:21:28.000000000 +0100
+++ configure.in	2008-03-16 21:24:42.000000000 +0100
@@ -86,9 +86,6 @@
 dnl Check that we meet the  dependencies
 dnl ==============================================
 
-ENABLE_DYNAMIC_LIBSLAB=0
-AM_CONDITIONAL(ENABLE_DYNAMIC_LIBSLAB, test "x$ENABLE_DYNAMIC_LIBSLAB" = "x1")
-
 COMMON_MODULES="gtk+-2.0 >= 2.11.6 dnl
  gio-2.0 dnl
  gconf-2.0 dnl
@@ -97,7 +94,6 @@
  libbonobo-2.0 dnl
  libbonoboui-2.0 dnl
  gnome-desktop-2.0 >= 2.21.4"
-PKG_CHECK_MODULES(LIBSLAB, $COMMON_MODULES gnome-desktop-2.0 librsvg-2.0 libgnome-menu pango eel-2.0)
 PKG_CHECK_MODULES(CAPPLET, $COMMON_MODULES)
 PKG_CHECK_MODULES(GNOMECC, $COMMON_MODULES libgnome-menu >= 2.10.1 libpanelapplet-2.0)
 PKG_CHECK_MODULES(GNOME, libgnomeui-2.0)
@@ -106,6 +102,7 @@
 PKG_CHECK_MODULES(GNOME_DESKTOP, gnome-desktop-2.0)
 PKG_CHECK_MODULES(DEFAULT_APPLICATIONS_CAPPLET, libxml-2.0)
 PKG_CHECK_MODULES(METACITY, libmetacity-private >= 2.21.21)
+PKG_CHECK_MODULES(SLAB, libslab)
 PKG_CHECK_MODULES(TYPING, glib-2.0 > 2.11 gconf-2.0 gtk+-2.0)
 PKG_CHECK_MODULES(LIBBACKGROUND, glib-2.0 gobject-2.0 gdk-2.0 gconf-2.0)
 PKG_CHECK_MODULES(LIBSOUNDS, gio-2.0 libgnome-2.0 libgnomeui-2.0)
@@ -446,8 +443,6 @@
 capplets/windows/window-properties.desktop.in
 help/Makefile
 libbackground/Makefile
-libslab/Makefile
-libslab/libslab.pc
 libsounds/Makefile
 libwindow-settings/Makefile
 libwindow-settings/gnome-window-settings-2.0.pc
--- shell/Makefile.am.old	2008-03-16 21:24:50.000000000 +0100
+++ shell/Makefile.am	2008-03-16 21:25:16.000000000 +0100
@@ -1,5 +1,5 @@
 INCLUDES =					\
-	-I$(top_srcdir)/libslab			\
+	$(SLAB_CFLAGS)				\
 	$(GNOMECC_CFLAGS)
 
 bin_PROGRAMS = gnome-control-center 
@@ -9,7 +9,7 @@
 
 gnome_control_center_LDADD =						\
 	$(GNOMECC_LIBS)							\
-	$(top_builddir)/libslab/libslab.la
+	$(SLAB_LIBS)
 
 AM_CPPFLAGS =							\
 	-DGNOMELOCALEDIR="\"$(datadir)/locale\""
