--- capplets/default-applications/gnome-default-applications-properties-structs.c.orig	2004-03-24 23:33:04.000000000 +0100
+++ capplets/default-applications/gnome-default-applications-properties-structs.c	2004-03-24 23:35:29.000000000 +0100
@@ -6,6 +6,7 @@
 	gboolean in_path;
 };
 BrowserDescription possible_browsers[] = {
+        { N_("Debian Sensible Browser"),"sensible-browser",    "sensible-browser %s",    FALSE, FALSE },
         { N_("Epiphany"), 		"epiphany",	"epiphany %s",		FALSE, FALSE },
         { N_("Galeon"), 		"galeon",	"galeon %s",		FALSE, FALSE },
         { N_("Encompass"), 		"encompass",	"encompass %s",		FALSE, FALSE },
@@ -47,6 +48,7 @@
 	gboolean in_path;
 };
 TerminalDescription possible_terminals[] = { 
+        { N_("Debian Terminal Emulator"), "x-terminal-emulator", "-e", FALSE },
         { N_("Gnome Terminal"),		"gnome-terminal",	"-x", FALSE },
         { N_("Standard XTerminal"),	"xterm",		"-e", FALSE },
         { N_("NXterm"),			"nxterm",		"-e", FALSE },
