--- capplets/default-applications/gnome-default-applications-properties-structs.c.orig	2004-11-08 22:38:55.000000000 +0100
+++ capplets/default-applications/gnome-default-applications-properties-structs.c	2004-11-08 22:40:20.000000000 +0100
@@ -6,10 +6,11 @@
 	gboolean in_path;
 };
 BrowserDescription possible_browsers[] = {
+        { N_("Debian Sensible Browser"),"sensible-browser",    "sensible-browser %s",    FALSE, FALSE },
         { N_("Epiphany"), 		"epiphany",	"epiphany %s",		FALSE, FALSE },
         { N_("Galeon"), 		"galeon",	"galeon %s",		FALSE, FALSE },
         { N_("Encompass"), 		"encompass",	"encompass %s",		FALSE, FALSE },
-        { N_("Firebird/FireFox"),  "mozilla-firebird",	"mozilla-firebird %s",	FALSE, FALSE },
+        { N_("FireFox"),		"mozilla-firefox",	"mozilla-firefox %s",	FALSE, FALSE },
         { N_("Mozilla/Netscape 6"), 	"mozilla-1.6",	"mozilla-1.6 %s",	FALSE, FALSE },
         { N_("Mozilla/Netscape 6"), 	"mozilla",	"mozilla %s",		FALSE, FALSE },
         { N_("Netscape Communicator"), 	"netscape",	"netscape %s",		FALSE, FALSE },
@@ -31,6 +32,7 @@
 	 * there is only one (the first entry in this list) Evolution entry 
 	 * in the list shown to the user
 	 */
+	{ N_("Evolution Mail Reader"),		"evolution-2.0",	"evolution-2.0 %s",	FALSE, FALSE, },
 	{ N_("Evolution Mail Reader"),		"evolution-1.6",	"evolution-1.6 %s",	FALSE, FALSE, },
 	{ N_("Evolution Mail Reader"),		"evolution-1.5",	"evolution-1.5 %s",	FALSE, FALSE, },
         { N_("Evolution Mail Reader"),		"evolution-1.4",	"evolution-1.4 %s",	FALSE, FALSE, },
@@ -38,6 +40,7 @@
 	{ N_("Balsa"),        			"balsa",		"balsa --compose=%s",	FALSE, FALSE },
 	{ N_("KMail"),        			"kmail",		"kmail %s",		FALSE, FALSE },
 	{ N_("Mozilla Mail"), 			"mozilla",		"mozilla -mail %s",	FALSE, FALSE},
+	{ N_("Thunderbird"),			"mozilla-thunderbird",	"mozilla-thunderbird -compose %s",     FALSE, FALSE},
         { N_("Mutt") , 	  			"mutt",			"mutt %s",		TRUE,  FALSE },
 
 };
@@ -49,6 +52,7 @@
 	gboolean in_path;
 };
 TerminalDescription possible_terminals[] = { 
+        { N_("Debian Terminal Emulator"), "x-terminal-emulator", "-e", FALSE },
         { N_("Gnome Terminal"),		"gnome-terminal",	"-x", FALSE },
         { N_("Standard XTerminal"),	"xterm",		"-e", FALSE },
         { N_("NXterm"),			"nxterm",		"-e", FALSE },
