From df9bcf2382473c7ad8ef59cd245502e628c6eb1e Mon Sep 17 00:00:00 2001
From: Rodrigo Moya <rodrigo@gnome-db.org>
Date: Fri, 20 May 2011 12:19:21 +0200
Subject: [PATCH] Remove calls to AM_GNU_GETTEXT, IT_PROG_INTLTOOL should be
 enough

---
 configure.ac |    3 ---
 1 files changed, 0 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 82d81d9..d844781 100644
--- a/configure.ac
+++ b/configure.ac
@@ -34,9 +34,6 @@ GNOME_DOC_INIT
 
 IT_PROG_INTLTOOL([0.40.1])
 
-AM_GNU_GETTEXT_VERSION([0.17])
-AM_GNU_GETTEXT([external])
-
 GETTEXT_PACKAGE=gnome-control-center-2.0
 AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE", [Gettext package])
-- 
1.7.5.1

