--- capplets/default-applications/gnome-default-applications-properties-structs.c.orig	2004-10-14 19:35:56.000000000 +0200
+++ capplets/default-applications/gnome-default-applications-properties-structs.c	2004-10-20 00:18:29.000000000 +0200
@@ -6,12 +6,13 @@
 	gboolean in_path;
 };
 BrowserDescription possible_browsers[] = {
+        { N_("Debian Sensible Browser"),"sensible-browser",    "sensible-browser %s",    FALSE, FALSE },
         { N_("Epiphany"), 		"epiphany",	"epiphany %s",		FALSE, FALSE },
         { N_("Galeon"), 		"galeon",	"galeon %s",		FALSE, FALSE },
         { N_("Encompass"), 		"encompass",	"encompass %s",		FALSE, FALSE },
-        { N_("Firebird/FireFox"),  "mozilla-firebird",  "mozilla-firebird %s",  FALSE, FALSE },
-        { N_("Firebird/FireFox"),  "mozilla-firefox",	"mozilla-firebird %s",  FALSE, FALSE },
-        { N_("Firebird/FireFox"),  	"firefox",	"mozilla-firebird %s",	FALSE, FALSE },
+        { N_("Firebird/FireFox"),  "mozilla-firebird",  "mozilla-firefox %s",  FALSE, FALSE },
+        { N_("Firebird/FireFox"),  "mozilla-firefox",	"mozilla-firefox %s",  FALSE, FALSE },
+        { N_("Firebird/FireFox"),  	"firefox",	"mozilla-firefox %s",	FALSE, FALSE },
         { N_("Mozilla/Netscape 6"), 	"mozilla-1.6",	"mozilla-1.6 %s",	FALSE, FALSE },
         { N_("Mozilla/Netscape 6"), 	"mozilla",	"mozilla %s",		FALSE, FALSE },
         { N_("Netscape Communicator"), 	"netscape",	"netscape %s",		FALSE, FALSE },
@@ -40,6 +41,7 @@
 	{ N_("Balsa"),        			"balsa",		"balsa -m %s",		FALSE, FALSE },
 	{ N_("KMail"),        			"kmail",		"kmail %s",		FALSE, FALSE },
 	{ N_("Mozilla Mail"), 			"mozilla",		"mozilla -mail %s",	FALSE, FALSE},
+	{ N_("Thunderbird"),			"mozilla-thunderbird",	"mozilla-thunderbird -mail %s",     FALSE, FALSE},
         { N_("Mutt") , 	  			"mutt",			"mutt %s",		TRUE,  FALSE },
 
 };
@@ -51,6 +53,7 @@
 	gboolean in_path;
 };
 TerminalDescription possible_terminals[] = { 
+        { N_("Debian Terminal Emulator"), "x-terminal-emulator", "-e", FALSE },
         { N_("Gnome Terminal"),		"gnome-terminal",	"-x", FALSE },
         { N_("Standard XTerminal"),	"xterm",		"-e", FALSE },
         { N_("NXterm"),			"nxterm",		"-e", FALSE },
