diff --git a/shell/gnome-control-center.desktop.in.in b/shell/gnome-control-center.desktop.in.in
index a7160bd..b8c543f 100644
--- a/shell/gnome-control-center.desktop.in.in
+++ b/shell/gnome-control-center.desktop.in.in
@@ -5,7 +5,7 @@ Exec=gnome-control-center --overview
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GNOME;GTK;System;
+Categories=GNOME;GTK;Settings;
 OnlyShowIn=GNOME;Unity;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
