Description: Remove the Debian entries from the default applications
Author: ?

diff -Nur -x '*.orig' -x '*~' gnome-control-center-2.22.1/capplets/default-applications/gnome-default-applications.xml.in gnome-control-center-2.22.1.new/capplets/default-applications/gnome-default-applications.xml.in
--- gnome-control-center-2.22.1/capplets/default-applications/gnome-default-applications.xml.in	2008-04-11 16:44:46.000000000 +0100
+++ gnome-control-center-2.22.1.new/capplets/default-applications/gnome-default-applications.xml.in	2008-04-14 15:57:42.000000000 +0100
@@ -16,14 +16,6 @@
       <win-command>opera -newwindow %s</win-command>
     </web-browser>
     <web-browser>
-      <_name>Debian Sensible Browser</_name>
-      <executable>sensible-browser</executable>
-      <command>sensible-browser %s</command>
-      <icon-name></icon-name>
-      <run-in-terminal>false</run-in-terminal>
-      <netscape-remote>false</netscape-remote>
-    </web-browser>
-    <web-browser>
       <_name>Epiphany Web Browser</_name>
       <executable>epiphany</executable>
       <command>epiphany %s</command>
@@ -245,13 +237,6 @@
 
   <terminals>
     <terminal>
-      <_name>Debian Terminal Emulator</_name>
-      <executable>x-terminal-emulator</executable>
-      <command>x-terminal-emulator</command>
-      <icon-name>gnome-mime-application-x-executable</icon-name>
-      <exec-flag>-e</exec-flag>
-    </terminal>
-    <terminal>
       <_name>GNOME Terminal</_name>
       <executable>gnome-terminal</executable>
       <command>gnome-terminal</command>
