Index: gnome-control-center-2.24.0.1/configure.in
===================================================================
--- gnome-control-center-2.24.0.1.orig/configure.in	2009-04-16 10:31:56.309352617 +0200
+++ gnome-control-center-2.24.0.1/configure.in	2009-04-16 10:32:14.016853698 +0200
@@ -98,7 +98,7 @@ PKG_CHECK_MODULES(CAPPLET, $COMMON_MODUL
 PKG_CHECK_MODULES(GNOMECC, $COMMON_MODULES libgnome-menu >= 2.10.1 libpanelapplet-2.0)
 PKG_CHECK_MODULES(GNOME, libgnomeui-2.0)
 PKG_CHECK_MODULES(DBUS, dbus-1 dbus-glib-1)
-PKG_CHECK_MODULES(GNOME_DESKTOP, gnome-desktop-2.0)
+PKG_CHECK_MODULES(GNOME_DESKTOP, gnome-desktop-2.0 libgnomeui-2.0)
 PKG_CHECK_MODULES(DEFAULT_APPLICATIONS_CAPPLET, libxml-2.0)
 PKG_CHECK_MODULES(METACITY, libmetacity-private >= 2.23.1)
 PKG_CHECK_MODULES(SLAB, libslab)
