--- control-center-2.17.91/gnome-settings-daemon/gnome-settings-xrdb.c	2007-02-05 12:11:42.000000000 +0100
+++ control-center-2.17.91.new/gnome-settings-daemon/gnome-settings-xrdb.c	2007-02-13 10:25:09.000000000 +0100
@@ -32,7 +32,7 @@
 #include "gnome-settings-daemon.h"
 #include "gnome-settings-xrdb.h"
 
-#define SYSTEM_AD_DIR GNOMECC_DATA_DIR "/xrdb"
+#define SYSTEM_AD_DIR "/etc/gnome/config"
 #define GENERAL_AD SYSTEM_AD_DIR "/General.ad"
 #define USER_AD_DIR ".gnome2/xrdb"
 #define USER_X_RESOURCES ".Xresources"
