--- gnome-settings-daemon/gnome-settings-xrdb.c.orig	2003-10-27 15:36:50.000000000 +0100
+++ gnome-settings-daemon/gnome-settings-xrdb.c	2003-10-27 15:37:52.000000000 +0100
@@ -32,7 +32,7 @@
 #include "gnome-settings-daemon.h"
 #include "gnome-settings-xrdb.h"
 
-#define SYSTEM_AD_DIR GNOMECC_DATA_DIR "/xrdb"
+#define SYSTEM_AD_DIR "/etc/gnome/config"
 #define GENERAL_AD SYSTEM_AD_DIR "/General.ad"
 #define USER_AD_DIR ".gnome2/xrdb"
 #define USER_X_RESOURCES ".Xresources"
 
