--- capplets/accessibility/at-properties/main.c.orig	2008-10-24 21:32:11.628329507 +0200
+++ capplets/accessibility/at-properties/main.c	2008-10-24 21:39:59.927826441 +0200
@@ -85,8 +85,12 @@
 
 			gtk_main_quit ();
 		}
-		gnome_client_request_save (client, GNOME_SAVE_GLOBAL, TRUE,
-					   GNOME_INTERACT_ANY, FALSE, TRUE);
+		const gchar *const argv[] = { "gnome-panel-logout", NULL };
+		if (!g_spawn_async (NULL, (gchar **) argv, NULL, G_SPAWN_SEARCH_PATH,
+		                    NULL, NULL, NULL, NULL)) {
+			gnome_client_request_save (client, GNOME_SAVE_GLOBAL, TRUE,
+			                           GNOME_INTERACT_ANY, FALSE, TRUE);
+		}
 	}
 }
 
