Index: gnome-control-center-2.24.0.1/capplets/accessibility/at-properties/main.c
===================================================================
--- gnome-control-center-2.24.0.1.orig/capplets/accessibility/at-properties/main.c	2008-09-24 18:36:22.000000000 +0200
+++ gnome-control-center-2.24.0.1/capplets/accessibility/at-properties/main.c	2008-12-28 13:22:48.211860165 +0100
@@ -94,8 +94,12 @@ cb_dialog_response (GtkDialog *dialog, g
 
 			gtk_main_quit ();
 		}
-		gnome_client_request_save (client, GNOME_SAVE_GLOBAL, TRUE,
-					   GNOME_INTERACT_ANY, FALSE, TRUE);
+		const gchar *const argv[] = { "gnome-panel-logout", NULL };
+		if (!g_spawn_async (NULL, (gchar **) argv, NULL, G_SPAWN_SEARCH_PATH,
+		                    NULL, NULL, NULL, NULL)) {
+			gnome_client_request_save (client, GNOME_SAVE_GLOBAL, TRUE,
+			                           GNOME_INTERACT_ANY, FALSE, TRUE);
+		}
 	}
 }
 
