Index: gnome-control-center-3.1.91/panels/wacom/gnome-wacom-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.1.91.orig/panels/wacom/gnome-wacom-panel.desktop.in.in	2011-09-06 23:06:36.851400013 +0200
+++ gnome-control-center-3.1.91/panels/wacom/gnome-wacom-panel.desktop.in.in	2011-09-06 23:06:44.319400373 +0200
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;
+OnlyShowIn=GNOME;Unity;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=wacom
