--- capplets/keybindings/gnome-keybinding-properties.c.orig	2004-04-07 01:33:59.000000000 +0200
+++ capplets/keybindings/gnome-keybinding-properties.c	2004-04-07 01:36:53.000000000 +0200
@@ -672,11 +672,11 @@
 		      -1);
 
   if (tmp_key_entry != NULL &&
-      strcmp (key_entry->gconf_key, tmp_key_entry->gconf_key) != 0 &&
-      ((key_entry->keyval == tmp_key_entry->keyval &&
-	key_entry->mask   == tmp_key_entry->mask) ||
-       key_entry->keycode == tmp_key_entry->keycode))
-      /* be sure we don't claim a key is a dup of itself */
+      key_entry->keyval == tmp_key_entry->keyval &&
+      key_entry->mask   == tmp_key_entry->mask &&
+      key_entry->keycode == tmp_key_entry->keycode &&
+       /* be sure we don't claim a key is a dup of itself */
+      strcmp (key_entry->gconf_key, tmp_key_entry->gconf_key) != 0)
     {
       key_entry->editable = FALSE;
       key_entry->gconf_key = tmp_key_entry->gconf_key;
