Index: gnome-control-center-3.4.0/configure.ac
===================================================================
--- gnome-control-center-3.4.0.orig/configure.ac	2012-03-26 18:52:56.000000000 +0200
+++ gnome-control-center-3.4.0/configure.ac	2012-04-05 12:09:36.665486863 +0200
@@ -93,7 +93,7 @@
 PKG_CHECK_MODULES(LIBSHORTCUTS, $COMMON_MODULES x11)
 PKG_CHECK_MODULES(SHELL, $COMMON_MODULES libgnome-menu-3.0 gio-unix-2.0 x11)
 PKG_CHECK_MODULES(BACKGROUND_PANEL, $COMMON_MODULES libxml-2.0 gnome-desktop-3.0
-                  gdk-pixbuf-2.0 >= $GDKPIXBUF_REQUIRED_VERSION)
+                  gdk-pixbuf-2.0 >= $GDKPIXBUF_REQUIRED_VERSION gconf-2.0)
 PKG_CHECK_MODULES(DATETIME_PANEL, $COMMON_MODULES
                   gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION
                   polkit-gobject-1 >= $POLKIT_REQUIRED_VERSION
Index: gnome-control-center-3.4.0/panels/background/background.ui
===================================================================
--- gnome-control-center-3.4.0.orig/panels/background/background.ui	2012-02-14 11:13:17.000000000 +0100
+++ gnome-control-center-3.4.0/panels/background/background.ui	2012-04-05 12:08:51.773264253 +0200
@@ -1,7 +1,6 @@
-<?xml version="1.0"?>
+<?xml version="1.0" encoding="UTF-8"?>
 <interface>
   <requires lib="gtk+" version="2.16"/>
-  <!-- interface-naming-policy project-wide -->
   <object class="GtkListStore" id="sources-liststore">
     <columns>
       <!-- column-name source-name -->
@@ -12,321 +11,921 @@
       <column type="gpointer"/>
     </columns>
   </object>
-  <object class="GtkHBox" id="background-panel">
-    <property name="visible">True</property>
-    <property name="border_width">10</property>
-    <property name="spacing">12</property>
+  <object class="GtkListStore" id="style-liststore">
+    <columns>
+      <!-- column-name name -->
+      <column type="gchararray"/>
+      <!-- column-name value -->
+      <column type="gint"/>
+    </columns>
+    <data>
+      <row>
+        <col id="0" translatable="yes">Tile</col>
+        <col id="1">1</col>
+      </row>
+      <row>
+        <col id="0" translatable="yes">Zoom</col>
+        <col id="1">5</col>
+      </row>
+      <row>
+        <col id="0" translatable="yes">Center</col>
+        <col id="1">2</col>
+      </row>
+      <row>
+        <col id="0" translatable="yes">Scale</col>
+        <col id="1">3</col>
+      </row>
+      <row>
+        <col id="0" translatable="yes">Fill</col>
+        <col id="1">4</col>
+      </row>
+      <row>
+        <col id="0" translatable="yes">Span</col>
+        <col id="1">6</col>
+      </row>
+    </data>
+  </object>
+  <object class="GtkListStore" id="theme-list-store">
+    <columns>
+      <!-- column-name THEME_ID -->
+      <column type="gchararray"/>
+      <!-- column-name THEME_NAME -->
+      <column type="gchararray"/>
+    </columns>
+  </object>
+  <object class="GtkWindow" id="window1">
+    <property name="can_focus">False</property>
     <child>
-      <object class="GtkVBox" id="vbox2">
+      <object class="GtkNotebook" id="main-notebook">
         <property name="visible">True</property>
-        <property name="orientation">vertical</property>
-        <property name="spacing">0</property>
-        <child>
-          <object class="GtkComboBox" id="sources-combobox">
-            <property name="visible">True</property>
-            <property name="model">sources-liststore</property>
-            <child>
-              <object class="GtkCellRendererText" id="cellrenderertext2"/>
-              <attributes>
-                <attribute name="text">0</attribute>
-              </attributes>
-            </child>
-          </object>
-          <packing>
-            <property name="expand">False</property>
-            <property name="position">0</property>
-            <property name="padding">6</property>
-          </packing>
-        </child>
+        <property name="can_focus">True</property>
+        <property name="border_width">12</property>
         <child>
-          <object class="GtkScrolledWindow" id="scrolledwindow1">
-            <property name="width_request">220</property>
-            <property name="height_request">380</property>
+          <object class="GtkBox" id="background-panel">
             <property name="visible">True</property>
-            <property name="can_focus">True</property>
-            <property name="hscrollbar_policy">automatic</property>
-            <property name="vscrollbar_policy">automatic</property>
-            <property name="shadow_type">in</property>
+            <property name="can_focus">False</property>
+            <property name="margin_top">3</property>
+            <property name="margin_bottom">3</property>
+            <property name="orientation">vertical</property>
+            <property name="spacing">3</property>
             <child>
-              <object class="GtkIconView" id="backgrounds-iconview">
+              <object class="GtkHBox" id="background-container">
                 <property name="visible">True</property>
-                <property name="can_focus">True</property>
-                <property name="row_spacing">0</property>
-                <property name="column_spacing">0</property>
-                <property name="margin">0</property>
+                <property name="can_focus">False</property>
+                <property name="margin_right">4</property>
+                <property name="border_width">10</property>
+                <property name="spacing">12</property>
                 <child>
-                  <object class="GtkCellRendererPixbuf" id="pixbuf-renderer">
-                    <property name="stock_size">6</property>
+                  <object class="GtkVBox" id="vbox3">
+                    <property name="visible">True</property>
+                    <property name="can_focus">False</property>
+                    <child>
+                      <object class="GtkLabel" id="label1">
+                        <property name="visible">True</property>
+                        <property name="can_focus">False</property>
+                        <property name="xalign">0</property>
+                        <property name="label" translatable="yes">&lt;b&gt;Background&lt;/b&gt;</property>
+                        <property name="use_markup">True</property>
+                      </object>
+                      <packing>
+                        <property name="expand">False</property>
+                        <property name="fill">True</property>
+                        <property name="position">0</property>
+                      </packing>
+                    </child>
+                    <child>
+                      <object class="GtkDrawingArea" id="preview-area">
+                        <property name="visible">True</property>
+                        <property name="can_focus">False</property>
+                      </object>
+                      <packing>
+                        <property name="expand">True</property>
+                        <property name="fill">True</property>
+                        <property name="position">1</property>
+                      </packing>
+                    </child>
+                    <child>
+                      <object class="GtkHBox" id="heading-hbox">
+                        <property name="visible">True</property>
+                        <property name="can_focus">False</property>
+                        <property name="halign">center</property>
+                        <property name="valign">end</property>
+                        <property name="spacing">4</property>
+                        <child>
+                          <object class="GtkLabel" id="background-label">
+                            <property name="visible">True</property>
+                            <property name="can_focus">False</property>
+                            <property name="halign">center</property>
+                            <property name="label">Current background</property>
+                            <attributes>
+                              <attribute name="style" value="italic"/>
+                            </attributes>
+                          </object>
+                          <packing>
+                            <property name="expand">False</property>
+                            <property name="fill">True</property>
+                            <property name="position">0</property>
+                          </packing>
+                        </child>
+                        <child>
+                          <object class="GtkLabel" id="size_label">
+                            <property name="width_request">15</property>
+                            <property name="visible">True</property>
+                            <property name="can_focus">False</property>
+                            <property name="label">(2560 x 2560)</property>
+                            <attributes>
+                              <attribute name="style" value="italic"/>
+                            </attributes>
+                          </object>
+                          <packing>
+                            <property name="expand">False</property>
+                            <property name="fill">True</property>
+                            <property name="position">1</property>
+                          </packing>
+                        </child>
+                      </object>
+                      <packing>
+                        <property name="expand">False</property>
+                        <property name="fill">True</property>
+                        <property name="padding">6</property>
+                        <property name="position">2</property>
+                      </packing>
+                    </child>
+                    <child>
+                      <object class="GtkHBox" id="bottom-hbox">
+                        <property name="visible">True</property>
+                        <property name="can_focus">False</property>
+                        <property name="spacing">12</property>
+                        <child>
+                          <object class="GtkHBox" id="hbox2">
+                            <property name="visible">True</property>
+                            <property name="can_focus">False</property>
+                            <property name="spacing">2</property>
+                            <child>
+                              <object class="GtkImage" id="slide_image">
+                                <property name="visible">True</property>
+                                <property name="can_focus">False</property>
+                                <property name="icon_name">slideshow-symbolic</property>
+                              </object>
+                              <packing>
+                                <property name="expand">False</property>
+                                <property name="fill">True</property>
+                                <property name="position">0</property>
+                              </packing>
+                            </child>
+                            <child>
+                              <object class="GtkLabel" id="slide-label">
+                                <property name="visible">True</property>
+                                <property name="can_focus">False</property>
+                                <property name="xalign">0</property>
+                                <property name="label" translatable="yes" comments="This refers to a slideshow background">Changes throughout the day</property>
+                              </object>
+                              <packing>
+                                <property name="expand">False</property>
+                                <property name="fill">True</property>
+                                <property name="position">1</property>
+                              </packing>
+                            </child>
+                            <child>
+                              <object class="GtkLabel" id="strut">
+                                <property name="visible">True</property>
+                                <property name="can_focus">False</property>
+                                <property name="label"> </property>
+                              </object>
+                              <packing>
+                                <property name="expand">False</property>
+                                <property name="fill">True</property>
+                                <property name="position">1</property>
+                              </packing>
+                            </child>
+                          </object>
+                          <packing>
+                            <property name="expand">True</property>
+                            <property name="fill">True</property>
+                            <property name="position">0</property>
+                          </packing>
+                        </child>
+                        <child>
+                          <object class="GtkHBox" id="edit-hbox">
+                            <property name="visible">True</property>
+                            <property name="can_focus">False</property>
+                            <property name="spacing">6</property>
+                            <child>
+                              <object class="GtkColorButton" id="style-pcolor">
+                                <property name="use_action_appearance">False</property>
+                                <property name="visible">True</property>
+                                <property name="can_focus">True</property>
+                                <property name="receives_default">True</property>
+                                <property name="use_action_appearance">False</property>
+                                <property name="color">#000000000000</property>
+                                <child internal-child="accessible">
+                                  <object class="AtkObject" id="style-pcolor-a11y">
+                                    <property name="accessible-name" translatable="yes">Primary Color</property>
+                                  </object>
+                                </child>
+                              </object>
+                              <packing>
+                                <property name="expand">False</property>
+                                <property name="fill">False</property>
+                                <property name="position">0</property>
+                              </packing>
+                            </child>
+                            <child>
+                              <object class="GtkButton" id="swap-color-button">
+                                <property name="use_action_appearance">False</property>
+                                <property name="visible">True</property>
+                                <property name="can_focus">True</property>
+                                <property name="receives_default">True</property>
+                                <property name="use_action_appearance">False</property>
+                                <child internal-child="accessible">
+                                  <object class="AtkObject" id="swap-color-a11y">
+                                    <property name="accessible-name" translatable="yes">Swap colors</property>
+                                  </object>
+                                </child>
+                                <child>
+                                  <object class="GtkImage" id="image1">
+                                    <property name="visible">True</property>
+                                    <property name="can_focus">False</property>
+                                    <property name="icon_name">object-flip-horizontal-symbolic</property>
+                                  </object>
+                                </child>
+                              </object>
+                              <packing>
+                                <property name="expand">False</property>
+                                <property name="fill">False</property>
+                                <property name="position">1</property>
+                              </packing>
+                            </child>
+                            <child>
+                              <object class="GtkColorButton" id="style-scolor">
+                                <property name="use_action_appearance">False</property>
+                                <property name="visible">True</property>
+                                <property name="can_focus">True</property>
+                                <property name="receives_default">True</property>
+                                <property name="use_action_appearance">False</property>
+                                <property name="color">#000000000000</property>
+                                <child internal-child="accessible">
+                                  <object class="AtkObject" id="style-scolor-a11y">
+                                    <property name="accessible-name" translatable="yes">Secondary color</property>
+                                  </object>
+                                </child>
+                              </object>
+                              <packing>
+                                <property name="expand">False</property>
+                                <property name="fill">False</property>
+                                <property name="position">2</property>
+                              </packing>
+                            </child>
+                            <child>
+                              <object class="GtkComboBox" id="style-combobox">
+                                <property name="visible">True</property>
+                                <property name="can_focus">False</property>
+                                <property name="model">style-liststore</property>
+                                <property name="active">0</property>
+                                <child>
+                                  <object class="GtkCellRendererText" id="cellrenderertext1"/>
+                                  <attributes>
+                                    <attribute name="text">0</attribute>
+                                  </attributes>
+                                </child>
+                              </object>
+                              <packing>
+                                <property name="expand">False</property>
+                                <property name="fill">False</property>
+                                <property name="pack_type">end</property>
+                                <property name="position">3</property>
+                              </packing>
+                            </child>
+                          </object>
+                          <packing>
+                            <property name="expand">False</property>
+                            <property name="fill">False</property>
+                            <property name="pack_type">end</property>
+                            <property name="position">1</property>
+                          </packing>
+                        </child>
+                      </object>
+                      <packing>
+                        <property name="expand">False</property>
+                        <property name="fill">True</property>
+                        <property name="position">3</property>
+                      </packing>
+                    </child>
                   </object>
-                  <attributes>
-                    <attribute name="gicon">0</attribute>
-                  </attributes>
+                  <packing>
+                    <property name="expand">True</property>
+                    <property name="fill">True</property>
+                    <property name="position">0</property>
+                  </packing>
                 </child>
                 <child>
-                  <!-- this cell renderer is here just to have
-                       the text spoken in orca -->
-                  <object class="GtkCellRendererText" id="text-renderer">
-                    <property name="visible">False</property>
+                  <object class="GtkVBox" id="vbox2">
+                    <property name="visible">True</property>
+                    <property name="can_focus">False</property>
+                    <child>
+                      <object class="GtkComboBox" id="sources-combobox">
+                        <property name="visible">True</property>
+                        <property name="can_focus">False</property>
+                        <property name="model">sources-liststore</property>
+                        <child>
+                          <object class="GtkCellRendererText" id="cellrenderertext2"/>
+                          <attributes>
+                            <attribute name="text">0</attribute>
+                          </attributes>
+                        </child>
+                      </object>
+                      <packing>
+                        <property name="expand">False</property>
+                        <property name="fill">True</property>
+                        <property name="position">0</property>
+                      </packing>
+                    </child>
+                    <child>
+                      <object class="GtkScrolledWindow" id="scrolledwindow1">
+                        <property name="width_request">262</property>
+                        <property name="height_request">260</property>
+                        <property name="visible">True</property>
+                        <property name="can_focus">True</property>
+                        <property name="shadow_type">in</property>
+                        <child>
+                          <object class="GtkIconView" id="backgrounds-iconview">
+                            <property name="visible">True</property>
+                            <property name="can_focus">True</property>
+                            <property name="row_spacing">0</property>
+                            <property name="column_spacing">0</property>
+                            <child>
+                              <object class="GtkCellRendererPixbuf" id="pixbuf-renderer">
+                                <property name="stock_size">6</property>
+                              </object>
+                              <attributes>
+                                <attribute name="gicon">0</attribute>
+                              </attributes>
+                            </child>
+                            <child>
+                              <object class="GtkCellRendererText" id="text-renderer">
+                                <property name="visible">False</property>
+                              </object>
+                              <attributes>
+                                <attribute name="text">2</attribute>
+                              </attributes>
+                            </child>
+                          </object>
+                        </child>
+                      </object>
+                      <packing>
+                        <property name="expand">True</property>
+                        <property name="fill">True</property>
+                        <property name="position">1</property>
+                      </packing>
+                    </child>
+                    <child>
+                      <object class="GtkToolbar" id="toolbar1">
+                        <property name="visible">True</property>
+                        <property name="can_focus">False</property>
+                        <property name="icon_size">1</property>
+                        <property name="toolbar-style">icons</property>
+                        <style>
+                          <class name="inline-toolbar"/>
+                        </style>
+                        <child>
+                          <object class="GtkToolButton" id="add_button">
+                            <property name="use_action_appearance">False</property>
+                            <property name="visible">True</property>
+                            <property name="can_focus">False</property>
+                            <property name="use_action_appearance">False</property>
+                            <property name="label" translatable="yes">Add wallpaper</property>
+                            <property name="use_underline">True</property>
+                            <property name="icon_name">list-add-symbolic</property>
+                          </object>
+                          <packing>
+                            <property name="expand">False</property>
+                            <property name="homogeneous">True</property>
+                          </packing>
+                        </child>
+                        <child>
+                          <object class="GtkToolButton" id="remove_button">
+                            <property name="use_action_appearance">False</property>
+                            <property name="visible">True</property>
+                            <property name="sensitive">False</property>
+                            <property name="can_focus">False</property>
+                            <property name="use_action_appearance">False</property>
+                            <property name="label" translatable="yes">Remove wallpaper</property>
+                            <property name="use_underline">True</property>
+                            <property name="icon_name">list-remove-symbolic</property>
+                          </object>
+                          <packing>
+                            <property name="expand">False</property>
+                            <property name="homogeneous">True</property>
+                          </packing>
+                        </child>
+                      </object>
+                      <packing>
+                        <property name="expand">False</property>
+                        <property name="fill">True</property>
+                        <property name="position">2</property>
+                      </packing>
+                    </child>
+                    <child>
+                      <object class="GtkCheckButton" id="add_wallpaper_dots">
+                        <property name="label" translatable="yes">Add dots</property>
+                        <property name="use_action_appearance">False</property>
+                        <property name="no_show_all">True</property>
+                        <property name="can_focus">True</property>
+                        <property name="receives_default">False</property>
+                        <property name="use_action_appearance">False</property>
+                        <property name="xalign">0</property>
+                        <property name="draw_indicator">True</property>
+                      </object>
+                      <packing>
+                        <property name="expand">True</property>
+                        <property name="fill">True</property>
+                        <property name="position">3</property>
+                      </packing>
+                    </child>
                   </object>
-                  <attributes>
-                    <attribute name="text">2</attribute>
-                  </attributes>
+                  <packing>
+                    <property name="expand">False</property>
+                    <property name="fill">True</property>
+                    <property name="position">1</property>
+                  </packing>
                 </child>
               </object>
-            </child>
-          </object>
-          <packing>
-            <property name="expand">True</property>
-            <property name="fill">True</property>
-            <property name="position">1</property>
-          </packing>
-        </child>
-        <child>
-          <object class="GtkToolbar" id="toolbar1">
-            <property name="visible">True</property>
-            <property name="can_focus">False</property>
-            <property name="icon_size">1</property>
-            <property name="toolbar-style">icons</property>
-            <style>
-              <class name="inline-toolbar"/>
-            </style>
-            <child>
-              <object class="GtkToolButton" id="add_button">
-                <property name="visible">True</property>
-                <property name="label" translatable="yes">Add wallpaper</property>
-                <property name="use_action_appearance">False</property>
-                <property name="use_underline">True</property>
-                <property name="icon_name">list-add-symbolic</property>
-              </object>
               <packing>
                 <property name="expand">False</property>
-                <property name="homogeneous">True</property>
+                <property name="fill">True</property>
+                <property name="position">0</property>
               </packing>
             </child>
             <child>
-              <object class="GtkToolButton" id="remove_button">
+              <object class="GtkSeparator" id="separator1">
                 <property name="visible">True</property>
-                <property name="label" translatable="yes">Remove wallpaper</property>
-                <property name="use_action_appearance">False</property>
-                <property name="use_underline">True</property>
-                <property name="icon_name">list-remove-symbolic</property>
-                <property name="sensitive">False</property>
+                <property name="can_focus">False</property>
+                <property name="margin_left">5</property>
+                <property name="margin_right">13</property>
               </object>
               <packing>
                 <property name="expand">False</property>
-                <property name="homogeneous">True</property>
+                <property name="fill">True</property>
+                <property name="position">1</property>
               </packing>
             </child>
-          </object>
-          <packing>
-            <property name="expand">False</property>
-            <property name="fill">True</property>
-            <property name="position">2</property>
-          </packing>
-        </child>
-      </object>
-      <packing>
-        <property name="expand">False</property>
-        <property name="fill">True</property>
-        <property name="position">0</property>
-      </packing>
-    </child>
-    <child>
-      <object class="GtkVBox" id="vbox3">
-        <property name="visible">True</property>
-        <property name="can_focus">False</property>
-        <property name="spacing">6</property>
-        <child>
-          <object class="GtkHBox" id="heading-hbox">
-            <property name="visible">True</property>
-            <property name="can_focus">False</property>
-            <property name="spacing">24</property>
             <child>
-              <object class="GtkLabel" id="background-label">
+              <object class="GtkBox" id="theme-box">
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
-                <property name="xalign">0</property>
-                <property name="label">Current background</property>
-                <attributes>
-                  <attribute name="weight" value="bold"/>
-                </attributes>
+                <property name="margin_bottom">8</property>
+                <property name="spacing">12</property>
+                <child>
+                  <object class="GtkLabel" id="theme-label">
+                    <property name="visible">True</property>
+                    <property name="can_focus">False</property>
+                    <property name="xalign">0</property>
+                    <property name="xpad">6</property>
+                    <property name="label" translatable="yes">&lt;b&gt;Theme&lt;/b&gt;</property>
+                    <property name="use_markup">True</property>
+                  </object>
+                  <packing>
+                    <property name="expand">False</property>
+                    <property name="fill">True</property>
+                    <property name="padding">6</property>
+                    <property name="position">0</property>
+                  </packing>
+                </child>
+                <child>
+                  <object class="GtkComboBox" id="theme-selector">
+                    <property name="width_request">268</property>
+                    <property name="visible">True</property>
+                    <property name="can_focus">False</property>
+                    <property name="margin_right">1</property>
+                    <property name="model">theme-list-store</property>
+                    <property name="id_column">0</property>
+                    <child>
+                      <object class="GtkCellRendererText" id="theme-name-renderer"/>
+                      <attributes>
+                        <attribute name="markup">1</attribute>
+                      </attributes>
+                    </child>
+                  </object>
+                  <packing>
+                    <property name="expand">False</property>
+                    <property name="fill">True</property>
+                    <property name="padding">9</property>
+                    <property name="pack_type">end</property>
+                    <property name="position">1</property>
+                  </packing>
+                </child>
               </object>
               <packing>
                 <property name="expand">False</property>
-                <property name="position">0</property>
+                <property name="fill">True</property>
+                <property name="position">2</property>
               </packing>
             </child>
             <child>
-              <object class="GtkLabel" id="size_label">
+              <object class="GtkSeparator" id="unity-separator1">
                 <property name="visible">True</property>
-                <property name="xalign">1</property>
-                <property name="label">1280x700px</property>
+                <property name="can_focus">False</property>
+                <property name="margin_left">5</property>
+                <property name="margin_right">13</property>
               </object>
               <packing>
-                <property name="position">1</property>
+                <property name="expand">False</property>
+                <property name="fill">True</property>
+                <property name="position">3</property>
               </packing>
             </child>
-          </object>
-          <packing>
-            <property name="expand">False</property>
-            <property name="position">0</property>
-          </packing>
-        </child>
-        <child>
-          <object class="GtkDrawingArea" id="preview-area">
-            <property name="visible">True</property>
-          </object>
-          <packing>
-            <property name="position">1</property>
-          </packing>
-        </child>
-        <child>
-          <object class="GtkHBox" id="bottom-hbox">
-            <property name="visible">True</property>
-            <property name="spacing">12</property>
             <child>
-              <object class="GtkHBox" id="hbox2">
+              <object class="GtkBox" id="unity-iconsize-box">
                 <property name="visible">True</property>
-                <property name="spacing">2</property>
-                <child>
-                  <object class="GtkImage" id="slide_image">
-                    <property name="visible">True</property>
-                    <property name="icon_name">slideshow-symbolic</property>
-                  </object>
-                  <packing>
-                    <property name="expand">False</property>
-                    <property name="position">0</property>
-                  </packing>
-                </child>
+                <property name="can_focus">False</property>
+                <property name="spacing">12</property>
                 <child>
-                  <object class="GtkLabel" id="slide-label">
+                  <object class="GtkLabel" id="unity-iconsize-label">
                     <property name="visible">True</property>
+                    <property name="can_focus">False</property>
                     <property name="xalign">0</property>
-                    <property name="label" translatable="yes" comments="This refers to a slideshow background">Changes throughout the day</property>
+                    <property name="xpad">6</property>
+                    <property name="label" translatable="yes">&lt;b&gt;Launcher icon size&lt;/b&gt;</property>
+                    <property name="use_markup">True</property>
                   </object>
                   <packing>
                     <property name="expand">False</property>
-                    <property name="position">1</property>
+                    <property name="fill">True</property>
+                    <property name="padding">6</property>
+                    <property name="position">0</property>
                   </packing>
                 </child>
                 <child>
-                  <object class="GtkLabel" id="strut">
+                  <object class="GtkScale" id="unity-iconsize-scale">
+                    <property name="width_request">264</property>
                     <property name="visible">True</property>
-                    <property name="label" translatable="no"> </property>
+                    <property name="can_focus">True</property>
+                    <property name="margin_right">2</property>
+                    <property name="round_digits">0</property>
+                    <property name="digits">0</property>
+                    <property name="value_pos">left</property>
                   </object>
                   <packing>
                     <property name="expand">False</property>
+                    <property name="fill">True</property>
+                    <property name="padding">9</property>
+                    <property name="pack_type">end</property>
                     <property name="position">1</property>
                   </packing>
                 </child>
               </object>
               <packing>
-                <property name="position">0</property>
+                <property name="expand">False</property>
+                <property name="fill">True</property>
+                <property name="position">4</property>
               </packing>
             </child>
+          </object>
+        </child>
+        <child type="tab">
+          <object class="GtkLabel" id="appearance_label">
+            <property name="visible">True</property>
+            <property name="can_focus">False</property>
+            <property name="label" translatable="yes">Look</property>
+          </object>
+          <packing>
+            <property name="tab_fill">False</property>
+          </packing>
+        </child>
+        <child>
+          <object class="GtkBox" id="unity-behavior">
+            <property name="visible">True</property>
+            <property name="can_focus">False</property>
+            <property name="margin_top">3</property>
+            <property name="margin_bottom">3</property>
+            <property name="orientation">vertical</property>
+            <property name="spacing">3</property>
             <child>
-              <object class="GtkHBox" id="edit-hbox">
+              <object class="GtkBox" id="box1">
                 <property name="visible">True</property>
-                <property name="spacing">6</property>
-                <child>
-                  <object class="GtkColorButton" id="style-pcolor">
-                    <property name="visible">True</property>
-                    <property name="can_focus">True</property>
-                    <property name="receives_default">True</property>
-                    <property name="color">#000000000000</property>
-                    <child internal-child="accessible">
-                      <object class="AtkObject" id="style-pcolor-a11y">
-                        <property name="accessible-name" translatable="yes">Primary Color</property>
-                      </object>
-                    </child>
-                  </object>
-                  <packing>
-                    <property name="expand">False</property>
-                    <property name="fill">False</property>
-                    <property name="position">0</property>
-                  </packing>
-                </child>
+                <property name="can_focus">False</property>
+                <property name="orientation">vertical</property>
                 <child>
-                  <object class="GtkButton" id="swap-color-button">
+                  <object class="GtkBox" id="box3">
                     <property name="visible">True</property>
-                    <property name="can_focus">True</property>
-                    <property name="receives_default">True</property>
-                    <property name="use_action_appearance">False</property>
-                    <child internal-child="accessible">
-                      <object class="AtkObject" id="swap-color-a11y">
-                        <property name="accessible-name" translatable="yes">Swap colors</property>
+                    <property name="can_focus">False</property>
+                    <property name="border_width">10</property>
+                    <property name="spacing">12</property>
+                    <child>
+                      <object class="GtkBox" id="box4">
+                        <property name="visible">True</property>
+                        <property name="can_focus">False</property>
+                        <property name="orientation">vertical</property>
+                        <property name="spacing">2</property>
+                        <child>
+                          <object class="GtkLabel" id="label4">
+                            <property name="visible">True</property>
+                            <property name="can_focus">False</property>
+                            <property name="xalign">0</property>
+                            <property name="label" translatable="yes">&lt;b&gt;Auto-hide the Launcher&lt;/b&gt;</property>
+                            <property name="use_markup">True</property>
+                          </object>
+                          <packing>
+                            <property name="expand">False</property>
+                            <property name="fill">True</property>
+                            <property name="position">0</property>
+                          </packing>
+                        </child>
+                        <child>
+                          <object class="GtkLabel" id="label2">
+                            <property name="visible">True</property>
+                            <property name="can_focus">False</property>
+                            <property name="xalign">0</property>
+                            <property name="label" translatable="yes">&lt;span size="small"&gt;The launcher will reveal when moving the pointer to the defined hot spot.&lt;/span&gt;</property>
+                            <property name="use_markup">True</property>
+                          </object>
+                          <packing>
+                            <property name="expand">False</property>
+                            <property name="fill">True</property>
+                            <property name="position">1</property>
+                          </packing>
+                        </child>
+                        <child>
+                          <object class="GtkBox" id="reveal_location_box">
+                            <property name="visible">True</property>
+                            <property name="can_focus">False</property>
+                            <property name="margin_left">10</property>
+                            <property name="margin_top">9</property>
+                            <property name="orientation">vertical</property>
+                            <child>
+                              <object class="GtkLabel" id="unity_reveal_label">
+                                <property name="visible">True</property>
+                                <property name="can_focus">False</property>
+                                <property name="xalign">0</property>
+                                <property name="label" translatable="yes">Reveal location:</property>
+                              </object>
+                              <packing>
+                                <property name="expand">False</property>
+                                <property name="fill">True</property>
+                                <property name="position">0</property>
+                              </packing>
+                            </child>
+                            <child>
+                              <object class="GtkRadioButton" id="unity_reveal_spot_left">
+                                <property name="label" translatable="yes">Left side</property>
+                                <property name="use_action_appearance">False</property>
+                                <property name="visible">True</property>
+                                <property name="can_focus">True</property>
+                                <property name="receives_default">False</property>
+                                <property name="use_action_appearance">False</property>
+                                <property name="xalign">0</property>
+                                <property name="active">True</property>
+                                <property name="draw_indicator">True</property>
+                              </object>
+                              <packing>
+                                <property name="expand">False</property>
+                                <property name="fill">True</property>
+                                <property name="position">1</property>
+                              </packing>
+                            </child>
+                            <child>
+                              <object class="GtkRadioButton" id="unity_reveal_spot_topleft">
+                                <property name="label" translatable="yes">Top left corner</property>
+                                <property name="use_action_appearance">False</property>
+                                <property name="visible">True</property>
+                                <property name="can_focus">True</property>
+                                <property name="receives_default">False</property>
+                                <property name="use_action_appearance">False</property>
+                                <property name="xalign">0</property>
+                                <property name="active">True</property>
+                                <property name="draw_indicator">True</property>
+                                <property name="group">unity_reveal_spot_left</property>
+                              </object>
+                              <packing>
+                                <property name="expand">False</property>
+                                <property name="fill">True</property>
+                                <property name="position">2</property>
+                              </packing>
+                            </child>
+                            <child>
+                              <object class="GtkRadioButton" id="unity_reveal_spot_otheroption">
+                                <property name="label" translatable="yes">Other reveal option</property>
+                                <property name="use_action_appearance">False</property>
+                                <property name="can_focus">True</property>
+                                <property name="receives_default">False</property>
+                                <property name="no_show_all">True</property>
+                                <property name="use_action_appearance">False</property>
+                                <property name="xalign">0</property>
+                                <property name="draw_indicator">True</property>
+                                <property name="group">unity_reveal_spot_left</property>
+                              </object>
+                              <packing>
+                                <property name="expand">False</property>
+                                <property name="fill">True</property>
+                                <property name="position">3</property>
+                              </packing>
+                            </child>
+                          </object>
+                          <packing>
+                            <property name="expand">False</property>
+                            <property name="fill">True</property>
+                            <property name="position">2</property>
+                          </packing>
+                        </child>
+                        <child>
+                          <object class="GtkBox" id="box6">
+                            <property name="visible">True</property>
+                            <property name="can_focus">False</property>
+                            <property name="spacing">6</property>
+                            <child>
+                              <object class="GtkLabel" id="unity-launcher-sensitivity-label">
+                                <property name="visible">True</property>
+                                <property name="can_focus">False</property>
+                                <property name="label" translatable="yes">Reveal sensitivity</property>
+                              </object>
+                              <packing>
+                                <property name="expand">False</property>
+                                <property name="fill">True</property>
+                                <property name="position">0</property>
+                              </packing>
+                            </child>
+                            <child>
+                              <object class="GtkLabel" id="unity-launcher-sensitivity-low-label">
+                                <property name="visible">True</property>
+                                <property name="can_focus">False</property>
+                                <property name="margin_left">5</property>
+                                <property name="label" translatable="yes">&lt;small&gt;Low&lt;/small&gt;</property>
+                                <property name="use_markup">True</property>
+                              </object>
+                              <packing>
+                                <property name="expand">False</property>
+                                <property name="fill">True</property>
+                                <property name="position">1</property>
+                              </packing>
+                            </child>
+                            <child>
+                              <object class="GtkScale" id="unity-launcher-sensitivity">
+                                <property name="width_request">300</property>
+                                <property name="visible">True</property>
+                                <property name="can_focus">True</property>
+                                <property name="round_digits">0</property>
+                                <property name="digits">0</property>
+                                <property name="draw_value">False</property>
+                              </object>
+                              <packing>
+                                <property name="expand">False</property>
+                                <property name="fill">True</property>
+                                <property name="padding">9</property>
+                                <property name="position">2</property>
+                              </packing>
+                            </child>
+                            <child>
+                              <object class="GtkLabel" id="unity-launcher-sensitivity-high-label">
+                                <property name="visible">True</property>
+                                <property name="can_focus">False</property>
+                                <property name="label" translatable="yes">&lt;small&gt;High&lt;/small&gt;</property>
+                                <property name="use_markup">True</property>
+                              </object>
+                              <packing>
+                                <property name="expand">False</property>
+                                <property name="fill">True</property>
+                                <property name="position">3</property>
+                              </packing>
+                            </child>
+                          </object>
+                          <packing>
+                            <property name="expand">False</property>
+                            <property name="fill">True</property>
+                            <property name="position">3</property>
+                          </packing>
+                        </child>
                       </object>
+                      <packing>
+                        <property name="expand">False</property>
+                        <property name="fill">True</property>
+                        <property name="padding">5</property>
+                        <property name="position">0</property>
+                      </packing>
                     </child>
                     <child>
-                      <object class="GtkImage" id="image1">
+                      <object class="GtkBox" id="box7">
                         <property name="visible">True</property>
                         <property name="can_focus">False</property>
-                        <property name="icon_name">object-flip-horizontal-symbolic</property>
+                        <property name="orientation">vertical</property>
+                        <child>
+                          <object class="GtkSwitch" id="unity_launcher_autohide">
+                            <property name="use_action_appearance">False</property>
+                            <property name="visible">True</property>
+                            <property name="can_focus">True</property>
+                            <property name="use_action_appearance">False</property>
+                            <property name="active">True</property>
+                          </object>
+                          <packing>
+                            <property name="expand">False</property>
+                            <property name="fill">True</property>
+                            <property name="position">0</property>
+                          </packing>
+                        </child>
+                        <child>
+                          <placeholder/>
+                        </child>
                       </object>
+                      <packing>
+                        <property name="expand">False</property>
+                        <property name="fill">False</property>
+                        <property name="pack_type">end</property>
+                        <property name="position">1</property>
+                      </packing>
                     </child>
                   </object>
                   <packing>
                     <property name="expand">False</property>
-                    <property name="fill">False</property>
-                    <property name="position">1</property>
+                    <property name="fill">True</property>
+                    <property name="position">0</property>
                   </packing>
                 </child>
                 <child>
-                  <object class="GtkComboBox" id="style-combobox">
-                    <property name="visible">True</property>
-                    <property name="model">style-liststore</property>
-                    <property name="active">0</property>
-                    <child>
-                      <object class="GtkCellRendererText" id="cellrenderertext1"/>
-                      <attributes>
-                        <attribute name="text">0</attribute>
-                      </attributes>
-                    </child>
+                  <object class="GtkLabel" id="unity-label-external-tool">
+                    <property name="can_focus">False</property>
+                    <property name="xalign">0</property>
+                    <property name="xpad">12</property>
+                    <property name="label" translatable="yes">Some settings have been overriden by an external program, press "Restore Default Behaviors"  to reset the behavior and return control to this panel.</property>
+                    <property name="wrap">True</property>
                   </object>
                   <packing>
                     <property name="expand">False</property>
-                    <property name="fill">False</property>
-                    <property name="pack_type">end</property>
-                    <property name="position">3</property>
+                    <property name="fill">True</property>
+                    <property name="position">1</property>
                   </packing>
                 </child>
+              </object>
+              <packing>
+                <property name="expand">False</property>
+                <property name="fill">True</property>
+                <property name="position">0</property>
+              </packing>
+            </child>
+            <child>
+              <object class="GtkSeparator" id="separator3">
+                <property name="visible">True</property>
+                <property name="can_focus">False</property>
+              </object>
+              <packing>
+                <property name="expand">False</property>
+                <property name="fill">True</property>
+                <property name="position">1</property>
+              </packing>
+            </child>
+            <child>
+              <object class="GtkBox" id="box5">
+                <property name="height_request">36</property>
+                <property name="visible">True</property>
+                <property name="can_focus">False</property>
                 <child>
-                  <object class="GtkColorButton" id="style-scolor">
+                  <object class="GtkButton" id="button-restore-unitybehavior">
+                    <property name="label" translatable="yes">Restore Default Behaviours</property>
+                    <property name="use_action_appearance">False</property>
+                    <property name="width_request">136</property>
                     <property name="visible">True</property>
                     <property name="can_focus">True</property>
                     <property name="receives_default">True</property>
                     <property name="use_action_appearance">False</property>
-                    <property name="color">#000000000000</property>
-                    <child internal-child="accessible">
-                      <object class="AtkObject" id="style-scolor-a11y">
-                        <property name="accessible-name" translatable="yes">Secondary color</property>
-                      </object>
-                    </child>
                   </object>
                   <packing>
                     <property name="expand">False</property>
-                    <property name="fill">False</property>
-                    <property name="position">2</property>
+                    <property name="fill">True</property>
+                    <property name="padding">10</property>
+                    <property name="position">0</property>
                   </packing>
                 </child>
+                <child>
+                  <placeholder/>
+                </child>
               </object>
               <packing>
                 <property name="expand">False</property>
-                <property name="fill">False</property>
-                <property name="pack_type">end</property>
-                <property name="position">1</property>
+                <property name="fill">True</property>
+                <property name="position">2</property>
               </packing>
             </child>
+            <child>
+              <placeholder/>
+            </child>
+            <child>
+              <placeholder/>
+            </child>
           </object>
           <packing>
-            <property name="expand">False</property>
-            <property name="position">2</property>
+            <property name="position">1</property>
+          </packing>
+        </child>
+        <child type="tab">
+          <object class="GtkLabel" id="behavior_label">
+            <property name="visible">True</property>
+            <property name="can_focus">False</property>
+            <property name="label" translatable="yes">Behavior</property>
+          </object>
+          <packing>
+            <property name="position">1</property>
+            <property name="tab_fill">False</property>
           </packing>
         </child>
       </object>
-      <packing>
-        <property name="position">1</property>
-      </packing>
     </child>
   </object>
   <object class="GtkSizeGroup" id="sizegroup">
@@ -338,38 +937,4 @@
       <widget name="strut"/>
     </widgets>
   </object>
-  <object class="GtkListStore" id="style-liststore">
-    <columns>
-      <!-- column-name name -->
-      <column type="gchararray"/>
-      <!-- column-name value -->
-      <column type="gint"/>
-    </columns>
-    <data>
-      <row>
-        <col id="0" translatable="yes">Tile</col>
-        <col id="1">1</col>
-      </row>
-      <row>
-        <col id="0" translatable="yes">Zoom</col>
-        <col id="1">5</col>
-      </row>
-      <row>
-        <col id="0" translatable="yes">Center</col>
-        <col id="1">2</col>
-      </row>
-      <row>
-        <col id="0" translatable="yes">Scale</col>
-        <col id="1">3</col>
-      </row>
-      <row>
-        <col id="0" translatable="yes">Fill</col>
-        <col id="1">4</col>
-      </row>
-      <row>
-        <col id="0" translatable="yes">Span</col>
-        <col id="1">6</col>
-      </row>
-    </data>
-  </object>
 </interface>
Index: gnome-control-center-3.4.0/panels/background/cc-background-panel.c
===================================================================
--- gnome-control-center-3.4.0.orig/panels/background/cc-background-panel.c	2012-02-14 11:12:23.000000000 +0100
+++ gnome-control-center-3.4.0/panels/background/cc-background-panel.c	2012-04-05 12:11:40.406100456 +0200
@@ -24,6 +24,7 @@
 #include <string.h>
 #include <glib/gi18n-lib.h>
 #include <gdesktop-enums.h>
+#include <gconf/gconf-client.h>
 
 #include "cc-background-panel.h"
 #include "bg-wallpapers-source.h"
@@ -69,6 +70,9 @@
 #endif
 
   GSettings *settings;
+  GSettings *interface_settings;
+  GSettings *wm_theme_settings;
+  GSettings *unity2d_settings;
 
   GnomeDesktopThumbnailFactory *thumb_factory;
 
@@ -81,6 +85,8 @@
 
   GdkPixbuf *display_base;
   GdkPixbuf *display_overlay;
+
+  GSList *gconf_notify_id;
 };
 
 enum
@@ -93,6 +99,26 @@
 #endif
 };
 
+#define UNITY_GCONF_OPTION_PATH "/apps/compiz-1/plugins/unityshell/screen0/options"
+#define UNITY_ICONSIZE_KEY UNITY_GCONF_OPTION_PATH"/icon_size"
+#define UNITY_LAUNCHERSENSITIVITY_KEY UNITY_GCONF_OPTION_PATH"/edge_responsiveness"
+#define UNITY_LAUNCHERHIDE_KEY UNITY_GCONF_OPTION_PATH"/launcher_hide_mode"
+#define UNITY_LAUNCHERREVEAL_KEY UNITY_GCONF_OPTION_PATH"/reveal_trigger"
+#define UNITY2D_GSETTINGS_LAUNCHER "com.canonical.Unity2d.Launcher"
+
+#define MIN_ICONSIZE 32.0
+#define MAX_ICONSIZE 64.0
+
+#define MIN_LAUNCHER_SENSIVITY 0.2
+#define MAX_LAUNCHER_SENSIVITY 8.0
+
+typedef struct
+{
+  gdouble min;
+  gdouble max;
+} MinMax;
+static MinMax iconsize_values;
+static MinMax launchersensitivity_values;
 
 #define WID(y) (GtkWidget *) gtk_builder_get_object (priv->builder, y)
 
@@ -126,6 +152,10 @@
 cc_background_panel_dispose (GObject *object)
 {
   CcBackgroundPanelPrivate *priv = CC_BACKGROUND_PANEL (object)->priv;
+  GConfClient *client;
+  GSList* iter;
+
+  client = gconf_client_get_default ();
 
   if (priv->builder)
     {
@@ -167,6 +197,24 @@
       priv->settings = NULL;
     }
 
+  if (priv->interface_settings)
+    {
+      g_object_unref (priv->interface_settings);
+      priv->interface_settings = NULL;
+    }
+
+  if (priv->wm_theme_settings)
+    {
+      g_object_unref (priv->wm_theme_settings);
+      priv->wm_theme_settings = NULL;
+    }
+
+  if (priv->unity2d_settings)
+    {
+      g_object_unref (priv->unity2d_settings);
+      priv->unity2d_settings = NULL;
+    }
+
   if (priv->copy_cancellable)
     {
       /* cancel any copy operation */
@@ -194,6 +242,16 @@
       priv->display_overlay = NULL;
     }
 
+  iter = priv->gconf_notify_id;
+  while (iter != NULL)
+    {
+      gconf_client_notify_remove(client, (guint) iter->data);
+      iter = g_slist_next (iter);
+    }
+  g_object_unref (client);
+  g_slist_free (priv->gconf_notify_id);
+  priv->gconf_notify_id = NULL;
+
   G_OBJECT_CLASS (cc_background_panel_parent_class)->dispose (object);
 }
 
@@ -340,12 +398,21 @@
   if (priv->current_background)
     {
       GdkColor pcolor, scolor;
+      const char* bgsize = NULL;
 
-      markup = g_strdup_printf ("<b>%s</b>", cc_background_item_get_name (priv->current_background));
+      markup = g_strdup_printf ("<i>%s</i>", cc_background_item_get_name (priv->current_background));
       gtk_label_set_markup (GTK_LABEL (WID ("background-label")), markup);
       g_free (markup);
 
-      gtk_label_set_text (GTK_LABEL (WID ("size_label")), cc_background_item_get_size (priv->current_background));
+      bgsize = cc_background_item_get_size (priv->current_background);
+      if (bgsize && *bgsize != '\0')
+       {
+          markup = g_strdup_printf ("(%s)", bgsize);
+          gtk_label_set_text (GTK_LABEL (WID ("size_label")), markup);
+          g_free (markup);
+       }
+      else
+          gtk_label_set_text (GTK_LABEL (WID ("size_label")), "");
 
       gdk_color_parse (cc_background_item_get_pcolor (priv->current_background), &pcolor);
       gdk_color_parse (cc_background_item_get_scolor (priv->current_background), &scolor);
@@ -1162,12 +1229,503 @@
   g_strfreev(uris);
 }
 
+static gchar *themes_id[] = { "Adwaita", "Ambiance", "Radiance", "HighContrast", "HighContrastInverse" };
+static gchar *themes_name[] = { "Adwaita", "Ambiance", "Radiance", "High Contrast", "High Contrast Inverse" };
+
+static gboolean
+get_theme_data (const gchar *theme_name,
+		gchar **gtk_theme,
+		gchar **icon_theme,
+		gchar **window_theme,
+		gchar **cursor_theme)
+{
+  gchar *path;
+  GKeyFile *theme_file;
+  GError *error = NULL;
+  gboolean result = FALSE;
+
+  *gtk_theme = *icon_theme = *window_theme = *cursor_theme = NULL;
+
+  theme_file = g_key_file_new ();
+  path = g_build_filename ("/usr/share/themes", theme_name, "index.theme", NULL);
+  if (g_key_file_load_from_file (theme_file, path, G_KEY_FILE_NONE, &error))
+    {
+      *gtk_theme = g_key_file_get_string (theme_file, "X-GNOME-Metatheme", "GtkTheme", NULL);
+      *icon_theme = g_key_file_get_string (theme_file, "X-GNOME-Metatheme", "IconTheme", NULL);
+      *window_theme = g_key_file_get_string (theme_file, "X-GNOME-Metatheme", "MetacityTheme", NULL);
+      *cursor_theme = g_key_file_get_string (theme_file, "X-GNOME-Metatheme", "CursorTheme", NULL);
+
+      result = TRUE;
+    }
+  else
+    {
+      g_warning ("Could not load %s: %s", path, error->message);
+      g_error_free (error);
+    }
+
+  g_key_file_free (theme_file);
+  g_free (path);
+
+  return result;
+}
+
+static void
+theme_selection_changed (GtkComboBox *combo, CcBackgroundPanel *self)
+{
+  gint active;
+  gchar *gtk_theme, *icon_theme, *window_theme, *cursor_theme;
+  GConfClient *client;
+
+  active = gtk_combo_box_get_active (combo);
+  g_return_if_fail (active >= 0 && active < G_N_ELEMENTS (themes_id));
+
+  if (!get_theme_data (gtk_combo_box_get_active_id (combo),
+                       &gtk_theme, &icon_theme, &window_theme, &cursor_theme))
+    return;
+
+  g_settings_delay (self->priv->interface_settings);
+
+  g_settings_set_string (self->priv->interface_settings, "gtk-theme", gtk_theme);
+  g_settings_set_string (self->priv->interface_settings, "icon-theme", icon_theme);
+  g_settings_set_string (self->priv->interface_settings, "cursor-theme", cursor_theme);
+  g_settings_set_string (self->priv->wm_theme_settings, "theme", window_theme);
+
+
+  /* disable overlay scrollbars for a11y */
+  if (g_strcmp0 (gtk_theme, "HighContrast") == 0 ||
+      g_strcmp0 (gtk_theme, "HighContrastInverse") == 0)
+    g_settings_set_boolean (self->priv->interface_settings, "ubuntu-overlay-scrollbars", FALSE);
+  else
+    g_settings_reset (self->priv->interface_settings, "ubuntu-overlay-scrollbars");
+
+  g_settings_apply (self->priv->interface_settings);
+
+  /* For metacity, we use GConf */
+  client = gconf_client_get_default ();
+  gconf_client_set_string (client, "/desktop/gnome/shell/windows/theme", window_theme, NULL);
+  gconf_client_set_string (client, "/apps/metacity/general/theme", window_theme, NULL);
+
+  g_object_unref (client);
+  g_free (gtk_theme);
+  g_free (icon_theme);
+  g_free (window_theme);
+  g_free (cursor_theme);
+}
+
+static void
+setup_theme_selector (CcBackgroundPanel *self)
+{
+  gchar *current_gtk_theme;
+  gchar *default_gtk_theme;
+  gint i, current_theme_index = 0;
+  GtkWidget *widget;
+  GtkWidget *liststore;
+  CcBackgroundPanelPrivate *priv = self->priv;
+  GSettings *defaults_settings = g_settings_new ("org.gnome.desktop.interface");
+
+  priv->interface_settings = g_settings_new ("org.gnome.desktop.interface");
+  priv->wm_theme_settings = g_settings_new ("org.gnome.desktop.wm.preferences");
+  current_gtk_theme = g_settings_get_string (priv->interface_settings, "gtk-theme");
+
+  /* gettint the default for the theme */
+  g_settings_delay (defaults_settings);
+  g_settings_reset (defaults_settings, "gtk-theme");
+  default_gtk_theme = g_settings_get_string (defaults_settings, "gtk-theme");
+  g_object_unref (defaults_settings);
+
+  widget = WID ("theme-selector");
+  liststore = WID ("theme-list-store");
+
+  for (i = 0; i < G_N_ELEMENTS (themes_id); i++, current_theme_index++)
+    {
+      gchar *gtk_theme, *icon_theme, *window_theme, *cursor_theme, *new_theme_name;
+      GtkTreeIter iter;
+
+      if (!get_theme_data (themes_id[i], &gtk_theme, &icon_theme, &window_theme, &cursor_theme))
+        {
+          current_theme_index--;
+          continue;
+        }
+
+      if (g_strcmp0 (gtk_theme, default_gtk_theme) == 0)
+        new_theme_name = g_strdup_printf ("%s <small><i>(%s)</i></small>", themes_name[i], _("default"));
+      else
+        new_theme_name = g_strdup (themes_name[i]);
+
+      gtk_list_store_append (GTK_LIST_STORE (liststore), &iter);
+      gtk_list_store_set (GTK_LIST_STORE (liststore), &iter, 0, themes_id[i], 1, new_theme_name, -1);
+
+      if (g_strcmp0 (gtk_theme, current_gtk_theme) == 0)
+	  /* This is the current theme, so select item in the combo box */
+         gtk_combo_box_set_active (GTK_COMBO_BOX (widget), current_theme_index);
+
+      g_free (gtk_theme);
+      g_free (new_theme_name);
+      g_free (icon_theme);
+      g_free (window_theme);
+      g_free (cursor_theme);
+    }
+    g_free (current_gtk_theme);
+    g_free (default_gtk_theme);
+
+    g_signal_connect (G_OBJECT (widget), "changed",
+		      G_CALLBACK (theme_selection_changed), self);
+}
+
+static void
+iconsize_widget_refresh (GtkAdjustment *iconsize_adj)
+{
+  gint value = gconf_client_get_int (gconf_client_get_default (), UNITY_ICONSIZE_KEY, NULL);
+  gtk_adjustment_set_value (iconsize_adj, (gdouble)value);
+}
+
+static void
+ext_iconsize_changed_callback (GConfClient* client,
+                               guint cnxn_id,
+                               GConfEntry *entry,
+                               gpointer user_data)
+{
+  iconsize_widget_refresh (GTK_ADJUSTMENT (user_data));
+}
+
+static void
+on_iconsize_changed (GtkAdjustment *adj, gpointer user_data)
+{
+  gconf_client_set_int (gconf_client_get_default (), UNITY_ICONSIZE_KEY, (gint)gtk_adjustment_get_value (adj), NULL);
+}
+
+static void
+refresh_was_modified_by_external_tool (CcBackgroundPanel *self)
+{
+  CcBackgroundPanelPrivate *priv = self->priv;
+  gboolean modified_ext_tool = FALSE;
+
+  // reveal side
+  modified_ext_tool = gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (WID ("unity_reveal_spot_otheroption")));
+
+  // autohide mode
+  if (!modified_ext_tool && (!gtk_widget_get_sensitive (WID ("unity_launcher_autohide"))))
+    modified_ext_tool = TRUE;
+
+  gtk_widget_set_visible (WID ("unity-label-external-tool"), modified_ext_tool);
+}
+
+static void
+hidelauncher_set_sensitivity_reveal (CcBackgroundPanel *self, gboolean autohide)
+{
+  CcBackgroundPanelPrivate *priv = self->priv;
+  gtk_widget_set_sensitive (WID ("unity_reveal_label"), autohide);
+  gtk_widget_set_sensitive (WID ("unity_reveal_spot_topleft"), autohide);
+  gtk_widget_set_sensitive (WID ("unity_reveal_spot_left"), autohide);
+  gtk_widget_set_sensitive (WID ("unity-launcher-sensitivity"), autohide);
+  gtk_widget_set_sensitive (WID ("unity-launcher-sensitivity-label"), autohide);
+  gtk_widget_set_sensitive (WID ("unity-launcher-sensitivity-low-label"), autohide);
+  gtk_widget_set_sensitive (WID ("unity-launcher-sensitivity-high-label"), autohide);
+  gtk_widget_set_sensitive (WID ("unity-launcher-sensitivity-high-label"), autohide);
+}
+
+static void
+hidelauncher_widget_refresh (CcBackgroundPanel *self)
+{
+  CcBackgroundPanelPrivate *priv = self->priv;
+  gint value = gconf_client_get_int (gconf_client_get_default (), UNITY_LAUNCHERHIDE_KEY, NULL);
+  gboolean autohide = (value != 0);
+
+  // handle not supported value
+  if (value != 0 && value != 1)
+    {
+      gtk_widget_set_sensitive (WID ("unity_launcher_autohide"), FALSE);
+    }
+  else
+    {
+      gtk_widget_set_sensitive (WID ("unity_launcher_autohide"), TRUE);
+      gtk_switch_set_active (GTK_SWITCH (WID ("unity_launcher_autohide")), autohide);
+      g_settings_set_int (priv->unity2d_settings, "hide-mode", value);
+    }
+
+  hidelauncher_set_sensitivity_reveal (self, autohide);
+  refresh_was_modified_by_external_tool (self);
+}
+
+static void
+ext_hidelauncher_changed_callback (GConfClient* client,
+                                   guint cnxn_id,
+                                   GConfEntry *entry,
+                                   gpointer user_data)
+{
+  hidelauncher_widget_refresh (CC_BACKGROUND_PANEL (user_data));
+}
+
+static void
+on_hidelauncher_changed (GtkSwitch *switcher, gboolean enabled, gpointer user_data)
+{
+  gint value = 0;
+  gint gconf_value = gconf_client_get_int (gconf_client_get_default (), UNITY_LAUNCHERHIDE_KEY, NULL);
+  gboolean gconf_autohide_enabled;
+
+  gconf_autohide_enabled = (gconf_value != 0);
+  if (gtk_switch_get_active (switcher))
+    {
+      /* change value to "active" if activation isn't due to gconf switching to any value */
+      if (gconf_autohide_enabled)
+        return;
+      value = 1;
+    }
+
+  /* 3d */
+  gconf_client_set_int (gconf_client_get_default (), UNITY_LAUNCHERHIDE_KEY, value, NULL);
+  hidelauncher_set_sensitivity_reveal (CC_BACKGROUND_PANEL (user_data), (value != -1));
+}
+
+static void
+reveallauncher_widget_refresh (CcBackgroundPanel *self)
+{
+  CcBackgroundPanelPrivate *priv = self->priv;
+  gint value = gconf_client_get_int (gconf_client_get_default (), UNITY_LAUNCHERREVEAL_KEY, NULL);
+
+  if (value == 1)
+    gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (WID ("unity_reveal_spot_topleft")), TRUE);
+  else if (value == 0)
+    gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (WID ("unity_reveal_spot_left")), TRUE);
+  else
+    /* this is a hidden spot when another option is selected (through ccsm) */
+    gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (WID ("unity_reveal_spot_otheroption")), TRUE);
+
+  refresh_was_modified_by_external_tool (self);
+}
+
+static void
+ext_reveallauncher_changed_callback (GConfClient* client,
+                                     guint cnxn_id,
+                                     GConfEntry *entry,
+                                     gpointer user_data)
+{
+  reveallauncher_widget_refresh (CC_BACKGROUND_PANEL (user_data));
+}
+
+static void
+on_reveallauncher_changed (GtkToggleButton *button, gpointer user_data)
+{
+  CcBackgroundPanel *self = CC_BACKGROUND_PANEL (user_data);
+  CcBackgroundPanelPrivate *priv = self->priv;
+  gint reveal_spot = 0;
+
+  if (gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (WID ("unity_reveal_spot_topleft"))))
+    reveal_spot = 1;
+  if (gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (WID ("unity_reveal_spot_left"))))
+    reveal_spot = 0;
+
+  gconf_client_set_int (gconf_client_get_default (), UNITY_LAUNCHERREVEAL_KEY, reveal_spot, NULL);
+  if (priv->unity2d_settings)
+    g_settings_set_int (priv->unity2d_settings, "reveal-mode", reveal_spot);
+  reveallauncher_widget_refresh (self);
+}
+
+static void
+launcher_sensitivity_widget_refresh (GtkAdjustment *launcher_sensitivity_adj)
+{
+  gdouble value = gconf_client_get_float (gconf_client_get_default (), UNITY_LAUNCHERSENSITIVITY_KEY, NULL);
+  gtk_adjustment_set_value (launcher_sensitivity_adj, (gdouble)value);
+}
+
+static void
+ext_launchersensitivity_changed_callback (GConfClient* client,
+                                          guint cnxn_id,
+                                          GConfEntry *entry,
+                                          gpointer user_data)
+{
+  launcher_sensitivity_widget_refresh (GTK_ADJUSTMENT (user_data));
+}
+
+static void
+on_launchersensitivity_changed (GtkAdjustment *adj, gpointer user_data)
+{
+  CcBackgroundPanel *self = CC_BACKGROUND_PANEL (user_data);
+  CcBackgroundPanelPrivate *priv = self->priv;
+  gdouble value = gtk_adjustment_get_value (adj);
+
+  gconf_client_set_float (gconf_client_get_default (), UNITY_LAUNCHERSENSITIVITY_KEY, value, NULL);
+  if (priv->unity2d_settings)
+    g_settings_set_double (priv->unity2d_settings, "edge-responsiveness", value);
+}
+
+static void
+on_restore_defaults_page2_clicked (GtkButton *button, gpointer user_data)
+{
+  GConfClient *client;
+  CcBackgroundPanel *self = CC_BACKGROUND_PANEL (user_data);
+  CcBackgroundPanelPrivate *priv = self->priv;
+
+  client = gconf_client_get_default ();
+
+  /* reset defaut for the profile and get the default */
+  gconf_client_unset (client, UNITY_LAUNCHERHIDE_KEY, NULL);
+  gconf_client_unset (client, UNITY_LAUNCHERSENSITIVITY_KEY, NULL);
+  gconf_client_unset (client, UNITY_LAUNCHERREVEAL_KEY, NULL);
+
+  if (priv->unity2d_settings)
+    {
+      g_settings_reset (priv->unity2d_settings, "hide-mode");
+      g_settings_reset (priv->unity2d_settings, "reveal-mode");
+      g_settings_reset (priv->unity2d_settings, "edge-responsiveness");
+    }
+}
+
+static gboolean
+background_is_unity_session (void)
+{
+  return (g_strcmp0 (g_getenv("XDG_CURRENT_DESKTOP"), "Unity") == 0);
+}
+
+/* <hacks> */
+
+/* Get scrolling in the right direction */
+static gboolean
+on_scale_scroll_event (GtkWidget      *widget,
+                       GdkEventScroll *event,
+                       gpointer *data)
+{
+  gdouble value;
+  GtkAdjustment *adj = gtk_range_get_adjustment (GTK_RANGE (widget));
+  MinMax *iconsize_values = (MinMax *) data;
+  gdouble delta = iconsize_values->max - iconsize_values->min;
+
+  value = gtk_adjustment_get_value (adj);
+
+  if ((event->direction == GDK_SCROLL_UP) ||
+     (event->direction == GDK_SCROLL_SMOOTH && event->delta_y < 0))
+    {
+      if (value + delta/8 > iconsize_values->max)
+        value = iconsize_values->max;
+      else
+        value = value + delta/8;
+      gtk_adjustment_set_value (adj, value);
+    }
+  else if ((event->direction == GDK_SCROLL_DOWN) ||
+           (event->direction == GDK_SCROLL_SMOOTH && event->delta_y > 0))
+    {
+      if (value - delta/8 < iconsize_values->min)
+        value = iconsize_values->min;
+      else
+        value = value - delta/8;
+      gtk_adjustment_set_value (adj, value);
+    }
+
+  return TRUE;
+}
+
+/* </hacks> */
+
+static void
+setup_unity_settings (CcBackgroundPanel *self)
+{
+  CcBackgroundPanelPrivate *priv = self->priv;
+  GConfClient *client;
+  GtkAdjustment* iconsize_adj;
+  GtkAdjustment* launcher_sensitivity_adj;
+  GtkScale* iconsize_scale;
+  GtkScale* launcher_sensitivity_scale;
+  const gchar * const *schemas;
+
+  client = gconf_client_get_default ();
+  gconf_client_add_dir (client, UNITY_GCONF_OPTION_PATH, 0, NULL);
+  priv->gconf_notify_id = NULL;
+
+  /* Only use the unity-2d schema if it's installed */
+  schemas = g_settings_list_schemas ();
+  while (*schemas != NULL)
+    {
+      if (g_strcmp0 (*schemas, UNITY2D_GSETTINGS_LAUNCHER) == 0)
+        {
+          priv->unity2d_settings = g_settings_new (UNITY2D_GSETTINGS_LAUNCHER);
+          break;
+        }
+      schemas++;
+    }
+
+  /* Option not supported in unity-2d */
+  if (g_strcmp0(g_getenv("DESKTOP_SESSION"), "ubuntu-2d") != 0)
+    {
+      /* Icon size change */
+      iconsize_values.min = MIN_ICONSIZE;
+      iconsize_values.max = MAX_ICONSIZE;
+      iconsize_adj = gtk_adjustment_new (48, iconsize_values.min, iconsize_values.max, 1, 5, 0);
+      iconsize_scale = GTK_SCALE (WID ("unity-iconsize-scale"));
+      gtk_range_set_adjustment (GTK_RANGE (iconsize_scale), iconsize_adj);
+      gtk_scale_add_mark (iconsize_scale, 48, GTK_POS_BOTTOM, NULL);
+      priv->gconf_notify_id = g_slist_append (priv->gconf_notify_id,
+                                              GINT_TO_POINTER (
+                                              gconf_client_notify_add (client, UNITY_ICONSIZE_KEY,
+                                                       ext_iconsize_changed_callback, iconsize_adj,
+                                                       NULL, NULL)));
+
+      g_signal_connect (iconsize_adj, "value_changed",
+                        G_CALLBACK (on_iconsize_changed), NULL);
+      g_signal_connect (G_OBJECT (iconsize_scale), "scroll-event",
+                        G_CALLBACK (on_scale_scroll_event), &iconsize_values);
+      iconsize_widget_refresh (iconsize_adj);
+    }
+  else
+    {
+      gtk_widget_hide (GTK_WIDGET (WID ("unity-separator1")));
+      gtk_widget_hide (GTK_WIDGET (WID ("unity-iconsize-box")));
+    }
+
+  /* Reveal spot setting */
+  priv->gconf_notify_id = g_slist_append (priv->gconf_notify_id,
+                                          GINT_TO_POINTER (
+                                          gconf_client_notify_add (client, UNITY_LAUNCHERREVEAL_KEY,
+                                                   ext_reveallauncher_changed_callback, self,
+                                                   NULL, NULL)));
+  g_signal_connect (WID ("unity_reveal_spot_topleft"), "toggled",
+                     G_CALLBACK (on_reveallauncher_changed), self);
+  g_signal_connect (WID ("unity_reveal_spot_left"), "toggled",
+                     G_CALLBACK (on_reveallauncher_changed), self);
+  reveallauncher_widget_refresh (self);
+
+  /* Launcher reveal */
+  launchersensitivity_values.min = MIN_LAUNCHER_SENSIVITY;
+  launchersensitivity_values.max = MAX_LAUNCHER_SENSIVITY;
+  launcher_sensitivity_adj = gtk_adjustment_new (2, launchersensitivity_values.min, launchersensitivity_values.max, 0.1, 1, 0);
+  launcher_sensitivity_scale = GTK_SCALE (WID ("unity-launcher-sensitivity"));
+  gtk_range_set_adjustment (GTK_RANGE (launcher_sensitivity_scale), launcher_sensitivity_adj);
+  gtk_scale_add_mark (launcher_sensitivity_scale, 2, GTK_POS_BOTTOM, NULL);
+  priv->gconf_notify_id = g_slist_append (priv->gconf_notify_id,
+                                          GINT_TO_POINTER (
+                                          gconf_client_notify_add (client, UNITY_LAUNCHERSENSITIVITY_KEY,
+                                                   ext_launchersensitivity_changed_callback, launcher_sensitivity_adj,
+                                                   NULL, NULL)));
+  g_signal_connect (launcher_sensitivity_adj, "value_changed",
+                    G_CALLBACK (on_launchersensitivity_changed), self);
+  g_signal_connect (G_OBJECT (launcher_sensitivity_scale), "scroll-event",
+                    G_CALLBACK (on_scale_scroll_event), &launchersensitivity_values);
+  launcher_sensitivity_widget_refresh (launcher_sensitivity_adj);
+
+  /* Autohide launcher setting */
+  priv->gconf_notify_id = g_slist_append (priv->gconf_notify_id,
+                                          GINT_TO_POINTER (
+                                          gconf_client_notify_add (client, UNITY_LAUNCHERHIDE_KEY,
+                                                   ext_hidelauncher_changed_callback, self,
+                                                   NULL, NULL)));
+  g_signal_connect (WID ("unity_launcher_autohide"), "notify::active",
+                    G_CALLBACK (on_hidelauncher_changed), self);
+  hidelauncher_widget_refresh (self);
+
+  /* Restore defaut on second page */
+  g_signal_connect (WID ("button-restore-unitybehavior"), "clicked",
+                    G_CALLBACK (on_restore_defaults_page2_clicked), self);
+
+  g_object_unref (client);
+}
+
 static void
 cc_background_panel_init (CcBackgroundPanel *self)
 {
   CcBackgroundPanelPrivate *priv;
   gchar *objects[] = { "style-liststore",
-      "sources-liststore", "background-panel", "sizegroup", NULL };
+      "sources-liststore", "theme-list-store", "background-panel", "sizegroup", NULL };
+  gchar *objects_unity[] = { "style-liststore",
+      "sources-liststore", "theme-list-store", "main-notebook", "sizegroup", NULL };
   GError *err = NULL;
   GtkWidget *widget;
   GtkListStore *store;
@@ -1177,9 +1735,14 @@
 
   priv->builder = gtk_builder_new ();
 
-  gtk_builder_add_objects_from_file (priv->builder,
-                                     DATADIR"/background.ui",
-                                     objects, &err);
+  if (background_is_unity_session ())
+    gtk_builder_add_objects_from_file (priv->builder,
+                                       DATADIR"/background.ui",
+                                       objects_unity, &err);
+  else
+    gtk_builder_add_objects_from_file (priv->builder,
+                                       DATADIR"/background.ui",
+                                       objects, &err);
 
   if (err)
     {
@@ -1230,7 +1793,10 @@
 
 
   /* add the top level widget */
-  widget = WID ("background-panel");
+  if (background_is_unity_session ())
+    widget = WID ("main-notebook");
+  else
+    widget = WID ("background-panel");
 
   gtk_container_add (GTK_CONTAINER (self), widget);
   gtk_widget_show_all (GTK_WIDGET (self));
@@ -1301,6 +1867,18 @@
 
   /* Setup the edit box with our current settings */
   source_update_edit_box (priv, TRUE);
+
+  /* Setup theme selector */
+  setup_theme_selector (self);
+
+  /* Setup unity settings */
+  if (background_is_unity_session ())
+    setup_unity_settings (self);
+  else
+    {
+      gtk_widget_hide (GTK_WIDGET (WID ("unity-separator1")));
+      gtk_widget_hide (GTK_WIDGET (WID ("unity-iconsize-box")));
+    }
 }
 
 void
Index: gnome-control-center-3.4.0/panels/background/gnome-background-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.4.0.orig/panels/background/gnome-background-panel.desktop.in.in	2012-02-14 11:13:17.000000000 +0100
+++ gnome-control-center-3.4.0/panels/background/gnome-background-panel.desktop.in.in	2012-04-05 12:08:51.777264279 +0200
@@ -1,6 +1,6 @@
 [Desktop Entry]
-_Name=Background
-_Comment=Change the background
+_Name=Appearance
+_Comment=Change the background and theme
 Exec=gnome-control-center background
 Icon=preferences-desktop-wallpaper
 Terminal=false
@@ -10,4 +10,4 @@
 OnlyShowIn=GNOME;Unity;
 X-GNOME-Settings-Panel=background
 # Translators: those are keywords for the background control-center panel
-_Keywords=Wallpaper;Screen;Desktop;
+_Keywords=Wallpaper;Screen;Desktop;Theme;Appearance;Launcher;Unity;Menus;
