diff -urNad control-center-2.18.0/configure.in control-center-2.18.0.new/configure.in
--- control-center-2.18.0/configure.in	2007-03-26 17:11:41.000000000 +0200
+++ control-center-2.18.0.new/configure.in	2007-03-26 17:13:46.000000000 +0200
@@ -52,7 +52,7 @@
 AC_CHECK_FUNCS(putenv)
 AC_CHECK_FUNCS(setenv)
 
-AC_PATH_XTRA
+AC_PATH_XTRA( [X11], [X11/Xlib.h], [XSync ()])
 x_libs="$X_PRE_LIBS $X_LIBS -lX11 $X_EXTRA_LIBS"
 
 AC_CHECK_FUNC(XScreenSaverQueryInfo, [
