diff -Nur control-center-2.6.1.orig/configure.in control-center-2.6.1/configure.in
--- control-center-2.6.1.orig/configure.in	2004-05-29 17:57:08.000000000 +0200
+++ control-center-2.6.1/configure.in	2004-05-29 17:57:25.000000000 +0200
@@ -49,7 +49,7 @@
 	AC_MSG_ERROR([Your Gettext installation doesn't seem to support ngettext to handle translation of plural forms. Please install GNU Gettext])
 fi;
 
-AC_PATH_XTRA
+AC_PATH_XTRA( [X11], [X11/Xlib.h], [XSync ()])
 x_libs="$X_PRE_LIBS $X_LIBS -lX11 $X_EXTRA_LIBS"
 
 AC_CHECK_FUNC(XScreenSaverQueryInfo, [
