Index: gnome-control-center-2.24.0.1/configure.in
===================================================================
--- gnome-control-center-2.24.0.1.orig/configure.in	2008-12-28 13:21:15.373512065 +0100
+++ gnome-control-center-2.24.0.1/configure.in	2008-12-28 13:22:57.644359897 +0100
@@ -48,7 +48,7 @@ GNOME_DEBUG_CHECK
 GNOME_COMPILE_WARNINGS([maximum])
 GNOME_MAINTAINER_MODE_DEFINES
 
-AC_PATH_XTRA
+AC_PATH_XTRA( [X11], [X11/Xlib.h], [XSync ()])
 x_libs="$X_PRE_LIBS $X_LIBS -lX11 $X_EXTRA_LIBS"
 
 AC_CHECK_FUNC(XScreenSaverQueryInfo, [
