Index: gnome-control-center-3.4.2/panels/info/cc-info-panel.c
===================================================================
--- gnome-control-center-3.4.2.orig/panels/info/cc-info-panel.c	2012-04-20 01:26:57.000000000 +1200
+++ gnome-control-center-3.4.2/panels/info/cc-info-panel.c	2012-05-22 12:13:07.610180837 +1200
@@ -1897,7 +1897,7 @@
 {
   GError *error;
   error = NULL;
-  g_spawn_command_line_async ("gpk-update-viewer", &error);
+  g_spawn_command_line_async ("update-manager", &error);
   if (error != NULL)
     {
       g_warning ("unable to launch Software Updates: %s", error->message);
