diff --git a/panels/info/cc-info-panel.c b/panels/info/cc-info-panel.c
index f4dba1b..0ad2b5a 100644
--- a/panels/info/cc-info-panel.c
+++ b/panels/info/cc-info-panel.c
@@ -1380,7 +1380,7 @@ on_updates_button_clicked (GtkWidget   *widget,
 {
   GError *error;
   error = NULL;
-  g_spawn_command_line_async ("gpk-update-viewer", &error);
+  g_spawn_command_line_async ("update-manager", &error);
   if (error != NULL)
     {
       g_warning ("unable to launch Software Updates: %s", error->message);
