Index: gnome-control-center-2.24.0.1/capplets/display/xrandr-capplet.c
===================================================================
--- gnome-control-center-2.24.0.1.orig/capplets/display/xrandr-capplet.c	2009-04-16 10:36:37.412850259 +0200
+++ gnome-control-center-2.24.0.1/capplets/display/xrandr-capplet.c	2009-04-16 10:37:37.653350312 +0200
@@ -435,7 +435,7 @@ rebuild_rotation_combo (App *app)
 
 	app->current_output->rotation = info->rotation;
 
-	if (gnome_rr_config_applicable (app->current_configuration, app->screen))
+	if (gnome_rr_config_applicable (app->current_configuration, app->screen, NULL))
 	{
  	    add_key (app->rotation_combo, info->name, 0, 0, 0, info->rotation);
 
@@ -1689,7 +1689,7 @@ on_detect_displays (GtkWidget *widget, g
 {
     App *app = data;
 
-    gnome_rr_screen_refresh (app->screen);
+    gnome_rr_screen_refresh (app->screen, NULL);
 }
 
 #define SHOW_ICON_KEY "/apps/gnome_settings_daemon/xrandr/show_notification_icon"
@@ -1723,7 +1723,7 @@ run_application (App *app)
     }
 
     app->screen = gnome_rr_screen_new (gdk_screen_get_default (),
-				       on_screen_changed, app);
+				       on_screen_changed, app, NULL);
     if (!app->screen)
     {
 	g_error ("Could not get screen info");
