diff --git a/typing-break/drwright.c b/typing-break/drwright.c
index 5e8674d..c7fc026 100644
--- a/typing-break/drwright.c
+++ b/typing-break/drwright.c
@@ -521,6 +521,8 @@ update_status (DrWright *dr)
 #ifdef HAVE_APP_INDICATOR
 	item = gtk_ui_manager_get_widget (dr->ui_manager, "/Pop/TakeABreak");
 	gtk_menu_item_set_label (GTK_MENU_ITEM (item), str);
+	app_indicator_set_attention_icon_full (dr->indicator, TYPING_MONITOR_ATTENTION_ICON, str);
+	app_indicator_set_icon_full (dr->indicator, TYPING_MONITOR_ACTIVE_ICON, str);
 #else
 	gtk_status_icon_set_tooltip_text (dr->icon, str);
 #endif /* HAVE_APP_INDICATOR */
