--- gnome-settings-daemon/gnome-settings-multimedia-keys.c.orig	2004-07-02 01:05:09.000000000 +0200
+++ gnome-settings-daemon/gnome-settings-multimedia-keys.c	2004-07-02 01:31:18.000000000 +0200
@@ -539,10 +539,10 @@
 	x = ((screen_w - orig_w) / 2) + geometry.x;
 	y = geometry.y + (screen_h / 2) + (screen_h / 2 - orig_h) / 2;
 
-	gdk_window_move (GTK_WIDGET (acme->dialog)->window, x, y);
-
 	gtk_widget_show (acme->dialog);
 
+	gdk_window_move (GTK_WIDGET (acme->dialog)->window, x, y);
+
 	/* this makes sure the dialog is actually shown */
 	while (gtk_events_pending())
 		gtk_main_iteration();
