diff -Nur control-center-2.18.1/schemas/apps_gnome_settings_daemon_keybindings.schemas.in control-center-2.18.1.new/schemas/apps_gnome_settings_daemon_keybindings.schemas.in
--- control-center-2.18.1/schemas/apps_gnome_settings_daemon_keybindings.schemas.in	2007-03-18 18:04:18.000000000 +0100
+++ control-center-2.18.1.new/schemas/apps_gnome_settings_daemon_keybindings.schemas.in	2007-04-12 12:56:58.000000000 +0200
@@ -17,7 +17,7 @@
             <key>/schemas/apps/gnome_settings_daemon/keybindings/volume_mute</key>
             <applyto>/apps/gnome_settings_daemon/keybindings/volume_mute</applyto>
             <type>string</type>
-            <default></default>
+            <default>0xa0</default>
             <locale name="C">
                 <short>Volume mute</short>
                 <long>Volume mute's shortcut</long>
@@ -29,7 +29,7 @@
             <key>/schemas/apps/gnome_settings_daemon/keybindings/volume_down</key>
             <applyto>/apps/gnome_settings_daemon/keybindings/volume_down</applyto>
             <type>string</type>
-            <default></default>
+            <default>0xae</default>
             <locale name="C">
                 <short>Volume down</short>
                 <long>Volume down's shortcut.</long>
@@ -41,7 +41,7 @@
             <key>/schemas/apps/gnome_settings_daemon/keybindings/volume_up</key>
             <applyto>/apps/gnome_settings_daemon/keybindings/volume_up</applyto>
             <type>string</type>
-            <default></default>
+            <default>0xb0</default>
             <locale name="C">
                 <short>Volume up</short>
                 <long>Volume up's shortcut.</long>
@@ -65,7 +65,7 @@
             <key>/schemas/apps/gnome_settings_daemon/keybindings/eject</key>
             <applyto>/apps/gnome_settings_daemon/keybindings/eject</applyto>
             <type>string</type>
-            <default></default>
+            <default>0xcc</default>
             <locale name="C">
                 <short>Eject</short>
                 <long>Eject's shortcut.</long>
@@ -89,7 +89,7 @@
             <key>/schemas/apps/gnome_settings_daemon/keybindings/search</key>
             <applyto>/apps/gnome_settings_daemon/keybindings/search</applyto>
             <type>string</type>
-            <default></default>
+            <default>0xe5</default>
             <locale name="C">
                 <short>Search</short>
                 <long>Search's shortcut.</long>
@@ -101,7 +101,7 @@
             <key>/schemas/apps/gnome_settings_daemon/keybindings/email</key>
             <applyto>/apps/gnome_settings_daemon/keybindings/email</applyto>
             <type>string</type>
-            <default></default>
+            <default>0xec</default>
             <locale name="C">
                 <short>E-mail</short>
                 <long>E-mail's shortcut.</long>
@@ -137,7 +137,7 @@
             <key>/schemas/apps/gnome_settings_daemon/keybindings/help</key>
             <applyto>/apps/gnome_settings_daemon/keybindings/help</applyto>
             <type>string</type>
-            <default></default>
+            <default>0xf5</default>
             <locale name="C">
                 <short>Launch help browser</short>
                 <long>Launch help browser's shortcut.</long>
@@ -149,7 +149,7 @@
             <key>/schemas/apps/gnome_settings_daemon/keybindings/www</key>
             <applyto>/apps/gnome_settings_daemon/keybindings/www</applyto>
             <type>string</type>
-            <default></default>
+            <default>0xb2</default>
             <locale name="C">
                 <short>Launch web browser</short>
                 <long>Launch web browser's shortcut.</long>
@@ -161,7 +161,7 @@
             <key>/schemas/apps/gnome_settings_daemon/keybindings/play</key>
             <applyto>/apps/gnome_settings_daemon/keybindings/play</applyto>
             <type>string</type>
-            <default></default>
+            <default>0xa2</default>
             <locale name="C">
                 <short>Play (or play/pause)</short>
                 <long>Play (or play/pause) key's shortcut.</long>
@@ -185,7 +185,7 @@
             <key>/schemas/apps/gnome_settings_daemon/keybindings/stop</key>
             <applyto>/apps/gnome_settings_daemon/keybindings/stop</applyto>
             <type>string</type>
-            <default></default>
+            <default>0xa4</default>
             <locale name="C">
                 <short>Stop playback key</short>
                 <long>Stop playback key's shortcut.</long>
@@ -197,7 +197,7 @@
             <key>/schemas/apps/gnome_settings_daemon/keybindings/previous</key>
             <applyto>/apps/gnome_settings_daemon/keybindings/previous</applyto>
             <type>string</type>
-            <default></default>
+            <default>0x90</default>
             <locale name="C">
                 <short>Skip to previous track</short>
                 <long>Previous track key's shortcut.</long>
@@ -209,7 +209,7 @@
             <key>/schemas/apps/gnome_settings_daemon/keybindings/next</key>
             <applyto>/apps/gnome_settings_daemon/keybindings/next</applyto>
             <type>string</type>
-            <default></default>
+            <default>0x99</default>
             <locale name="C">
                 <short>Skip to next track</short>
                 <long>Next track key's shortcut.</long>
