diff -Nur control-center-2.14.1/schemas/apps_gnome_settings_daemon_keybindings.schemas.in control-center-2.14.1.new/schemas/apps_gnome_settings_daemon_keybindings.schemas.in
--- control-center-2.14.1/schemas/apps_gnome_settings_daemon_keybindings.schemas.in	2006-04-10 22:21:48.000000000 +0200
+++ control-center-2.14.1.new/schemas/apps_gnome_settings_daemon_keybindings.schemas.in	2006-04-10 22:26:09.000000000 +0200
@@ -17,7 +17,7 @@
             <key>/schemas/apps/gnome_settings_daemon/keybindings/volume_mute</key>
             <applyto>/apps/gnome_settings_daemon/keybindings/volume_mute</applyto>
             <type>string</type>
-            <default></default>
+            <default>0xa0</default>
             <locale name="C">
                 <short>Volume mute</short>
                 <long>Volume mute's shortcut</long>
@@ -29,7 +29,7 @@
             <key>/schemas/apps/gnome_settings_daemon/keybindings/volume_down</key>
             <applyto>/apps/gnome_settings_daemon/keybindings/volume_down</applyto>
             <type>string</type>
-            <default></default>
+            <default>0xae</default>
             <locale name="C">
                 <short>Volume down</short>
                 <long>Volume down's shortcut.</long>
@@ -41,7 +41,7 @@
             <key>/schemas/apps/gnome_settings_daemon/keybindings/volume_up</key>
             <applyto>/apps/gnome_settings_daemon/keybindings/volume_up</applyto>
             <type>string</type>
-            <default></default>
+            <default>0xb0</default>
             <locale name="C">
                 <short>Volume up</short>
                 <long>Volume up's shortcut.</long>
@@ -65,7 +65,7 @@
             <key>/schemas/apps/gnome_settings_daemon/keybindings/eject</key>
             <applyto>/apps/gnome_settings_daemon/keybindings/eject</applyto>
             <type>string</type>
-            <default></default>
+            <default>0xcc</default>
             <locale name="C">
                 <short>Eject</short>
                 <long>Eject's shortcut.</long>
@@ -89,7 +89,7 @@
             <key>/schemas/apps/gnome_settings_daemon/keybindings/search</key>
             <applyto>/apps/gnome_settings_daemon/keybindings/search</applyto>
             <type>string</type>
-            <default></default>
+            <default>0xe5</default>
             <locale name="C">
                 <short>Search</short>
                 <long>Search's shortcut.</long>
@@ -101,7 +101,7 @@
             <key>/schemas/apps/gnome_settings_daemon/keybindings/email</key>
             <applyto>/apps/gnome_settings_daemon/keybindings/email</applyto>
             <type>string</type>
-            <default></default>
+            <default>0xec</default>
             <locale name="C">
                 <short>E-mail</short>
                 <long>E-mail's shortcut.</long>
@@ -137,7 +137,7 @@
             <key>/schemas/apps/gnome_settings_daemon/keybindings/help</key>
             <applyto>/apps/gnome_settings_daemon/keybindings/help</applyto>
             <type>string</type>
-            <default></default>
+            <default>0xf5</default>
             <locale name="C">
                 <short>Launch help browser</short>
                 <long>Launch help browser's shortcut.</long>
@@ -149,7 +149,7 @@
             <key>/schemas/apps/gnome_settings_daemon/keybindings/www</key>
             <applyto>/apps/gnome_settings_daemon/keybindings/www</applyto>
             <type>string</type>
-            <default></default>
+            <default>0xb2</default>
             <locale name="C">
                 <short>Launch web browser</short>
                 <long>Launch web browser's shortcut.</long>
@@ -173,7 +173,7 @@
             <key>/schemas/apps/gnome_settings_daemon/keybindings/brightness_up</key>
             <applyto>/apps/gnome_settings_daemon/keybindings/brightness_up</applyto>
             <type>string</type>
-            <default></default>
+            <default>0xa2</default>
             <locale name="C">
                 <short>Brightness up</short>
                 <long>Brightness up's shortcut.</long>
