Index: gnome-control-center-3.1.5/shell/gnome-control-center.c
===================================================================
--- gnome-control-center-3.1.5.orig/shell/gnome-control-center.c	2011-08-28 23:38:09.023225450 -0400
+++ gnome-control-center-3.1.5/shell/gnome-control-center.c	2011-08-28 23:38:18.173238867 -0400
@@ -81,7 +81,7 @@
 
 /* Use a fixed width for the shell, since resizing horizontally is more awkward
  * for the user than resizing vertically */
-#define FIXED_WIDTH 675
+#define FIXED_WIDTH 740
 
 
 static const gchar *
