diff --git a/panels/common/gdm-languages.c b/panels/common/gdm-languages.c
index 2dfc436..50a1cd6 100644
--- a/panels/common/gdm-languages.c
+++ b/panels/common/gdm-languages.c
@@ -635,9 +635,11 @@ count_languages_and_territories (void)
 		count++;
 		g_hash_table_insert (gdm_language_count_map, g_strdup (locale->language_code), GINT_TO_POINTER (count));
 
-		count = GPOINTER_TO_INT (g_hash_table_lookup (gdm_territory_count_map, locale->territory_code));
-		count++;
-		g_hash_table_insert (gdm_territory_count_map, g_strdup (locale->territory_code), GINT_TO_POINTER (count));
+		if (locale->territory_code != NULL) {
+			count = GPOINTER_TO_INT (g_hash_table_lookup (gdm_territory_count_map, locale->territory_code));
+			count++;
+			g_hash_table_insert (gdm_territory_count_map, g_strdup (locale->territory_code), GINT_TO_POINTER (count));
+		}
         }
 }
 
