diff -Nur -x '*.orig' -x '*~' gnome-control-center-2.25.2/capplets/display/xrandr-capplet.c gnome-control-center-2.25.2-patched/capplets/display/xrandr-capplet.c
--- gnome-control-center-2.25.2/capplets/display/xrandr-capplet.c	2008-10-09 11:58:09.000000000 +0200
+++ gnome-control-center-2.25.2-patched/capplets/display/xrandr-capplet.c	2008-12-16 22:16:46.000000000 +0100
@@ -335,7 +335,8 @@
     const char *selection;
     GnomeRRRotation current;
     int i;
-
+	GError *error = NULL;
+	
     clear_combo (app->rotation_combo);
 
     gtk_widget_set_sensitive (
@@ -353,7 +354,7 @@
 
 	app->current_output->rotation = info->rotation;
 
-	if (gnome_rr_config_applicable (app->current_configuration, app->screen))
+	if (gnome_rr_config_applicable (app->current_configuration, app->screen, error))
 	{
  	    add_key (app->rotation_combo, info->name, 0, 0, 0, info->rotation);
 
@@ -1608,8 +1609,8 @@
 on_detect_displays (GtkWidget *widget, gpointer data)
 {
     App *app = data;
-
-    gnome_rr_screen_refresh (app->screen);
+	GError *error = NULL;
+    gnome_rr_screen_refresh (app->screen, error);
 }
 
 #define SHOW_ICON_KEY "/apps/gnome_settings_daemon/xrandr/show_notification_icon"
@@ -1641,9 +1642,9 @@
 	g_warning ("Could not open " GLADE_FILE);
 	return;
     }
-
+	GError *error = NULL;
     app->screen = gnome_rr_screen_new (gdk_screen_get_default (),
-				       on_screen_changed, app);
+				       on_screen_changed, app, error);
     if (!app->screen)
     {
 	g_warning (_("The X server does not support the XRANDR extension.  Runtime resolution changes to the display size are not available."));
