Index: gnome-control-center/capplets/default-applications/gnome-default-applications.xml.in
===================================================================
--- gnome-control-center.orig/capplets/default-applications/gnome-default-applications.xml.in	2010-03-07 11:51:56.573220885 -0500
+++ gnome-control-center/capplets/default-applications/gnome-default-applications.xml.in	2010-03-07 11:53:41.000000000 -0500
@@ -312,8 +312,8 @@
   <media-players>
     <media-player>
       <_name>Banshee Music Player</_name>
-      <executable>banshee</executable>
-      <command>banshee</command>
+      <executable>banshee-1</executable>
+      <command>banshee-1</command>
       <icon-name>music-player-banshee</icon-name>
       <run-in-terminal>false</run-in-terminal>
     </media-player>
