--- control-center-2.6.1/capplets/default-applications/gnome-default-applications-properties.c	2004-02-12 15:48:28.000000000 +0100
+++ capplets/default-applications/gnome-default-applications-properties.c	2004-08-01 16:01:04.000000000 +0200
@@ -392,9 +392,15 @@
 			g_free (browser);
 			return;
 		}
-        }
-	gtk_toggle_button_set_inconsistent (GTK_TOGGLE_BUTTON (WID ("web_select_radio")), TRUE);
-	gtk_toggle_button_set_inconsistent (GTK_TOGGLE_BUTTON (WID ("web_custom_radio")), TRUE);
+	}
+	if (browser && strlen(browser) != 0) {
+		gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (WID ("web_select_radio")), TRUE);
+		gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (WID ("web_custom_radio")), TRUE);
+	} else {
+		gtk_toggle_button_set_inconsistent (GTK_TOGGLE_BUTTON (WID ("web_select_radio")), TRUE);
+		gtk_toggle_button_set_inconsistent (GTK_TOGGLE_BUTTON (WID ("web_custom_radio")), TRUE);
+	}
+	
 	g_free (browser);
 }
 
@@ -431,8 +437,14 @@
 			return;
 		}
         }
-	gtk_toggle_button_set_inconsistent (GTK_TOGGLE_BUTTON (WID ("mail_select_radio")), TRUE);
-	gtk_toggle_button_set_inconsistent (GTK_TOGGLE_BUTTON (WID ("mail_custom_radio")), TRUE);
+		
+	if (mailer && strlen(mailer) != 0) {
+		gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (WID ("mail_select_radio")), TRUE);
+		gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (WID ("mail_custom_radio")), TRUE);
+	} else {
+		gtk_toggle_button_set_inconsistent (GTK_TOGGLE_BUTTON (WID ("mail_select_radio")), TRUE);
+		gtk_toggle_button_set_inconsistent (GTK_TOGGLE_BUTTON (WID ("mail_custom_radio")), TRUE);
+	}
 	g_free (mailer);
 
 
@@ -446,7 +458,7 @@
 	const gchar *browser = gtk_entry_get_text (GTK_ENTRY (entry));
 
 	for (i = 0; i < G_N_ELEMENTS (possible_browsers); i++ ) {
-		if (! strcmp (_(possible_browsers[i].name), browser)) {
+		if (! strcmp (_(possible_browsers[i].name), browser) && possible_browsers[i].in_path) {
 		        gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (WID ("web_custom_terminal_toggle")),
 						      possible_browsers[i].needs_term);
 			gtk_entry_set_text (GTK_ENTRY (WID ("web_custom_command_entry")),
@@ -464,7 +476,7 @@
 	const gchar *mailer = gtk_entry_get_text (GTK_ENTRY (entry));
 
 	for (i = 0; i < G_N_ELEMENTS (possible_mailers); i++ ) {
-		if (! strcmp (_(possible_mailers[i].name), mailer)) {
+		if (! strcmp (_(possible_mailers[i].name), mailer) && possible_mailers[i].in_path) {
 		        gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (WID ("mail_custom_terminal_toggle")),
 						      possible_mailers[i].needs_term);
 			gtk_entry_set_text (GTK_ENTRY (WID ("mail_custom_command_entry")),
@@ -525,7 +537,7 @@
 	const gchar *terminal = gtk_entry_get_text (GTK_ENTRY (entry));
 
 	for (i = 0; i < G_N_ELEMENTS (possible_terminals); i++ ) {
-		if (! strcmp (_(possible_terminals[i].name), terminal)) {
+		if (! strcmp (_(possible_terminals[i].name), terminal) && possible_terminals[i].in_path) {
 			gtk_entry_set_text (GTK_ENTRY (WID ("terminal_custom_command_entry")), possible_terminals[i].exec);
 			gtk_entry_set_text (GTK_ENTRY (WID ("terminal_custom_exec_entry")), possible_terminals[i].exec_arg);
 			return;
