Revert upstream patch a39b9219652c14bb7a39a8639d03ab1462d311c5
Index: gnome-control-center-3.4.2+git20121016.29d7c0/panels/printers/cc-printers-panel.c
===================================================================
--- gnome-control-center-3.4.2+git20121016.29d7c0.orig/panels/printers/cc-printers-panel.c	2012-11-07 18:52:59.000000000 +0100
+++ gnome-control-center-3.4.2+git20121016.29d7c0/panels/printers/cc-printers-panel.c	2012-11-07 19:25:59.292279172 +0100
@@ -216,6 +216,11 @@ cc_printers_panel_class_init (CcPrinters
 }
 
 static void
+cc_printers_panel_class_finalize (CcPrintersPanelClass *klass)
+{
+}
+
+static void
 on_cups_notification (GDBusConnection *connection,
                       const char      *sender_name,
                       const char      *object_path,
