diff -ur control-center-2.10.0/gnome-settings-daemon/gnome-settings-multimedia-keys.c control-center-2.10.0.new/gnome-settings-daemon/gnome-settings-multimedia-keys.c
--- control-center-2.10.0/gnome-settings-daemon/gnome-settings-multimedia-keys.c	2005-02-17 18:19:06.000000000 +0000
+++ control-center-2.10.0.new/gnome-settings-daemon/gnome-settings-multimedia-keys.c	2005-03-13 16:34:12.434483912 +0000
@@ -823,7 +823,7 @@
 		do_mail_action (acme);
 		break;
 	case SLEEP_KEY:
-		do_sleep_action ("apm", "xset dpms force off");
+		do_sleep_action ("gdm-signal -s", "xset dpms force off");
 		break;
 	case SCREENSAVER_KEY:
 		execute ("xscreensaver-command -lock", FALSE);
Only in control-center-2.10.0.new/gnome-settings-daemon: gnome-settings-multimedia-keys.c~
