diff --git a/panels/screen/cc-screen-panel.c b/panels/screen/cc-screen-panel.c
index 4bf5bfa..bd91d17 100644
--- a/panels/screen/cc-screen-panel.c
+++ b/panels/screen/cc-screen-panel.c
@@ -393,8 +393,7 @@ set_dpms_value_for_combo (GtkComboBox *combo_box, CcScreenPanel *self)
       gtk_tree_model_get (model, &iter,
                           1, &value_tmp,
                           -1);
-      if (value == value_tmp ||
-          (value_tmp > value_prev && value < value_tmp))
+      if (value == value_tmp)
         {
           gtk_combo_box_set_active_iter (combo_box, &iter);
           return;
diff --git a/panels/screen/screen.ui b/panels/screen/screen.ui
index 3af6c4c..892fb6e 100644
--- a/panels/screen/screen.ui
+++ b/panels/screen/screen.ui
@@ -84,6 +84,10 @@
         <col id="0" translatable="yes">1 hour</col>
         <col id="1">3600</col>
       </row>
+      <row>
+	<col id="0" translatable="yes">Never</col>
+	<col id="1">0</col>
+      </row>
     </data>
   </object>
   <object class="GtkWindow" id="window1">
