commit 56d33f88cd118b70c3e1200a989f6b82f9ac0e03
Author: Frédéric Péters <fpeters@0d.be>
Date:   Sat Mar 12 20:49:20 2011 +0100

    Add missing timezone files

diff --git a/panels/datetime/Makefile.am b/panels/datetime/Makefile.am
index 4384a28..05bd7ef 100644
--- a/panels/datetime/Makefile.am
+++ b/panels/datetime/Makefile.am
@@ -34,6 +34,7 @@ dist_ui_DATA = \
 	data/timezone_12.png \
 	data/timezone_12.75.png \
 	data/timezone_13.png \
+	data/timezone_14.png \
 	data/timezone_-2.png \
 	data/timezone_2.png \
 	data/timezone_-3.png \
@@ -56,6 +57,7 @@ dist_ui_DATA = \
 	data/timezone_7.png \
 	data/timezone_-8.png \
 	data/timezone_8.png \
+	data/timezone_8.75.png \
 	data/timezone_-9.png \
 	data/timezone_9.png \
 	data/timezone_-9.5.png \
