--- typing-break/drw-monitor.c.orig	2004-09-15 03:35:59.000000000 +0200
+++ typing-break/drw-monitor.c	2004-09-15 03:36:04.000000000 +0200
@@ -39,8 +39,6 @@
 	XScreenSaverInfo *ss_info;
 	guint             timeout_id;
 	unsigned long     last_idle;
-
-	time_t            last_activity;
 };
 
 /* Signals */
@@ -142,19 +140,12 @@
 drw_monitor_timeout (DrwMonitor *monitor)
 {
 	DrwMonitorPriv *priv;
- 	time_t          now;
 
 	priv = monitor->priv;
 	
 	if (XScreenSaverQueryInfo (GDK_DISPLAY (), DefaultRootWindow (GDK_DISPLAY ()), priv->ss_info) != 0) {
 		if (priv->ss_info->idle < priv->last_idle) {
- 			now = time (NULL);
- 
- 			if (now - priv->last_activity < 25) {
- 				g_signal_emit (monitor, signals[ACTIVITY], 0, NULL);
- 			}
- 			
- 			priv->last_activity = now;
+			g_signal_emit (monitor, signals[ACTIVITY], 0, NULL);			
 		}
 
 		priv->last_idle = priv->ss_info->idle;
@@ -178,8 +169,6 @@
 
 	priv->ss_info = XScreenSaverAllocInfo ();
 
-	priv->last_activity = time (NULL);
-	
 	priv->timeout_id = g_timeout_add (3000, (GSourceFunc) drw_monitor_timeout, monitor);
 	
 	return TRUE;
