diff -Nur control-center-2.19.5/capplets/common/gconf-property-editor.c control-center-2.19.5.new/capplets/common/gconf-property-editor.c
--- control-center-2.19.5/capplets/common/gconf-property-editor.c	2007-07-02 14:30:11.000000000 +0200
+++ control-center-2.19.5.new/capplets/common/gconf-property-editor.c	2007-07-11 13:50:33.000000000 +0200
@@ -511,12 +511,14 @@
 
 	if (entry && (value = gconf_entry_get_value (entry))) {
 		value_wid = peditor->p->conv_to_widget_cb (peditor, value);
-		entry_current_text = gtk_entry_get_text (GTK_ENTRY (peditor->p->ui_control));
-		entry_current_integer = strtol (entry_current_text, NULL, 10);
-		if (entry_current_integer != gconf_value_get_int (value)) {
-			char *buf = g_strdup_printf ("%d", gconf_value_get_int (value_wid));
-			gtk_entry_set_text (GTK_ENTRY (peditor->p->ui_control), buf);
-			g_free (buf);
+		if (value->type == GCONF_VALUE_STRING) {
+			entry_current_text = gtk_entry_get_text (GTK_ENTRY (peditor->p->ui_control));
+			entry_current_integer = strtol (entry_current_text, NULL, 10);
+			if (entry_current_integer != gconf_value_get_int (value)) {
+				char *buf = g_strdup_printf ("%d", gconf_value_get_int (value_wid));
+				gtk_entry_set_text (GTK_ENTRY (peditor->p->ui_control), buf);
+				g_free (buf);
+			}
 		}
 		gconf_value_free (value_wid);
 	}
