diff -u -r1.13 gnome-default-applications-properties.glade
--- capplets/default-applications/gnome-default-applications-properties.glade.orig	9 Jan 2004 14:53:58 -0000	1.13
+++ capplets/default-applications/gnome-default-applications-properties.glade	22 Nov 2004 09:25:59 -0000
@@ -641,7 +641,6 @@
 	  <child>
 	    <widget class="GtkFrame" id="frame1">
 	      <property name="border_width">12</property>
-	      <property name="visible">True</property>
 	      <property name="label_xalign">0</property>
 	      <property name="label_yalign">0.5</property>
 	      <property name="shadow_type">GTK_SHADOW_NONE</property>
