Description: Don't list Mozilla commands which are compatibility symlinks
Bug: http://bugzilla.gnome.org/show_bug.cgi?id=499327
Bug-Ubuntu: http://bugs.launchpad.net/bugs/55963

diff -Nur -x '*.orig' -x '*~' gnome-control-center-2.27.3/capplets/default-applications/gnome-default-applications.xml.in gnome-control-center-2.27.3.new/capplets/default-applications/gnome-default-applications.xml.in
--- gnome-control-center-2.27.3/capplets/default-applications/gnome-default-applications.xml.in	2009-07-13 19:45:12.376642974 +0100
+++ gnome-control-center-2.27.3.new/capplets/default-applications/gnome-default-applications.xml.in	2009-07-13 19:45:14.536643756 +0100
@@ -74,26 +74,6 @@
       <win-command>iceweasel -new-window "%s"</win-command>
     </web-browser>
     <web-browser>
-      <_name>Mozilla 1.6</_name>
-      <executable>mozilla-1.6</executable>
-      <command>mozilla-1.6 %s</command>
-      <icon-name>mozilla-icon</icon-name>
-      <run-in-terminal>false</run-in-terminal>
-      <netscape-remote>true</netscape-remote>
-      <tab-command>mozilla-1.6 -remote "openurl(%s,new-tab)"</tab-command>
-      <win-command>mozilla-1.6 -remote "openurl(%s,new-window)"</win-command>
-    </web-browser>
-    <web-browser>
-      <_name>Mozilla</_name>
-      <executable>mozilla</executable>
-      <command>mozilla %s</command>
-      <icon-name>mozilla-icon</icon-name>
-      <run-in-terminal>false</run-in-terminal>
-      <netscape-remote>true</netscape-remote>
-      <tab-command>mozilla -remote "openurl(%s,new-tab)"</tab-command>
-      <win-command>mozilla -remote "openurl(%s,new-window)"</win-command>
-    </web-browser>
-    <web-browser>
       <_name>SeaMonkey</_name>
       <executable>seamonkey</executable>
       <command>seamonkey %s</command>
@@ -185,13 +165,6 @@
       <run-in-terminal>false</run-in-terminal>
     </mail-reader>
     <mail-reader>
-      <_name>Mozilla Mail</_name>
-      <executable>mozilla</executable>
-      <command>mozilla -mail %s</command>
-      <icon-name>mozilla-mail-icon</icon-name>
-      <run-in-terminal>false</run-in-terminal>
-    </mail-reader>
-    <mail-reader>
       <_name>SeaMonkey Mail</_name>
       <executable>seamonkey</executable>
       <command>seamonkey -mail %s</command>
