diff -Nur -x '*.orig' -x '*~' gnome-control-center-2.22.1/capplets/default-applications/gnome-default-applications.xml.in gnome-control-center-2.22.1.new/capplets/default-applications/gnome-default-applications.xml.in
--- gnome-control-center-2.22.1/capplets/default-applications/gnome-default-applications.xml.in	2008-04-14 16:00:59.000000000 +0100
+++ gnome-control-center-2.22.1.new/capplets/default-applications/gnome-default-applications.xml.in	2008-04-14 16:01:36.000000000 +0100
@@ -57,7 +57,7 @@
       <_name>Firefox</_name>
       <executable>firefox</executable>
       <command>firefox %s</command>
-      <icon-name>firefox</icon-name>
+      <icon-name>firefox-3.0</icon-name>
       <run-in-terminal>false</run-in-terminal>
       <netscape-remote>true</netscape-remote>
       <tab-command>firefox -new-tab "%s"</tab-command>
