diff -Nur gnome-control-center-2.19.1/capplets/keybindings/01-desktop-key.xml.in gnome-control-center-2.19.1.new/capplets/keybindings/01-desktop-key.xml.in
--- gnome-control-center-2.19.1/capplets/keybindings/01-desktop-key.xml.in	2007-03-21 12:35:48.000000000 +0100
+++ gnome-control-center-2.19.1.new/capplets/keybindings/01-desktop-key.xml.in	2007-05-03 23:32:13.000000000 +0200
@@ -25,5 +25,8 @@
 	<KeyListEntry
 	name="/apps/gnome_settings_daemon/keybindings/www" />
 
+	<KeyListEntry	
+	name="/apps/gnome_settings_daemon/keybindings/calculator" />
+
 </KeyListEntries>
 
diff -Nur gnome-control-center-2.19.1/gnome-settings-daemon/actions/acme.h gnome-control-center-2.19.1.new/gnome-settings-daemon/actions/acme.h
--- gnome-control-center-2.19.1/gnome-settings-daemon/actions/acme.h	2007-03-21 12:35:47.000000000 +0100
+++ gnome-control-center-2.19.1.new/gnome-settings-daemon/actions/acme.h	2007-05-03 23:31:22.000000000 +0200
@@ -33,6 +33,7 @@
 	EJECT_KEY,
 	HOME_KEY,
 	MEDIA_KEY,
+	CALCULATOR_KEY,
 	SEARCH_KEY,
 	EMAIL_KEY,
 	SLEEP_KEY,
@@ -65,6 +66,7 @@
 	{ EJECT_KEY, GCONF_BINDING_DIR "/eject", NULL },
 	{ HOME_KEY, GCONF_BINDING_DIR "/home", NULL },
 	{ MEDIA_KEY, GCONF_BINDING_DIR "/media", NULL },
+	{ CALCULATOR_KEY, GCONF_BINDING_DIR "/calculator", NULL },
 	{ SEARCH_KEY, GCONF_BINDING_DIR "/search", NULL },
 	{ EMAIL_KEY, GCONF_BINDING_DIR "/email", NULL },
 	{ SLEEP_KEY, GCONF_BINDING_DIR "/sleep", NULL },
diff -Nur gnome-control-center-2.19.1/gnome-settings-daemon/gnome-settings-multimedia-keys.c gnome-control-center-2.19.1.new/gnome-settings-daemon/gnome-settings-multimedia-keys.c
--- gnome-control-center-2.19.1/gnome-settings-daemon/gnome-settings-multimedia-keys.c	2007-04-02 12:35:01.000000000 +0200
+++ gnome-control-center-2.19.1.new/gnome-settings-daemon/gnome-settings-multimedia-keys.c	2007-05-03 23:31:22.000000000 +0200
@@ -673,6 +673,8 @@
 	case MEDIA_KEY:
 		do_media_action (acme);
 		break;
+	case CALCULATOR_KEY:
+		execute (acme, "gcalctool", FALSE, FALSE);
 	case PLAY_KEY:
 		return do_multimedia_player_action (acme, "Play");
 		break;
diff -Nur gnome-control-center-2.19.1/schemas/apps_gnome_settings_daemon_keybindings.schemas.in gnome-control-center-2.19.1.new/schemas/apps_gnome_settings_daemon_keybindings.schemas.in
--- gnome-control-center-2.19.1/schemas/apps_gnome_settings_daemon_keybindings.schemas.in	2007-03-21 12:31:15.000000000 +0100
+++ gnome-control-center-2.19.1.new/schemas/apps_gnome_settings_daemon_keybindings.schemas.in	2007-05-03 23:31:22.000000000 +0200
@@ -144,6 +144,17 @@
             </locale>
         </schema>
 
+        <schema>
+            <key>/schemas/apps/gnome_settings_daemon/keybindings/calculator</key>
+            <applyto>/apps/gnome_settings_daemon/keybindings/calculator</applyto>
+            <type>string</type>
+            <default></default>
+            <locale name="C">
+                <short>Launch calculator</short>
+                <long>Launch calculator's shortcut</long>
+            </locale>
+        </schema>
+
 
         <schema>
             <key>/schemas/apps/gnome_settings_daemon/keybindings/www</key>
