--- control-center-2.12.1/control-center/control-center-categories.c.orig	2005-11-21 15:16:35.000000000 +0100
+++ control-center-2.12.1/control-center/control-center-categories.c	2005-11-21 16:08:43.000000000 +0100
@@ -251,7 +251,7 @@
 
 	information = g_new0 (ControlCenterInformation, 1);
 
-	menu_tree = gmenu_tree_lookup ("preferences.menu", GMENU_TREE_FLAGS_NONE);
+	menu_tree = gmenu_tree_lookup ("gnome-preferences.menu", GMENU_TREE_FLAGS_NONE);
 
 	if (!(menu_root = gmenu_tree_get_root_directory (menu_tree))) {
 		gmenu_tree_unref (menu_tree);
