--- control-center/control-center-categories.c.orig	2005-06-14 23:15:20.298224184 +0200
+++ control-center/control-center-categories.c	2005-06-14 23:15:33.778174920 +0200
@@ -234,7 +234,7 @@
 
 	information = g_new0 (ControlCenterInformation, 1);
 
-	menu_tree = menu_tree_lookup ("preferences.menu");
+	menu_tree = menu_tree_lookup ("gnome-preferences.menu");
 
 	if (!(menu_root = menu_tree_get_root_directory (menu_tree))) {
 		menu_tree_unref (menu_tree);
