diff -Nur control-center-2.16.3/capplets/windows/Makefile.am control-center-2.16.3.new/capplets/windows/Makefile.am
--- control-center-2.16.3/capplets/windows/Makefile.am	2007-01-29 12:10:38.000000000 +0100
+++ control-center-2.16.3.new/capplets/windows/Makefile.am	2007-03-17 15:25:59.000000000 +0100
@@ -17,7 +17,7 @@
 
 INCLUDES   = $(GNOMECC_CAPPLETS_CFLAGS)					\
 	     -I$(top_srcdir)/libwindow-settings \
-	     -DGNOME_WINDOW_MANAGER_MODULE_PATH=\""$(libdir)/window-manager-settings"\" \
+	     -DGNOME_WINDOW_MANAGER_MODULE_PATH=\""$(libdir)/libgnome-window-settings1"\" \
 	     -DGLADEDIR=\""$(gladedir)"\"  	\
 	     -DPIXMAPDIR=\""$(pixmapdir)"\"
 
diff -Nur control-center-2.16.3/capplets/windows/Makefile.in control-center-2.16.3.new/capplets/windows/Makefile.in
--- control-center-2.16.3/capplets/windows/Makefile.in	2007-01-30 10:28:40.000000000 +0100
+++ control-center-2.16.3.new/capplets/windows/Makefile.in	2007-03-17 15:25:59.000000000 +0100
@@ -343,7 +343,7 @@
 desktop_DATA = $(Desktop_in_files:.desktop.in=.desktop)
 INCLUDES = $(GNOMECC_CAPPLETS_CFLAGS)					\
 	     -I$(top_srcdir)/libwindow-settings \
-	     -DGNOME_WINDOW_MANAGER_MODULE_PATH=\""$(libdir)/window-manager-settings"\" \
+	     -DGNOME_WINDOW_MANAGER_MODULE_PATH=\""$(libdir)/libgnome-window-settings1"\" \
 	     -DGLADEDIR=\""$(gladedir)"\"  	\
 	     -DPIXMAPDIR=\""$(pixmapdir)"\"
 
diff -Nur control-center-2.16.3/libwindow-settings/Makefile.am control-center-2.16.3.new/libwindow-settings/Makefile.am
--- control-center-2.16.3/libwindow-settings/Makefile.am	2007-01-29 12:10:31.000000000 +0100
+++ control-center-2.16.3.new/libwindow-settings/Makefile.am	2007-03-17 15:25:59.000000000 +0100
@@ -1,4 +1,4 @@
-WM_MODULE_DIR=$(libdir)/window-manager-settings
+WM_MODULE_DIR=$(libdir)/libgnome-window-settings1
 
 INCLUDES = 								\
         -DGNOMELOCALEDIR=\""$(prefix)/$(DATADIRNAME)/locale"\" 		\
diff -Nur control-center-2.16.3/libwindow-settings/Makefile.in control-center-2.16.3.new/libwindow-settings/Makefile.in
--- control-center-2.16.3/libwindow-settings/Makefile.in	2007-01-30 10:28:47.000000000 +0100
+++ control-center-2.16.3.new/libwindow-settings/Makefile.in	2007-03-17 15:26:58.000000000 +0100
@@ -345,7 +345,7 @@
 target_alias = @target_alias@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-WM_MODULE_DIR = $(libdir)/window-manager-settings
+WM_MODULE_DIR = $(libdir)/libgnome-window-settings1
 INCLUDES = \
         -DGNOMELOCALEDIR=\""$(prefix)/$(DATADIRNAME)/locale"\" 		\
 	-DGNOME_ICONDIR=\""${prefix}/share/pixmaps"\"			\
