diff -Nur -x '*.orig' -x '*~' gnome-control-center-2.27.91/capplets/windows/Makefile.am gnome-control-center-2.27.91.new/capplets/windows/Makefile.am
--- gnome-control-center-2.27.91/capplets/windows/Makefile.am	2009-08-31 02:07:27.188856277 +0100
+++ gnome-control-center-2.27.91.new/capplets/windows/Makefile.am	2009-08-31 02:07:31.860858186 +0100
@@ -20,7 +20,7 @@
 
 INCLUDES   = $(GNOMECC_CAPPLETS_CFLAGS)					\
 	     -I$(top_srcdir)/libwindow-settings \
-	     -DGNOME_WINDOW_MANAGER_MODULE_PATH=\""$(libdir)/window-manager-settings"\" \
+	     -DGNOME_WINDOW_MANAGER_MODULE_PATH=\""$(libdir)/libgnome-window-settings1"\" \
 	     -DUIDIR=\""$(uidir)"\"  	\
 	     -DGNOMELOCALEDIR="\"$(datadir)/locale\""			\
 	     -DPIXMAPDIR=\""$(pixmapdir)"\"
diff -Nur -x '*.orig' -x '*~' gnome-control-center-2.27.91/libwindow-settings/Makefile.am gnome-control-center-2.27.91.new/libwindow-settings/Makefile.am
--- gnome-control-center-2.27.91/libwindow-settings/Makefile.am	2009-08-31 02:07:25.120854240 +0100
+++ gnome-control-center-2.27.91.new/libwindow-settings/Makefile.am	2009-08-31 02:07:31.860858186 +0100
@@ -1,4 +1,4 @@
-WM_MODULE_DIR=$(libdir)/window-manager-settings
+WM_MODULE_DIR=$(libdir)/libgnome-window-settings1
 
 INCLUDES = 								\
 	-DGNOMELOCALEDIR="\"$(datadir)/locale\""			\
