Index: gnome-control-center-2.24.0.1/capplets/windows/Makefile.am
===================================================================
--- gnome-control-center-2.24.0.1.orig/capplets/windows/Makefile.am	2008-12-28 13:23:47.476361870 +0100
+++ gnome-control-center-2.24.0.1/capplets/windows/Makefile.am	2008-12-28 13:25:03.059860400 +0100
@@ -20,7 +20,7 @@ desktop_DATA = $(Desktop_in_files:.deskt
 
 INCLUDES   = $(GNOMECC_CAPPLETS_CFLAGS)					\
 	     -I$(top_srcdir)/libwindow-settings \
-	     -DGNOME_WINDOW_MANAGER_MODULE_PATH=\""$(libdir)/window-manager-settings"\" \
+	     -DGNOME_WINDOW_MANAGER_MODULE_PATH=\""$(libdir)/libgnome-window-settings1"\" \
 	     -DGLADEDIR=\""$(gladedir)"\"  	\
 	     -DGNOMELOCALEDIR="\"$(datadir)/locale\""			\
 	     -DPIXMAPDIR=\""$(pixmapdir)"\"
Index: gnome-control-center-2.24.0.1/libwindow-settings/Makefile.am
===================================================================
--- gnome-control-center-2.24.0.1.orig/libwindow-settings/Makefile.am	2008-12-28 13:23:47.380355813 +0100
+++ gnome-control-center-2.24.0.1/libwindow-settings/Makefile.am	2008-12-28 13:25:03.059860400 +0100
@@ -1,4 +1,4 @@
-WM_MODULE_DIR=$(libdir)/window-manager-settings
+WM_MODULE_DIR=$(libdir)/libgnome-window-settings1
 
 INCLUDES = 								\
 	-DGNOMELOCALEDIR="\"$(datadir)/locale\""			\
