diff -Nur -x '*.orig' -x '*~' gnome-control-center-2.27.3/capplets/windows/Makefile.am gnome-control-center-2.27.3.new/capplets/windows/Makefile.am
--- gnome-control-center-2.27.3/capplets/windows/Makefile.am	2009-07-13 19:46:22.300645017 +0100
+++ gnome-control-center-2.27.3.new/capplets/windows/Makefile.am	2009-07-13 19:46:26.256643387 +0100
@@ -20,7 +20,7 @@
 
 INCLUDES   = $(GNOMECC_CAPPLETS_CFLAGS)					\
 	     -I$(top_srcdir)/libwindow-settings \
-	     -DGNOME_WINDOW_MANAGER_MODULE_PATH=\""$(libdir)/window-manager-settings"\" \
+	     -DGNOME_WINDOW_MANAGER_MODULE_PATH=\""$(libdir)/libgnome-window-settings1"\" \
 	     -DGLADEDIR=\""$(gladedir)"\"  	\
 	     -DGNOMELOCALEDIR="\"$(datadir)/locale\""			\
 	     -DPIXMAPDIR=\""$(pixmapdir)"\"
diff -Nur -x '*.orig' -x '*~' gnome-control-center-2.27.3/libwindow-settings/Makefile.am gnome-control-center-2.27.3.new/libwindow-settings/Makefile.am
--- gnome-control-center-2.27.3/libwindow-settings/Makefile.am	2009-07-13 19:46:20.384645826 +0100
+++ gnome-control-center-2.27.3.new/libwindow-settings/Makefile.am	2009-07-13 19:46:26.256643387 +0100
@@ -1,4 +1,4 @@
-WM_MODULE_DIR=$(libdir)/window-manager-settings
+WM_MODULE_DIR=$(libdir)/libgnome-window-settings1
 
 INCLUDES = 								\
 	-DGNOMELOCALEDIR="\"$(datadir)/locale\""			\
