--- libwindow-settings/Makefile.am	2005-07-20 14:10:55.000000000 +0200
+++ libwindow-settings/Makefile.am.new	2006-05-10 17:38:27.000000000 +0200
@@ -1,4 +1,4 @@
-WM_MODULE_DIR=$(libdir)/window-manager-settings
+WM_MODULE_DIR=$(libdir)/libgnome-window-settings1
 
 INCLUDES = 								\
         -DGNOMELOCALEDIR=\""$(prefix)/$(DATADIRNAME)/locale"\" 		\
--- libwindow-settings/Makefile.in	2006-04-10 19:03:25.000000000 +0200
+++ libwindow-settings/Makefile.in.new	2006-05-10 17:38:53.000000000 +0200
@@ -353,7 +353,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-WM_MODULE_DIR = $(libdir)/window-manager-settings
+WM_MODULE_DIR = $(libdir)/libgnome-window-settings1
 INCLUDES = \
         -DGNOMELOCALEDIR=\""$(prefix)/$(DATADIRNAME)/locale"\" 		\
 	-DGNOME_ICONDIR=\""${prefix}/share/pixmaps"\"			\
--- capplets/windows/Makefile.am	2004-11-28 09:14:05.000000000 +0100
+++ capplets/windows/Makefile.am.new	2006-05-10 17:37:25.000000000 +0200
@@ -17,7 +17,7 @@
 
 INCLUDES   = $(GNOMECC_CAPPLETS_CFLAGS)					\
 	     -I$(top_srcdir)/libwindow-settings \
-	     -DGNOME_WINDOW_MANAGER_MODULE_PATH=\""$(libdir)/window-manager-settings"\" \
+	     -DGNOME_WINDOW_MANAGER_MODULE_PATH=\""$(libdir)/libgnome-window-settings1"\" \
 	     -DGLADEDIR=\""$(gladedir)"\"  	\
 	     -DPIXMAPDIR=\""$(pixmapdir)"\"
 
--- capplets/windows/Makefile.in	2006-04-10 19:03:22.000000000 +0200
+++ capplets/windows/Makefile.in.new	2006-05-10 17:37:41.000000000 +0200
@@ -358,7 +358,7 @@
 desktop_DATA = $(Desktop_in_files:.desktop.in=.desktop)
 INCLUDES = $(GNOMECC_CAPPLETS_CFLAGS)					\
 	     -I$(top_srcdir)/libwindow-settings \
-	     -DGNOME_WINDOW_MANAGER_MODULE_PATH=\""$(libdir)/window-manager-settings"\" \
+	     -DGNOME_WINDOW_MANAGER_MODULE_PATH=\""$(libdir)/libgnome-window-settings1"\" \
 	     -DGLADEDIR=\""$(gladedir)"\"  	\
 	     -DPIXMAPDIR=\""$(pixmapdir)"\"
 
