diff -Nur control-center-2.14.2/gnome-settings-daemon/gnome-settings-multimedia-keys.c control-center-2.14.2.new/gnome-settings-daemon/gnome-settings-multimedia-keys.c
--- control-center-2.14.2/gnome-settings-daemon/gnome-settings-multimedia-keys.c	2006-06-13 23:34:24.000000000 +0200
+++ control-center-2.14.2.new/gnome-settings-daemon/gnome-settings-multimedia-keys.c	2006-06-13 23:38:08.000000000 +0200
@@ -935,9 +935,11 @@
 	/* initialise Frame Buffer level handler */
 	acme->levobj = acme_fblevel_new (&err);
 	if (acme->levobj == NULL && err != NULL) {
+    /*
 		if (!g_error_matches (err, ACME_FBLEVEL_ERROR,
 				      ACME_FBLEVEL_ERROR_NO_POWERBOOK))
 			acme_error (err->message);
+      */
 		g_error_free (err);
 	}
 #endif
