--- control-center-2.14.2/capplets/theme-switcher/gnome-theme-installer.c~      2006-01-05 20:55:00.000000000 +0100
+++ control-center-2.14.2/capplets/theme-switcher/gnome-theme-installer.c       2006-07-13 21:17:47.000000000 +0200
@@ -176,7 +176,7 @@
 	
 	filename = g_shell_quote(theme_props->filename);
 	/* this should be something more clever and nonblocking */
-	command = g_strdup_printf ("sh -c 'cd \"%s\"; /usr/bin/bzip2 -d -c < \"%s\" | /bin/tar xf - '",
+	command = g_strdup_printf ("sh -c 'cd \"%s\"; bzip2 -d -c < \"%s\" | tar xf - '",
 				   theme_props->target_tmp_dir, filename);
 	g_free (filename);
 	if (g_spawn_command_line_sync (command, NULL, NULL, &status, NULL) && status == 0) {
@@ -215,7 +215,7 @@
 		g_free (filename);
 	} else if (path && len > 8 && !strcmp (path + len - 8, ".tar.bz2")) {
 		filename = g_shell_quote (path);
-		command = g_strdup_printf ("sh -c '/usr/bin/bzip2 -d -c < \"%s\" | /bin/tar ft - | head -n 1'",
+		command = g_strdup_printf ("sh -c 'bzip2 -d -c < \"%s\" | tar ft - | head -n 1'",
 					    filename);
 		theme_props->filetype=TARBZ;
 		g_free (filename);
@@ -252,7 +252,7 @@
 	theme_props->filename=g_strdup(path);
 	
 	if (theme_props->filetype == TARBZ ) {
-		if (!g_file_test ("/usr/bin/bzip2", G_FILE_TEST_EXISTS)) {
+		if (g_find_program_in_path ("bzip2") == NULL) {
 			dialog = gtk_message_dialog_new (NULL,
 						GTK_DIALOG_MODAL,
 						GTK_MESSAGE_ERROR,
