diff -Nur gnome-control-center-2.19.4/capplets/default-applications/gnome-at-commandline.in.in gnome-control-center-2.19.4.new/capplets/default-applications/gnome-at-commandline.in.in
--- gnome-control-center-2.19.4/capplets/default-applications/gnome-at-commandline.in.in	2007-04-23 16:58:28.000000000 +0200
+++ gnome-control-center-2.19.4.new/capplets/default-applications/gnome-at-commandline.in.in	2007-07-04 23:33:08.000000000 +0200
@@ -51,7 +51,7 @@
 	
 	if [ ! -z "$AUTOSTART" ]; then
 		# assuming ran from /usr/share/gnome/autostart
-		if [ "$STARTUP" == "true" ]; then
+		if [ "$STARTUP" = "true" ]; then
 			# gconf indicated requested autostart
 			($CMDLINE &)
 		fi
