diff -urN control-center-2.14.2.orig/capplets/default-applications/gnome-default-applications.xml.in control-center-2.14.2/capplets/default-applications/gnome-default-applications.xml.in
--- control-center-2.14.2.orig/capplets/default-applications/gnome-default-applications.xml.in	2006-03-28 21:52:57.000000000 +0200
+++ control-center-2.14.2/capplets/default-applications/gnome-default-applications.xml.in	2006-08-18 15:31:30.000000000 +0200
@@ -202,7 +202,7 @@
     <mail-reader>
       <_name>Thunderbird</_name>
       <executable>thunderbird</executable>
-      <command>thunderbird -compose %s</command>
+      <command>thunderbird %s</command>
       <icon-name>thunderbird</icon-name>
       <run-in-terminal>false</run-in-terminal>
     </mail-reader>
