diff -rup ../control-center-2.14.2-orig/capplets/default-applications/gnome-default-applications.xml.in ./capplets/default-applications/gnome-default-applications.xml.in
--- ../control-center-2.14.2-orig/capplets/default-applications/gnome-default-applications.xml.in	2006-11-21 20:23:23.000000000 +0100
+++ ./capplets/default-applications/gnome-default-applications.xml.in	2006-11-23 16:56:50.000000000 +0100
@@ -72,6 +72,16 @@
       <win-command>firefox -remote "openurl(%s,new-window)"</win-command>
     </web-browser>
     <web-browser>
+      <_name>Iceweasel</_name>
+      <executable>iceweasel</executable>
+      <command>iceweasel %s</command>
+      <icon-name>iceweasel</icon-name>
+      <run-in-terminal>false</run-in-terminal>
+      <netscape-remote>true</netscape-remote>
+      <tab-command>iceweasel -remote "openurl(%s,new-tab)"</tab-command>
+      <win-command>iceweasel -remote "openurl(%s,new-window)"</win-command>
+    </web-browser>
+    <web-browser>
       <_name>Mozilla 1.6</_name>
       <executable>mozilla-1.6</executable>
       <command>mozilla-1.6 %s</command>
@@ -200,6 +210,13 @@
       <run-in-terminal>false</run-in-terminal>
     </mail-reader>
     <mail-reader>
+      <_name>Icedove</_name>
+      <executable>icedove</executable>
+      <command>icedove -compose %s</command>
+      <icon-name>icedove</icon-name>
+      <run-in-terminal>false</run-in-terminal>
+    </mail-reader>
+    <mail-reader>
       <_name>Thunderbird</_name>
       <executable>thunderbird</executable>
       <command>thunderbird -compose %s</command>
