--- gnome-settings-daemon/gnome-settings-daemon.pc.in.orig	2007-11-03 02:19:59.340704956 +0100
+++ gnome-settings-daemon/gnome-settings-daemon.pc.in	2007-11-03 02:20:30.994508804 +0100
@@ -2,6 +2,8 @@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
+libexecdir=@libexecdir@
+binary=${libexecdir}/gnome-settings-daemon
 
 Name: gnome-settings-daemon
 Description: Utility library for accessing gnome-settings-daemon over DBUS
