diff -Nur control-center-2.16.3/capplets/sound/sound-properties-capplet.c control-center-2.16.3.new/capplets/sound/sound-properties-capplet.c
--- control-center-2.16.3/capplets/sound/sound-properties-capplet.c	2007-01-29 12:10:37.000000000 +0100
+++ control-center-2.16.3.new/capplets/sound/sound-properties-capplet.c	2007-03-17 15:32:57.000000000 +0100
@@ -662,8 +662,8 @@
 	add_device (AUDIO_CAPTURE, "esdmon", _("ESD - Enlightenment Sound Daemon"), NULL);
 	add_device (AUDIO_PLAYBACK, "osssink", _("OSS - Open Sound System"), NULL);
 	add_device (AUDIO_CAPTURE, "osssrc", _("OSS - Open Sound System"), NULL);
-	add_device (AUDIO_PLAYBACK, "polypsink", _("Polypaudio Sound Server"), NULL);
-	add_device (AUDIO_CAPTURE, "polypsrc", _("Polypaudio Sound Server"), NULL);
+	add_device (AUDIO_PLAYBACK, "pulsesink", _("PulseAudio Sound Server"), NULL);
+	add_device (AUDIO_CAPTURE, "pulsesrc", _("PulseAudio Sound Server"), NULL);
 	add_device (AUDIO_CAPTURE, "audiotestsrc wave=triangle is-live=true", _("Test Sound"), NULL);
 	add_device (AUDIO_CAPTURE, "audiotestsrc wave=silence is-live=true", _("Silence"), NULL);
 	
