diff -rup ../control-center-2.16.2.orig/capplets/sound/sound-properties-capplet.c ./capplets/sound/sound-properties-capplet.c
--- ../control-center-2.16.2.orig/capplets/sound/sound-properties-capplet.c	2007-02-05 10:12:50.000000000 +0100
+++ ./capplets/sound/sound-properties-capplet.c	2007-02-05 10:16:42.000000000 +0100
@@ -642,8 +642,8 @@ setup_devices ()
 	add_device (AUDIO_CAPTURE, "esdmon", "ESD - Enlightenment Sound Daemon", NULL);
 	add_device (AUDIO_PLAYBACK, "osssink", "OSS - Open Sound System", NULL);
 	add_device (AUDIO_CAPTURE, "osssrc", "OSS - Open Sound System", NULL);
-	add_device (AUDIO_PLAYBACK, "polypsink", "Polypaudio Sound Server", NULL);
-	add_device (AUDIO_CAPTURE, "polypsrc", "Polypaudio Sound Server", NULL);
+	add_device (AUDIO_PLAYBACK, "pulsesink", "PulseAudio Sound Server", NULL);
+	add_device (AUDIO_CAPTURE, "pulsesrc", "PulseAudio Sound Server", NULL);
 	add_device (AUDIO_CAPTURE, "audiotestsrc wave=triangle is-live=true", "Test Sound", NULL);
 	add_device (AUDIO_CAPTURE, "audiotestsrc wave=silence is-live=true", "Silence", NULL);
 	
