diff -ruN control-center-2.17.91/capplets/common/wm-common.h control-center-2.17.91.new/capplets/common/wm-common.h
--- control-center-2.17.91/capplets/common/wm-common.h	2007-01-02 13:47:02.000000000 +0100
+++ control-center-2.17.91.new/capplets/common/wm-common.h	2007-02-18 14:56:08.000000000 +0100
@@ -3,6 +3,7 @@
 
 #define WM_COMMON_METACITY "Metacity"
 #define WM_COMMON_SAWFISH  "Sawfish"
+#define WM_COMMON_COMPIZ "compiz"
 #define WM_COMMON_UNKNOWN  "Unknown"
 
 gchar *wm_common_get_current_window_manager (void);
diff -ruN control-center-2.17.91/capplets/keybindings/gnome-keybinding-properties.c control-center-2.17.91.new/capplets/keybindings/gnome-keybinding-properties.c
--- control-center-2.17.91/capplets/keybindings/gnome-keybinding-properties.c	2007-02-12 10:28:24.000000000 +0100
+++ control-center-2.17.91.new/capplets/keybindings/gnome-keybinding-properties.c	2007-02-18 14:55:20.000000000 +0100
@@ -568,7 +568,7 @@
   append_keys_to_tree (dialog, _("Desktop"), desktop_key_list);
   append_keys_to_tree (dialog, _("Sound"), sounds_key_list);
   
-  if (strcmp((char *) wm_name, WM_COMMON_METACITY) == 0)
+  if (strcmp((char *) wm_name, WM_COMMON_METACITY) == 0 || strcmp((char *) wm_name, WM_COMMON_COMPIZ) == 0)
     {
       append_keys_to_tree (dialog, _("Window Management"), metacity_key_list);
     }
