Index: gnome-control-center-3.16.2/panels/info/cc-info-panel.c
===================================================================
--- gnome-control-center-3.16.2.orig/panels/info/cc-info-panel.c
+++ gnome-control-center-3.16.2/panels/info/cc-info-panel.c
@@ -1482,6 +1482,8 @@ info_panel_setup_overview (CcInfoPanel
       g_free (text);
     }
 
+  gtk_widget_hide (WID ("version_label"));
+
   glibtop_get_mem (&mem);
   text = g_format_size_full (mem.total, G_FORMAT_SIZE_IEC_UNITS);
   widget = WID ("memory_label");
Index: gnome-control-center-3.16.2/panels/info/info.ui
===================================================================
--- gnome-control-center-3.16.2.orig/panels/info/info.ui
+++ gnome-control-center-3.16.2/panels/info/info.ui
@@ -205,7 +205,7 @@
                               <object class="GtkImage" id="system_image">
                                 <property name="visible">True</property>
                                 <property name="can_focus">False</property>
-                                <property name="resource">/org/gnome/control-center/info/GnomeLogoVerticalMedium.svg</property>
+                                <property name="pixbuf">UbuntuLogo.png</property>
                               </object>
                               <packing>
                                 <property name="expand">False</property>
Index: gnome-control-center-3.16.2/panels/info/info.gresource.xml
===================================================================
--- gnome-control-center-3.16.2.orig/panels/info/info.gresource.xml
+++ gnome-control-center-3.16.2/panels/info/info.gresource.xml
@@ -3,5 +3,6 @@
   <gresource prefix="/org/gnome/control-center/info">
     <file preprocess="xml-stripblanks">info.ui</file>
     <file>GnomeLogoVerticalMedium.svg</file>
+    <file>UbuntuLogo.png</file>
   </gresource>
 </gresources>
