Index: gnome-control-center-3.2.0/panels/keyboard/gnome-keyboard-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.2.0.orig/panels/keyboard/gnome-keyboard-panel.desktop.in.in	2011-10-02 23:52:37.454580214 -0400
+++ gnome-control-center-3.2.0/panels/keyboard/gnome-keyboard-panel.desktop.in.in	2011-10-02 23:53:01.322698572 -0400
@@ -2,7 +2,7 @@
 _Name=Keyboard
 _Comment=Change keyboard settings
 Exec=gnome-control-center keyboard
-Icon=input-keyboard
+Icon=preferences-desktop-keyboard
 Terminal=false
 Type=Application
 StartupNotify=true
