Index: gnome-control-center-3.3.91/shell/gnome-control-center.c
===================================================================
--- gnome-control-center-3.3.91.orig/shell/gnome-control-center.c	2012-03-05 15:04:56.000000000 +0100
+++ gnome-control-center-3.3.91/shell/gnome-control-center.c	2012-03-06 17:52:28.087083642 +0100
@@ -49,8 +49,8 @@
 /* Use a fixed width for the shell, since resizing horizontally is more awkward
  * for the user than resizing vertically
  * Both sizes are defined in https://live.gnome.org/Design/SystemSettings/ */
-#define FIXED_WIDTH 740
-#define FIXED_HEIGHT 636
+#define FIXED_WIDTH 850
+#define FIXED_HEIGHT 650
 #define SMALL_SCREEN_FIXED_HEIGHT 400
 
 #define MIN_ICON_VIEW_HEIGHT 300
