# Description: files to translate from ubuntu patches
Index: b/po/POTFILES.in
===================================================================
--- a/po/POTFILES.in
+++ b/po/POTFILES.in
@@ -112,3 +112,17 @@
 shell/gnomecc.directory.in
 shell/gnome-control-center.desktop.in.in
 [type: gettext/glade]shell/shell.ui
+panels/sound-nua/applet-main.c
+panels/sound-nua/data/gnome-sound-nua-panel.desktop.in.in
+panels/sound-nua/gvc-applet.c
+panels/sound-nua/gvc-balance-bar.c
+panels/sound-nua/gvc-channel-bar.c
+panels/sound-nua/gvc-combo-box.c
+panels/sound-nua/gvc-mixer-control.c
+panels/sound-nua/gvc-mixer-dialog.c
+panels/sound-nua/gvc-sound-theme-chooser.c
+panels/sound-nua/gvc-speaker-test.c
+panels/sound-nua/gvc-stream-status-icon.c
+panels/sound-nua/sound-theme-file-utils.c
+shell/cc-shell-nav-bar.c
+
