--- gnome-control-center-3.3.90/po/POTFILES.in	2012-02-20 11:22:59.000000000 +0100
+++ gnome-control-center-3.3.90/po/POTFILES.in.ubuntu	2012-02-25 12:36:32.787967246 +0100
@@ -112,3 +112,18 @@
 shell/gnomecc.directory.in
 shell/gnome-control-center.desktop.in.in
 [type: gettext/glade]shell/shell.ui
+panels/sound-nua/applet-main.c
+panels/sound-nua/data/gnome-sound-nua-panel.desktop.in.in
+panels/sound-nua/gvc-applet.c
+panels/sound-nua/gvc-balance-bar.c
+panels/sound-nua/gvc-channel-bar.c
+panels/sound-nua/gvc-combo-box.c
+panels/sound-nua/gvc-mixer-control.c
+panels/sound-nua/gvc-mixer-dialog.c
+panels/sound-nua/gvc-sound-theme-chooser.c
+panels/sound-nua/gvc-speaker-test.c
+panels/sound-nua/gvc-stream-status-icon.c
+panels/sound-nua/sound-theme-file-utils.c
+panels/universal-access/gconf-property-editor.c
+shell/cc-shell-nav-bar.c
+
