# Description: files to translate from ubuntu patches
Index: gnome-control-center-3.6.3/po/POTFILES.in
===================================================================
--- gnome-control-center-3.6.3.orig/po/POTFILES.in	2013-04-10 16:06:23.226657784 -0400
+++ gnome-control-center-3.6.3/po/POTFILES.in	2013-04-10 16:06:23.222657784 -0400
@@ -78,6 +78,7 @@
 panels/region/gnome-region-panel-formats.c
 panels/region/gnome-region-panel-system.c
 [type: gettext/glade]panels/region/gnome-region-panel.ui
+panels/region/ubuntu-region-panel.desktop.in.in
 panels/screen/gnome-screen-panel.desktop.in.in
 [type: gettext/glade]panels/screen/screen.ui
 panels/sound/applet-main.c
@@ -136,4 +137,8 @@
 [type: gettext/glade]panels/region/gnome-region-panel-options-dialog.ui
 panels/region/gnome-region-panel-xkblt.c
 panels/region/gnome-region-panel-xkbot.c
+panels/sound-nua/data/gnome-sound-nua-panel.desktop.in.in
+panels/sound-nua/gvc-balance-bar.c
+panels/sound-nua/gvc-mixer-dialog.c
+shell/cc-shell-nav-bar.c
 
