# Description: files to translate from ubuntu patches
--- a/po/POTFILES.in
+++ b/po/POTFILES.in
@@ -81,6 +81,7 @@
 [type: gettext/glade]panels/region/gnome-region-panel-input-chooser.ui
 panels/region/gnome-region-panel-system.c
 [type: gettext/glade]panels/region/gnome-region-panel.ui
+panels/region/ubuntu-region-panel.desktop.in.in
 panels/screen/gnome-screen-panel.desktop.in.in
 [type: gettext/glade]panels/screen/screen.ui
 panels/sound/applet-main.c
@@ -135,3 +136,7 @@
 shell/gnomecc.directory.in
 shell/gnome-control-center.desktop.in.in
 [type: gettext/glade]shell/shell.ui
+panels/sound-nua/data/gnome-sound-nua-panel.desktop.in.in
+panels/sound-nua/gvc-balance-bar.c
+panels/sound-nua/gvc-mixer-dialog.c
+shell/cc-shell-nav-bar.c
