Index: gnome-control-center-3.4.2/panels/universal-access/cc-ua-panel.c
===================================================================
--- gnome-control-center-3.4.2.orig/panels/universal-access/cc-ua-panel.c	2012-04-10 17:01:29.000000000 +0200
+++ gnome-control-center-3.4.2/panels/universal-access/cc-ua-panel.c	2012-10-02 18:07:53.320752834 +0200
@@ -44,6 +44,7 @@
   GtkBuilder *builder;
   GSettings *wm_settings;
   GSettings *interface_settings;
+  GSettings *canonical_interface_settings;
   GSettings *kb_settings;
   GSettings *mouse_settings;
   GSettings *application_settings;
@@ -109,6 +110,12 @@
       priv->interface_settings = NULL;
     }
 
+  if (priv->canonical_interface_settings)
+    {
+      g_object_unref (priv->canonical_interface_settings);
+      priv->canonical_interface_settings = NULL;
+    }
+
   if (priv->kb_settings)
     {
       g_object_unref (priv->kb_settings);
@@ -423,11 +430,21 @@
 
   if (g_strcmp0 (theme_name, ""))
     {
+      /* disable overlay scrollbars for a11y */
+      if (g_strcmp0 (theme_name, "HighContrast") == 0 ||
+          g_strcmp0 (theme_name, "HighContrastInverse") == 0)
+        g_settings_set_string (priv->canonical_interface_settings, "scrollbar-mode", "normal");
+      else
+        g_settings_reset (priv->canonical_interface_settings, "scrollbar-mode");
+
       g_settings_set_string (priv->interface_settings, GTK_THEME_KEY, theme_name);
       g_settings_set_string (priv->interface_settings, ICON_THEME_KEY, theme_name);
     }
   else
     {
+      /* reset overlay scrollbars key */
+      g_settings_reset (priv->canonical_interface_settings, "scrollbar-mode");
+
       g_settings_reset (priv->interface_settings, GTK_THEME_KEY);
       g_settings_reset (priv->interface_settings, ICON_THEME_KEY);
     }
@@ -772,6 +789,7 @@
     }
 
   priv->interface_settings = g_settings_new ("org.gnome.desktop.interface");
+  priv->canonical_interface_settings = g_settings_new ("com.canonical.desktop.interface");
   g_signal_connect (priv->interface_settings, "changed",
                     G_CALLBACK (interface_settings_changed_cb), self);
   interface_settings_changed_cb (priv->interface_settings, "gtk-theme", self);
