Index: gnome-control-center-3.6.1/panels/universal-access/cc-ua-panel.c
===================================================================
--- gnome-control-center-3.6.1.orig/panels/universal-access/cc-ua-panel.c	2012-10-01 05:39:00.000000000 -0400
+++ gnome-control-center-3.6.1/panels/universal-access/cc-ua-panel.c	2012-11-01 05:01:09.228520016 -0400
@@ -31,6 +31,8 @@
 
 #include "zoom-options.h"
 
+#define CANONICAL_DESKTOP_INTERFACE "com.canonical.desktop.interface"
+
 #define WID(b, w) (GtkWidget *) gtk_builder_get_object (b, w)
 
 #define DPI_FACTOR_LARGE 1.25
@@ -52,6 +54,7 @@
   GtkBuilder *builder;
   GSettings *wm_settings;
   GSettings *interface_settings;
+  GSettings *canonical_interface_settings;
   GSettings *kb_settings;
   GSettings *mouse_settings;
   GSettings *application_settings;
@@ -117,6 +120,12 @@
       priv->interface_settings = NULL;
     }
 
+  if (priv->canonical_interface_settings)
+    {
+      g_object_unref (priv->canonical_interface_settings);
+      priv->canonical_interface_settings = NULL;
+    }
+
   if (priv->kb_settings)
     {
       g_object_unref (priv->kb_settings);
@@ -385,11 +394,25 @@
   hc = g_value_get_boolean (value);
   if (hc)
     {
+      /* disable overlay scrollbars for a11y */
+      if (priv->canonical_interface_settings != NULL)
+        {
+          if (g_strcmp0 (theme_name, "HighContrast") == 0 ||
+              g_strcmp0 (theme_name, "HighContrastInverse") == 0)
+            g_settings_set_string (priv->canonical_interface_settings, "scrollbar-mode", "normal");
+          else
+            g_settings_reset (priv->canonical_interface_settings, "scrollbar-mode");
+        }
+
       ret = g_variant_new_string (HIGH_CONTRAST_THEME);
       g_settings_set_string (settings, KEY_ICON_THEME, HIGH_CONTRAST_THEME);
     }
   else
     {
+      /* reset overlay scrollbars key */
+      if (priv->canonical_interface_settings != NULL)
+        g_settings_reset (priv->canonical_interface_settings, "scrollbar-mode");
+
       g_settings_reset (settings, KEY_GTK_THEME);
       g_settings_reset (settings, KEY_ICON_THEME);
     }
@@ -634,6 +657,7 @@
 static void
 cc_ua_panel_init (CcUaPanel *self)
 {
+  GSettingsSchemaSource *source;
   CcUaPanelPrivate *priv;
   GtkWidget *widget;
   GError *err = NULL;
@@ -667,6 +691,10 @@
       return;
     }
 
+  source = g_settings_schema_source_get_default ();
+  if (source != NULL && g_settings_schema_source_lookup (source, CANONICAL_DESKTOP_INTERFACE, TRUE) != NULL)
+    priv->canonical_interface_settings = g_settings_new (CANONICAL_DESKTOP_INTERFACE);
+
   priv->interface_settings = g_settings_new ("org.gnome.desktop.interface");
   priv->wm_settings = g_settings_new ("org.gnome.desktop.wm.preferences");
   priv->kb_settings = g_settings_new ("org.gnome.desktop.a11y.keyboard");
