Index: gnome-control-center-3.4.2/panels/universal-access/cc-ua-panel.c
===================================================================
--- gnome-control-center-3.4.2.orig/panels/universal-access/cc-ua-panel.c	2012-04-11 03:01:29.000000000 +1200
+++ gnome-control-center-3.4.2/panels/universal-access/cc-ua-panel.c	2012-05-22 12:13:26.554181099 +1200
@@ -423,11 +423,21 @@
 
   if (g_strcmp0 (theme_name, ""))
     {
+      /* disable overlay scrollbars for a11y */
+      if (g_strcmp0 (theme_name, "HighContrast") == 0 ||
+          g_strcmp0 (theme_name, "HighContrastInverse") == 0)
+        g_settings_set_boolean (priv->interface_settings, "ubuntu-overlay-scrollbars", FALSE);
+      else
+        g_settings_reset (priv->interface_settings, "ubuntu-overlay-scrollbars");
+
       g_settings_set_string (priv->interface_settings, GTK_THEME_KEY, theme_name);
       g_settings_set_string (priv->interface_settings, ICON_THEME_KEY, theme_name);
     }
   else
     {
+      /* reset overlay scrollbars key */
+      g_settings_reset (priv->interface_settings, "ubuntu-overlay-scrollbars");
+
       g_settings_reset (priv->interface_settings, GTK_THEME_KEY);
       g_settings_reset (priv->interface_settings, ICON_THEME_KEY);
     }
