Index: gnome-control-center-3.16.2/panels/keyboard/01-launchers.xml.in
===================================================================
--- gnome-control-center-3.16.2.orig/panels/keyboard/01-launchers.xml.in
+++ gnome-control-center-3.16.2/panels/keyboard/01-launchers.xml.in
@@ -9,6 +9,8 @@
 
 	<KeyListEntry name="email" _description="Launch email client"/>
 
+	<KeyListEntry name="terminal" _description="Launch terminal"/>
+
 	<KeyListEntry name="www" _description="Launch web browser"/>
 
 	<KeyListEntry name="home" _description="Home folder"/>
