diff -Nur -x '*.orig' -x '*~' gnome-control-center-2.27.3/capplets/accessibility/at-properties/main.c gnome-control-center-2.27.3.new/capplets/accessibility/at-properties/main.c
--- gnome-control-center-2.27.3/capplets/accessibility/at-properties/main.c	2009-07-13 19:50:05.780646620 +0100
+++ gnome-control-center-2.27.3.new/capplets/accessibility/at-properties/main.c	2009-07-13 19:50:07.100648131 +0100
@@ -91,7 +91,7 @@
 static void
 cb_login_preferences (GtkDialog *dialog, gint response_id)
 {
-	g_spawn_command_line_async ("gdmsetup", NULL);
+	g_spawn_command_line_async ("gksu gdmsetup", NULL);
 }
 
 /* get_session_bus(), get_sm_proxy(), and do_logout() are all
