diff -Nur -x '*.orig' -x '*~' gnome-control-center-2.23.3/capplets/accessibility/at-properties/main.c gnome-control-center-2.23.3.new/capplets/accessibility/at-properties/main.c
--- gnome-control-center-2.23.3/capplets/accessibility/at-properties/main.c	2008-06-09 18:23:26.000000000 +0200
+++ gnome-control-center-2.23.3.new/capplets/accessibility/at-properties/main.c	2008-06-09 18:23:55.000000000 +0200
@@ -78,7 +78,7 @@
 static void
 cb_login_preferences (GtkDialog *dialog, gint response_id)
 {
-	g_spawn_command_line_async ("gdmsetup", NULL);
+	g_spawn_command_line_async ("gksu gdmsetup", NULL);
 }
 
 static void
