diff -Nur -x '*.orig' -x '*~' gnome-control-center-2.27.4.1/capplets/accessibility/at-properties/main.c gnome-control-center-2.27.4.1.new/capplets/accessibility/at-properties/main.c
--- gnome-control-center-2.27.4.1/capplets/accessibility/at-properties/main.c	2009-07-27 21:58:16.512552173 +0100
+++ gnome-control-center-2.27.4.1.new/capplets/accessibility/at-properties/main.c	2009-07-27 21:58:19.300549689 +0100
@@ -87,7 +87,7 @@
 static void
 cb_login_preferences (GtkDialog *dialog, gint response_id)
 {
-	g_spawn_command_line_async ("gdmsetup", NULL);
+	g_spawn_command_line_async ("gksu gdmsetup", NULL);
 }
 
 /* get_session_bus(), get_sm_proxy(), and do_logout() are all
