diff -Nur -x '*.orig' -x '*~' gnome-control-center-2.26.0/libslab/search-entry.h gnome-control-center-2.26.0.new/libslab/search-entry.h
--- gnome-control-center-2.26.0/libslab/search-entry.h	2009-07-14 13:30:35.759913598 +0200
+++ gnome-control-center-2.26.0.new/libslab/search-entry.h	2009-07-14 13:31:05.418131993 +0200
@@ -21,7 +21,7 @@
 #ifndef __NLD_SEARCH_ENTRY_H__
 #define __NLD_SEARCH_ENTRY_H__
 
-#include <gtk/gtkentry.h>
+#include <gtk/gtk.h>
 
 G_BEGIN_DECLS
 
