diff -up gnome-control-center-2.28.0/capplets/about-me/gnome-about-me.c.about-me-delete gnome-control-center-2.28.0/capplets/about-me/gnome-about-me.c
--- gnome-control-center-2.28.0/capplets/about-me/gnome-about-me.c.about-me-delete	2009-08-17 04:40:16.000000000 -0400
+++ gnome-control-center-2.28.0/capplets/about-me/gnome-about-me.c	2009-10-06 00:18:40.622398272 -0400
@@ -72,6 +71,7 @@ typedef struct {
 	gchar 		*username;
 
 	guint	      	 commit_timeout_id;
+	gboolean	 destroyed;
 } GnomeAboutMe;
 
 static GnomeAboutMe *me = NULL;
@@ -183,7 +183,8 @@ about_me_destroy (GnomeAboutMe *me)
 	g_free (me->person);
 	g_free (me->login);
 	g_free (me->username);
-	g_free (me);
+
+	me->destroyed = TRUE;
 }
 
 static void
@@ -275,6 +276,9 @@ about_me_focus_out (GtkWidget *widget, G
 	const gchar *wid;
 	gint i;
 
+	if (me->destroyed)
+		return FALSE;
+
 	wid = gtk_widget_get_name (widget);
 
 	if (wid == NULL)

