Index: gnome-control-center-3.3.91/panels/info/info.ui
===================================================================
--- gnome-control-center-3.3.91.orig/panels/info/info.ui	2012-03-06 17:52:14.787082997 +0100
+++ gnome-control-center-3.3.91/panels/info/info.ui	2012-03-06 17:52:14.943083005 +0100
@@ -1191,7 +1191,7 @@
                             </child>
                             <child>
                               <object class="GtkLabel" id="fallback-label">
-                                <property name="visible">True</property>
+                                <property name="visible">False</property>
                                 <property name="can_focus">False</property>
                                 <property name="xalign">1</property>
                                 <property name="label" translatable="yes" comments="Hardware is not able to run GNOME 3's shell, so we might want to force running the 'Fallback' experience.">Forced _Fallback Mode</property>
@@ -1240,7 +1240,7 @@
                             </child>
                             <child>
                               <object class="GtkHBox" id="graphics_fallback_switch_box">
-                                <property name="visible">True</property>
+                                <property name="visible">False</property>
                                 <property name="can_focus">False</property>
                                 <property name="halign">start</property>
                                 <property name="valign">center</property>
