Index: gnome-control-center-3.1.92/panels/info/info.ui
===================================================================
--- gnome-control-center-3.1.92.orig/panels/info/info.ui	2011-09-23 18:03:09.000000000 +0200
+++ gnome-control-center-3.1.92/panels/info/info.ui	2011-09-23 18:04:30.684825851 +0200
@@ -661,7 +661,7 @@
                             </child>
                             <child>
                               <object class="GtkLabel" id="fallback-label">
-                                <property name="visible">True</property>
+                                <property name="visible">False</property>
                                 <property name="can_focus">False</property>
                                 <property name="xalign">1</property>
                                 <property name="label" translatable="yes" comments="Hardware is not able to run GNOME 3's shell, so we might want to force running the 'Fallback' experience.">Forced _Fallback Mode</property>
@@ -710,7 +710,7 @@
                             </child>
                             <child>
                               <object class="GtkHBox" id="graphics_fallback_switch_box">
-                                <property name="visible">True</property>
+                                <property name="visible">False</property>
                                 <property name="can_focus">False</property>
                                 <property name="halign">start</property>
                                 <property name="valign">center</property>
