Index: gnome-control-center-3.0.2/panels/info/info.ui
===================================================================
--- gnome-control-center-3.0.2.orig/panels/info/info.ui	2011-09-04 14:24:36.525033824 +0200
+++ gnome-control-center-3.0.2/panels/info/info.ui	2011-09-04 14:24:39.997050810 +0200
@@ -656,7 +656,7 @@
                             </child>
                             <child>
                               <object class="GtkLabel" id="label16">
-                                <property name="visible">True</property>
+                                <property name="visible">False</property>
                                 <property name="can_focus">False</property>
                                 <property name="xalign">1</property>
                                 <property name="label" translatable="yes" comments="Hardware is not able to run GNOME 3's shell, so we might want to force running the 'Fallback' experience.">Forced Fallback Mode</property>
@@ -704,7 +704,7 @@
                             </child>
                             <child>
                               <object class="GtkHBox" id="graphics_fallback_switch_box">
-                                <property name="visible">True</property>
+                                <property name="visible">False</property>
                                 <property name="can_focus">False</property>
                                 <property name="halign">start</property>
                                 <property name="valign">center</property>
