diff -Nur control-center-2.18.1/help/Makefile.in control-center-2.18.1.new/help/Makefile.in
--- control-center-2.18.1/help/Makefile.in	2007-04-10 13:56:16.000000000 +0200
+++ control-center-2.18.1.new/help/Makefile.in	2007-04-10 22:13:17.000000000 +0200
@@ -138,6 +138,8 @@
 GCONF_SCHEMA_FILE_DIR = @GCONF_SCHEMA_FILE_DIR@
 GDK_PIXBUF_CSOURCE = @GDK_PIXBUF_CSOURCE@
 GDK_PIXBUF_CSOURCE_32 = @GDK_PIXBUF_CSOURCE_32@
+GDU_MODULE_VERSION_CHECK_CFLAGS = @GDU_MODULE_VERSION_CHECK_CFLAGS@
+GDU_MODULE_VERSION_CHECK_LIBS = @GDU_MODULE_VERSION_CHECK_LIBS@
 GETTEXT_PACKAGE = @GETTEXT_PACKAGE@
 GLIB_GENMARSHAL = @GLIB_GENMARSHAL@
 GMOFILES = @GMOFILES@
@@ -322,7 +324,6 @@
 	$(filter $(LINGUAS),$(DOC_LINGUAS)),					\
 	$(DOC_LINGUAS))
 
-_DOC_ABS_SRCDIR = @abs_srcdir@
 
 ################################################################################
 db2omf_args = \
@@ -334,11 +335,11 @@
 	--stringparam db2omf.lang $(notdir $(patsubst %/$(notdir $(2)),%,$(2)))	\
 	--stringparam db2omf.omf_dir "$(OMF_DIR)"				\
 	--stringparam db2omf.help_dir "$(HELP_DIR)"				\
-	--stringparam db2omf.omf_in "$(_DOC_OMF_IN)"				\
+	--stringparam db2omf.omf_in "`pwd`/$(_DOC_OMF_IN)"			\
 	--stringparam db2omf.scrollkeeper_cl "$(_skcontentslist)"		\
 	$(_db2omf) $(2)
 
-_DOC_OMF_IN = $(if $(DOC_MODULE),$(wildcard $(_DOC_ABS_SRCDIR)/$(DOC_MODULE).omf.in))
+_DOC_OMF_IN = $(if $(DOC_MODULE),$(wildcard $(srcdir)/$(DOC_MODULE).omf.in))
 _DOC_OMF_DB = $(if $(_DOC_OMF_IN),						\
 	$(foreach lc,C $(_DOC_REAL_LINGUAS),$(DOC_MODULE)-$(lc).omf))
 
@@ -629,17 +630,13 @@
 	done;
 	cp $@.tmp $@ && rm -f $@.tmp
 
-dist-check-gdu:
-@HAVE_GNOME_DOC_UTILS_FALSE@	@echo "*** GNOME Doc Utils must be installed in order to make dist"
-@HAVE_GNOME_DOC_UTILS_FALSE@	@false
-
 .PHONY: dist-doc-header
 dist-doc-header: $(DOC_H_FILE)
 	@if test -f "$(DOC_H_FILE)"; then d=; else d="$(srcdir)/"; fi; \
 	echo "$(INSTALL_DATA) $${d}$(DOC_H_FILE) $(distdir)/$(DOC_H_FILE)"; \
 	$(INSTALL_DATA) "$${d}$(DOC_H_FILE)" "$(distdir)/$(DOC_H_FILE)";
 
-doc-dist-hook: dist-check-gdu $(if $(DOC_H_FILE),dist-doc-header)
+doc-dist-hook: $(if $(DOC_H_FILE),dist-doc-header)
 
 .PHONY: clean-doc-header
 clean-local: $(_clean_doc_header)
@@ -675,9 +672,9 @@
 _credits ?= `$(PKG_CONFIG) --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/credits.xsl
 _ids ?= `$(PKG_CONFIG) --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/ids.xsl
 
-@ENABLE_SK_TRUE@_skpkgdatadir ?= `scrollkeeper-config --pkgdatadir`
-@ENABLE_SK_TRUE@_sklocalstatedir ?= `scrollkeeper-config --pkglocalstatedir`
-@ENABLE_SK_TRUE@_skcontentslist ?= $(_skpkgdatadir)/Templates/C/scrollkeeper_cl.xml
+_skpkgdatadir ?= `scrollkeeper-config --pkgdatadir`
+_sklocalstatedir ?= `scrollkeeper-config --pkglocalstatedir`
+_skcontentslist ?= $(_skpkgdatadir)/Templates/C/scrollkeeper_cl.xml
 
 $(_DOC_OMF_DB) : $(_DOC_OMF_IN)
 $(_DOC_OMF_DB) : $(DOC_MODULE)-%.omf : %/$(DOC_MODULE).xml
@@ -712,7 +709,11 @@
 	fi;
 	@docs=; \
 	list='$(_DOC_C_DOCS_NOENT)'; for doc in $$list; do \
-	  docs="$$docs $(_DOC_ABS_SRCDIR)/$$doc"; \
+	  if test -f $$doc; then \
+	    docs="$$docs ../$$doc"; \
+	  else \
+	    docs="$$docs ../$(srcdir)/$$doc"; \
+	  fi; \
 	done; \
 	if ! test -f $@; then \
 	  echo "(cd $(dir $@) && \
@@ -733,7 +734,8 @@
 $(_DOC_LC_DOCS) : $(_DOC_POFILES)
 $(_DOC_LC_DOCS) : $(_DOC_C_DOCS)
 	if ! test -d $(dir $@); then mkdir $(dir $@); fi
-	if [ -f "C/$(notdir $@)" ]; then d="../"; else d="$(_DOC_ABS_SRCDIR)/"; fi; \
+	case "$(srcdir)" in /*) sd="$(srcdir)";; *) sd="../$(srcdir)";;	esac; \
+	if [ -f "C/$(notdir $@)" ]; then d="../"; else d="$$sd/"; fi; \
 	(cd $(dir $@) && \
 	  $(_xml2po) -e -p \
 	    "$${d}$(dir $@)$(patsubst %/$(notdir $@),%,$@).po" \
