Index: gnome-control-center-3.3.5/configure.ac
===================================================================
--- gnome-control-center-3.3.5.orig/configure.ac	2012-02-12 11:03:42.198272412 -0500
+++ gnome-control-center-3.3.5/configure.ac	2012-02-12 11:04:07.174396268 -0500
@@ -191,7 +191,7 @@
             with_cheese=auto)
 
 if test x"$with_cheese" != x"no" ; then
-   PKG_CHECK_MODULES(CHEESE, gstreamer-0.10 cheese-gtk >= 3.3.5, [have_cheese=yes], [have_cheese=no])
+   PKG_CHECK_MODULES(CHEESE, gstreamer-0.10 cheese >= 3.3.5 cheese-gtk >= 3.3.5, [have_cheese=yes], [have_cheese=no])
    if test x${have_cheese} = xyes; then
      AC_DEFINE(HAVE_CHEESE, 1, [Define to 1 to enable cheese webcam support])
    fi
