--- trunk/capplets/sound/sound-properties-capplet.c	2007/03/18 13:53:58	7387
+++ trunk/capplets/sound/sound-properties-capplet.c	2007/03/18 14:27:53	7389
@@ -1019,7 +1019,15 @@
 	bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
 	textdomain (GETTEXT_PACKAGE);
 
- 	context = g_option_context_new (_("- GNOME Sound Preferences"));
+        /* Since gstreamer and gnome-vfs require threads, we
+         * have to initialise threads here as the first call to glib.
+         */
+        g_thread_init (NULL);
+
+ 	context = g_option_context_new (N_("- GNOME Sound Preferences"));
+#if GLIB_CHECK_VERSION (2, 12, 0)
+        g_option_context_set_translation_domain (context, GETTEXT_PACKAGE);
+#endif
  	g_option_context_add_main_entries (context, cap_options, GETTEXT_PACKAGE);
 	g_option_context_add_group (context, gst_init_get_option_group ());
 
