From 8bbd3cba915641366fa1a342beb94d26db37388b Mon Sep 17 00:00:00 2001
From: Rodrigo Moya <rodrigo@gnome-db.org>
Date: Tue, 10 Jan 2012 09:31:40 +0000
Subject: build: Link language-related code to fontconfig

---
diff --git a/configure.ac b/configure.ac
index be52551..1d7ac0d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -87,7 +87,7 @@
  libnotify >= $LIBNOTIFY_REQUIRED_VERSION"
 
 PKG_CHECK_MODULES(LIBGNOME_CONTROL_CENTER, $COMMON_MODULES gconf-2.0)
-PKG_CHECK_MODULES(LIBLANGUAGE, $COMMON_MODULES gnome-desktop-3.0)
+PKG_CHECK_MODULES(LIBLANGUAGE, $COMMON_MODULES gnome-desktop-3.0 fontconfig)
 PKG_CHECK_MODULES(LIBSHORTCUTS, $COMMON_MODULES x11)
 PKG_CHECK_MODULES(SHELL, $COMMON_MODULES libgnome-menu-3.0 gio-unix-2.0)
 PKG_CHECK_MODULES(BACKGROUND_PANEL, $COMMON_MODULES libxml-2.0 gnome-desktop-3.0
