diff -Nur gnome-control-center-3.6.3/panels/printers/gnome-printers-panel.desktop.in.in gnome-control-center-3.6.3.new/panels/printers/gnome-printers-panel.desktop.in.in
--- gnome-control-center-3.6.3/panels/printers/gnome-printers-panel.desktop.in.in	2013-03-19 11:48:12.950319345 +0100
+++ gnome-control-center-3.6.3.new/panels/printers/gnome-printers-panel.desktop.in.in	2013-03-19 11:47:56.902318811 +0100
@@ -8,7 +8,7 @@
 StartupNotify=true
 # The X-GNOME-Settings-Panel is necessary to show in the main shell UI
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Settings-Panel=printers
 # Translators: those are keywords for the printing control-center panel
 _Keywords=Printer;Queue;Print;Paper;Ink;Toner;
