commit 8bbd3cba915641366fa1a342beb94d26db37388b
Author: Rodrigo Moya <rodrigo@gnome-db.org>
Date:   Tue Jan 10 10:31:40 2012 +0100

    build: Link language-related code to fontconfig

Index: gnome-control-center-3.2.2/configure.ac
===================================================================
--- gnome-control-center-3.2.2.orig/configure.ac	2011-11-08 21:23:58.000000000 +0100
+++ gnome-control-center-3.2.2/configure.ac	2012-03-29 05:11:01.152879301 +0200
@@ -87,7 +87,7 @@
  libnotify >= $LIBNOTIFY_REQUIRED_VERSION"
 
 PKG_CHECK_MODULES(LIBGNOME_CONTROL_CENTER, $COMMON_MODULES gconf-2.0)
-PKG_CHECK_MODULES(LIBLANGUAGE, $COMMON_MODULES gnome-desktop-3.0)
+PKG_CHECK_MODULES(LIBLANGUAGE, $COMMON_MODULES gnome-desktop-3.0 fontconfig)
 PKG_CHECK_MODULES(LIBSHORTCUTS, $COMMON_MODULES x11)
 PKG_CHECK_MODULES(SHELL, $COMMON_MODULES libgnome-menu-3.0 gio-unix-2.0)
 PKG_CHECK_MODULES(BACKGROUND_PANEL, $COMMON_MODULES libxml-2.0 gnome-desktop-3.0
