Index: capplets/appearance/appearance-style.c
===================================================================
--- capplets/appearance/appearance-style.c	(révision 8996)
+++ capplets/appearance/appearance-style.c	(copie de travail)
@@ -113,7 +113,8 @@
                                                     &sort_iter, &iter);
     path = gtk_tree_model_get_string_from_iter (store, &sort_iter);
 
-    create_thumbnail (curr_value, thumbnail, data);
+    if (data)
+      create_thumbnail (curr_value, thumbnail, data);
   }
 
   new_value = gconf_value_new (GCONF_VALUE_STRING);
