diff -Nur -x '*.orig' -x '*~' gnome-control-center-2.27.91/capplets/accessibility/at-properties/at-enable-dialog.ui gnome-control-center-2.27.91.new/capplets/accessibility/at-properties/at-enable-dialog.ui
--- gnome-control-center-2.27.91/capplets/accessibility/at-properties/at-enable-dialog.ui	2009-08-31 20:09:39.668853804 +0100
+++ gnome-control-center-2.27.91.new/capplets/accessibility/at-properties/at-enable-dialog.ui	2009-08-31 20:20:52.737855918 +0100
@@ -229,7 +229,7 @@
                                 <child>
                                   <object class="GtkButton" id="login_button">
                                     <property name="label" translatable="yes">Accessible Lo_gin</property>
-                                    <property name="visible">True</property>
+                                    <property name="visible">False</property>
                                     <property name="can_focus">True</property>
                                     <property name="receives_default">False</property>
                                     <property name="image">login_button_img</property>
diff -Nur -x '*.orig' -x '*~' gnome-control-center-2.27.91/capplets/accessibility/at-properties/main.c gnome-control-center-2.27.91.new/capplets/accessibility/at-properties/main.c
--- gnome-control-center-2.27.91/capplets/accessibility/at-properties/main.c	2009-08-31 20:09:39.672857767 +0100
+++ gnome-control-center-2.27.91.new/capplets/accessibility/at-properties/main.c	2009-08-31 20:23:25.356855134 +0100
@@ -47,7 +47,7 @@
 						 "at_applications_image");
 		gtk_image_set_from_file (GTK_IMAGE (object),
 					 PIXMAPDIR "/at-support.png");
-
+#if 0
 		prog = g_find_program_in_path ("gdmsetup");
 		if (prog == NULL) {
 			object = gtk_builder_get_object (builder,
@@ -56,6 +56,7 @@
 		}
 
 		g_free (prog);
+#endif
 	} else {
 		g_warning ("Could not load UI: %s", error->message);
 		g_error_free (error);
@@ -309,11 +310,11 @@
 	object = gtk_builder_get_object (builder, "mouse_button");
 	g_signal_connect (object, "clicked",
 			  G_CALLBACK (cb_mouse_preferences), NULL);
-
+#if 0
 	object = gtk_builder_get_object (builder, "login_button");
 	g_signal_connect (object, "clicked",
 			  G_CALLBACK (cb_login_preferences), NULL);
-
+#endif
 	widget = GTK_WIDGET (gtk_builder_get_object (builder,
 						     "at_properties_dialog"));
 	capplet_set_icon (widget, "preferences-desktop-accessibility");
