diff -Nur -x '*.orig' -x '*~' gnome-control-center-2.27.4.1/capplets/appearance/appearance-font.c gnome-control-center-2.27.4.1.new/capplets/appearance/appearance-font.c
--- gnome-control-center-2.27.4.1/capplets/appearance/appearance-font.c	2009-07-28 00:24:23.540815819 +0100
+++ gnome-control-center-2.27.4.1.new/capplets/appearance/appearance-font.c	2009-07-28 00:24:51.888817942 +0100
@@ -956,7 +956,7 @@
                    ANTIALIAS_GRAYSCALE, HINT_FULL);
   setup_font_pair (appearance_capplet_get_widget (data, "subpixel_radio"),
                    appearance_capplet_get_widget (data, "subpixel_sample"),
-                   ANTIALIAS_RGBA, HINT_FULL);
+                   ANTIALIAS_RGBA, HINT_SLIGHT);
 
   font_render_load (data->client);
 
