diff -Nur -x '*.orig' -x '*~' gnome-control-center-2.22.1/capplets/appearance/appearance-font.c gnome-control-center-2.22.1.new/capplets/appearance/appearance-font.c
--- gnome-control-center-2.22.1/capplets/appearance/appearance-font.c	2008-03-27 13:12:29.000000000 +0000
+++ gnome-control-center-2.22.1.new/capplets/appearance/appearance-font.c	2008-06-04 16:53:06.000000000 +0100
@@ -956,7 +956,7 @@
 		   ANTIALIAS_GRAYSCALE, HINT_FULL);
   setup_font_pair (glade_xml_get_widget (data->xml, "subpixel_radio"),
 		   glade_xml_get_widget (data->xml, "subpixel_sample"),
-		   ANTIALIAS_RGBA, HINT_FULL);
+		   ANTIALIAS_RGBA, HINT_SLIGHT);
 
   font_render_load (data->client);
 
