Index: gnome-control-center-2.28.1/capplets/appearance/appearance-font.c
===================================================================
--- gnome-control-center-2.28.1.orig/capplets/appearance/appearance-font.c	2009-12-10 13:11:37.101719614 +1100
+++ gnome-control-center-2.28.1/capplets/appearance/appearance-font.c	2009-12-10 13:12:14.461719769 +1100
@@ -958,7 +958,7 @@
                    ANTIALIAS_GRAYSCALE, HINT_FULL);
   setup_font_pair (appearance_capplet_get_widget (data, "subpixel_radio"),
                    appearance_capplet_get_widget (data, "subpixel_sample"),
-                   ANTIALIAS_RGBA, HINT_FULL);
+                   ANTIALIAS_RGBA, HINT_SLIGHT);
 
   font_render_load (data->client);
 
