diff -Nur -x '*.orig' -x '*~' gnome-control-center-2.25.92/po/Makefile.in.in gnome-control-center-2.25.92.new/po/Makefile.in.in
--- gnome-control-center-2.25.92/po/Makefile.in.in	2009-03-06 10:56:25.000000000 +0100
+++ gnome-control-center-2.25.92.new/po/Makefile.in.in	2009-03-06 10:57:26.000000000 +0100
@@ -21,7 +21,7 @@
 PACKAGE = @PACKAGE@
 VERSION = @VERSION@
 
-SHELL = /bin/sh
+SHELL = @SHELL@
 
 srcdir = @srcdir@
 top_srcdir = @top_srcdir@
