diff -Nur gnome-control-center-3.2.2/panels/sound/gvc-sound-theme-chooser.c gnome-control-center-3.2.2.ubuntu/panels/sound/gvc-sound-theme-chooser.c
--- gnome-control-center-3.2.2/panels/sound/gvc-sound-theme-chooser.c	2011-08-25 17:09:31.000000000 +0200
+++ gnome-control-center-3.2.2.ubuntu/panels/sound/gvc-sound-theme-chooser.c	2012-02-14 23:03:27.071026620 +0100
@@ -68,7 +68,7 @@
 #define DEFAULT_ALERT_ID        "__default"
 #define CUSTOM_THEME_NAME       "__custom"
 #define NO_SOUNDS_THEME_NAME    "__no_sounds"
-#define DEFAULT_THEME           "freedesktop"
+#define DEFAULT_THEME           "ubuntu"
 
 enum {
         THEME_DISPLAY_COL,
