Index: gnome-control-center-3.3.5/panels/sound/gvc-sound-theme-chooser.c
===================================================================
--- gnome-control-center-3.3.5.orig/panels/sound/gvc-sound-theme-chooser.c	2012-01-30 15:13:20.000000000 +0100
+++ gnome-control-center-3.3.5/panels/sound/gvc-sound-theme-chooser.c	2012-02-15 23:16:10.391706858 +0100
@@ -67,7 +67,7 @@
 #define DEFAULT_ALERT_ID        "__default"
 #define CUSTOM_THEME_NAME       "__custom"
 #define NO_SOUNDS_THEME_NAME    "__no_sounds"
-#define DEFAULT_THEME           "freedesktop"
+#define DEFAULT_THEME           "ubuntu"
 
 enum {
         THEME_DISPLAY_COL,
