Index: gnome-control-center-2.32.0/capplets/keyboard/Makefile.am
===================================================================
--- gnome-control-center-2.32.0.orig/capplets/keyboard/Makefile.am	2010-11-10 16:43:24.228649002 +0100
+++ gnome-control-center-2.32.0/capplets/keyboard/Makefile.am	2010-11-10 16:44:20.368649000 +0100
@@ -15,7 +15,7 @@
 	gnome-keyboard-properties-xkbpv.c \
 	gnome-keyboard-properties-xkb.h
 
-gnome_keyboard_properties_LDADD = $(GNOMECC_CAPPLETS_LIBS) $(LIBGNOMEKBDUI_LIBS)
+gnome_keyboard_properties_LDADD = $(GNOMECC_CAPPLETS_LIBS) $(LIBGNOMEKBDUI_LIBS) $(LIBGNOMEKBD_LIBS)
 
 @INTLTOOL_DESKTOP_RULE@
 
@@ -33,6 +33,7 @@
 INCLUDES = \
 	$(GNOMECC_CAPPLETS_CFLAGS) \
 	$(LIBGNOMEKBDUI_CFLAGS) \
+	$(LIBGNOMEKBD_CFLAGS) \
 	-DGNOMELOCALEDIR="\"$(datadir)/locale\"" \
 	-DGNOMECC_DATA_DIR="\"$(pkgdatadir)\"" \
 	-DGNOMECC_UI_DIR="\"$(uidir)\""
