Index: gnome-control-center-3.3.90/panels/display/cc-display-panel.c
===================================================================
--- gnome-control-center-3.3.90.orig/panels/display/cc-display-panel.c	2012-02-20 05:22:59.000000000 -0500
+++ gnome-control-center-3.3.90/panels/display/cc-display-panel.c	2012-02-25 09:35:48.099889088 -0500
@@ -117,6 +117,7 @@
 static GObject *cc_display_panel_constructor (GType                  gtype,
 					      guint                  n_properties,
 					      GObjectConstructParam *properties);
+static void on_screen_changed (GnomeRRScreen *scr, gpointer data);
 
 static void
 cc_display_panel_get_property (GObject    *object,
@@ -159,6 +160,7 @@
 
   self = CC_DISPLAY_PANEL (object);
 
+  g_signal_handlers_disconnect_by_func (self->priv->screen, on_screen_changed, self);
   g_object_unref (self->priv->screen);
   g_object_unref (self->priv->builder);
 
