Index: gnome-control-center-3.4.0/panels/user-accounts/data/user-accounts-dialog.ui
===================================================================
--- gnome-control-center-3.4.0.orig/panels/user-accounts/data/user-accounts-dialog.ui	2012-02-14 05:13:17.000000000 -0500
+++ gnome-control-center-3.4.0/panels/user-accounts/data/user-accounts-dialog.ui	2012-04-09 20:29:33.637981288 -0400
@@ -126,6 +126,7 @@
                 <child>
                   <object class="GtkVBox" id="main-user-vbox">
                     <property name="visible">True</property>
+                    <property name="sensitive">False</property>
                     <property name="orientation">vertical</property>
                     <property name="spacing">6</property>
                     <child>
Index: gnome-control-center-3.4.0/panels/user-accounts/um-user-panel.c
===================================================================
--- gnome-control-center-3.4.0.orig/panels/user-accounts/um-user-panel.c	2012-04-09 20:19:34.289962522 -0400
+++ gnome-control-center-3.4.0/panels/user-accounts/um-user-panel.c	2012-04-09 20:31:01.313984033 -0400
@@ -636,7 +636,10 @@
                 gtk_tree_model_get (model, &iter, USER_COL, &user, -1);
                 show_user (user, d);
                 on_permission_changed (d->permission, NULL, d);
+                gtk_widget_set_sensitive (get_widget (d, "main-user-vbox"), TRUE);
                 g_object_unref (user);
+        } else {                
+                gtk_widget_set_sensitive (get_widget (d, "main-user-vbox"), FALSE);
         }
 }
 
