diff -Nur gnome-control-center-3.4.0.gnome/panels/sound/data/gnome-sound-applet.desktop.in gnome-control-center-3.4.0/panels/sound/data/gnome-sound-applet.desktop.in
--- gnome-control-center-3.4.0.gnome/panels/sound/data/gnome-sound-applet.desktop.in	2012-03-30 11:23:25.267561164 +0200
+++ gnome-control-center-3.4.0/panels/sound/data/gnome-sound-applet.desktop.in	2012-03-30 11:24:03.743563026 +0200
@@ -14,4 +14,4 @@
 #X-GNOME-Autostart-Phase=Panel
 X-GNOME-Autostart-Notify=true
 AutostartCondition=GNOME3 if-session gnome-fallback
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=;
