Index: gnome-control-center-3.4.2/configure.ac
===================================================================
--- gnome-control-center-3.4.2.orig/configure.ac	2012-08-23 15:50:04.972386899 -0400
+++ gnome-control-center-3.4.2/configure.ac	2012-08-23 15:50:05.032386901 -0400
@@ -200,6 +200,7 @@
    PKG_CHECK_MODULES(CHEESE, gstreamer-0.10 cheese-gtk >= 3.3.5 cheese, [have_cheese=yes], [have_cheese=no])
    if test x${have_cheese} = xyes; then
      AC_DEFINE(HAVE_CHEESE, 1, [Define to 1 to enable cheese webcam support])
+     AC_DEFINE(HAVE_CLUTTER, 1, [Define to 1 to enable clutter support])
    fi
    if test x${with_cheese} = xyes && test x${have_cheese} = xno; then
      AC_MSG_ERROR([Cheese configured but not found])
@@ -209,6 +210,24 @@
 fi
 AM_CONDITIONAL(BUILD_CHEESE, test x${have_cheese} = xyes)
 
+# Optional dependency
+AC_ARG_WITH([clutter],
+            AS_HELP_STRING([--with-clutter], [enable clutter support]),,
+            with_clutter=auto)
+
+if test x"$with_clutter" != x"no" ; then
+  PKG_CHECK_MODULES(CLUTTER, clutter-gtk-1.0 clutter-1.0 >= 1.11.10, [have_clutter=yes], [have_clutter=no])
+  if test x${have_clutter} = xyes; then
+    AC_DEFINE(HAVE_CLUTTER, 1, [Define to 1 to enable clutter support])
+  fi
+  if test x${with_clutter} = xyes && test x${have_clutter} = xno; then
+    AC_MSG_ERROR([Clutter configured but not found])
+  fi
+else
+   have_clutter=no
+fi
+AM_CONDITIONAL(BUILD_CLUTTER, test x${have_clutter} = xyes)
+
 # This is a hard-dependency for the region and user-accounts panels
 PKG_CHECK_MODULES(ISOCODES, iso-codes)
 
@@ -451,6 +470,11 @@
 else
 	AC_MSG_NOTICE([   Users panel webcam support disabled])
 fi
+if test "x$have_clutter" = "xyes"; then
+	AC_MSG_NOTICE([** Clutter)])
+else
+	AC_MSG_NOTICE([   Clutter support disabled])
+fi
 if test "x$with_libsocialweb" = "xyes"; then
 	AC_MSG_NOTICE([** libsocialweb (Background panel Flickr support)])
 else
Index: gnome-control-center-3.4.2/shell/gnome-control-center.c
===================================================================
--- gnome-control-center-3.4.2.orig/shell/gnome-control-center.c	2012-08-23 15:50:04.864386895 -0400
+++ gnome-control-center-3.4.2/shell/gnome-control-center.c	2012-08-23 16:00:12.852406898 -0400
@@ -22,12 +22,16 @@
 
 #include "gnome-control-center.h"
 
+#include "config.h"
 #include <glib/gi18n.h>
 #include <gio/gio.h>
 #include <gio/gdesktopappinfo.h>
 #include <gtk/gtk.h>
 #include <gdk/gdkkeysyms.h>
 #include <string.h>
+#ifdef HAVE_CLUTTER
+#include <clutter-gtk/clutter-gtk.h>
+#endif /* HAVE_CLUTTER */
 #define GMENU_I_KNOW_THIS_IS_UNSTABLE
 #include <gmenu-tree.h>
 
@@ -1181,6 +1185,14 @@
 
   priv = self->priv = CONTROL_CENTER_PRIVATE (self);
 
+#ifdef HAVE_CLUTTER
+  if (gtk_clutter_init (NULL, NULL) != CLUTTER_INIT_SUCCESS)
+    {
+      g_critical ("Clutter-GTK init failed");
+      return;
+    }
+#endif /* HAVE_CLUTTER */
+
   priv->small_screen = get_monitor_height() <= FIXED_HEIGHT;
 
   /* load the user interface */
Index: gnome-control-center-3.4.2/shell/Makefile.am
===================================================================
--- gnome-control-center-3.4.2.orig/shell/Makefile.am	2012-08-23 15:50:04.864386895 -0400
+++ gnome-control-center-3.4.2/shell/Makefile.am	2012-08-23 15:50:05.032386901 -0400
@@ -1,6 +1,7 @@
 INCLUDES =					\
 	-I$(top_srcdir)				\
-	$(SHELL_CFLAGS)
+	$(SHELL_CFLAGS)				\
+	$(CLUTTER_CFLAGS)
 
 bin_PROGRAMS = gnome-control-center
 
@@ -35,7 +36,8 @@
 
 gnome_control_center_LDADD =			\
    libgnome-control-center.la \
-	$(SHELL_LIBS)
+	$(SHELL_LIBS)				\
+	$(CLUTTER_LIBS)
 
 gnome_control_center_LDFLAGS = -export-dynamic
 
