Index: gnome-control-center/panels/info-overview/cc-info-overview-panel.ui
===================================================================
--- gnome-control-center.orig/panels/info-overview/cc-info-overview-panel.ui
+++ gnome-control-center/panels/info-overview/cc-info-overview-panel.ui
@@ -30,7 +30,8 @@
                   <object class="GtkImage">
                     <property name="visible">True</property>
                     <property name="can_focus">False</property>
-                    <property name="resource">/org/gnome/control-center/info-overview/GnomeLogoVerticalMedium.svg</property>
+                    <property name="pixel_size">160</property>
+                    <property name="icon_name">distributor-logo</property>
                   </object>
                   <packing>
                     <property name="expand">False</property>
