---
 shell/Makefile.am |    7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

--- a/shell/Makefile.am
+++ b/shell/Makefile.am
@@ -22,9 +22,8 @@ libshell_la_SOURCES = \
 	cc-shell-model.c			\
 	cc-shell-model.h
 
-bin_PROGRAMS = gnome-control-center
-
-noinst_PROGRAMS = gnome-control-center-alt
+bin_PROGRAMS = gnome-control-center \
+	gnome-control-center-alt
 
 common_sources =				\
 	$(BUILT_SOURCES)			\
@@ -165,7 +164,7 @@ CLEANFILES = $(BUILT_SOURCES) $(completi
 DISTCLEANFILES = gnome-control-center.desktop gnome-control-center.desktop.in
 
 TEST_PROGS += test-hostname
-noinst_PROGRAMS += test-hostname
+noinst_PROGRAMS = test-hostname
 test_hostname_SOURCES = hostname-helper.c hostname-helper.h test-hostname.c
 test_hostname_LDADD = $(PANEL_LIBS) $(INFO_PANEL_LIBS)
 
