Index: b/panels/user-accounts/um-user.c
===================================================================
--- a/panels/user-accounts/um-user.c
+++ b/panels/user-accounts/um-user.c
@@ -1061,6 +1061,8 @@
         GVariant *result;
         GError *error = NULL;
 
+        g_return_if_fail (UM_IS_USER (user));
+
         result = g_dbus_proxy_call_sync (user->proxy, "SetAutomaticLogin", g_variant_new ("(b)", enabled), G_DBUS_CALL_FLAGS_NONE, -1, NULL, &error);
         if (!result) {
                 g_warning ("SetAutomaticLogin call failed: %s", error->message);
Index: b/panels/user-accounts/um-user-panel.c
===================================================================
--- a/panels/user-accounts/um-user-panel.c
+++ b/panels/user-accounts/um-user-panel.c
@@ -535,6 +535,9 @@
 
         active = gtk_switch_get_active (GTK_SWITCH (object));
         user = get_selected_user (d);
+        if (user == NULL) {
+                return;
+        }
 
         if (active != um_user_get_automatic_login (user)) {
                 um_user_set_automatic_login (user, active);
