Index: gnome-control-center-3.6.3/panels/bluetooth/gnome-bluetooth-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.6.3.orig/panels/bluetooth/gnome-bluetooth-panel.desktop.in.in	2014-03-04 19:17:38.334473755 +0100
+++ gnome-control-center-3.6.3/panels/bluetooth/gnome-bluetooth-panel.desktop.in.in	2014-03-04 19:17:38.322473754 +0100
@@ -6,7 +6,7 @@
 Terminal=false
 Type=Application
 Categories=GTK;GNOME;Settings;X-GNOME-NetworkSettings;HardwareSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 StartupNotify=true
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-bluetooth
Index: gnome-control-center-3.6.3/panels/color/gnome-color-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.6.3.orig/panels/color/gnome-color-panel.desktop.in.in	2014-03-04 19:17:38.334473755 +0100
+++ gnome-control-center-3.6.3/panels/color/gnome-color-panel.desktop.in.in	2014-03-04 19:17:38.322473754 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;X-GNOME-Settings-Panel;HardwareSettings
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=color
Index: gnome-control-center-3.6.3/panels/display/gnome-display-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.6.3.orig/panels/display/gnome-display-panel.desktop.in.in	2014-03-04 19:17:38.334473755 +0100
+++ gnome-control-center-3.6.3/panels/display/gnome-display-panel.desktop.in.in	2014-03-04 19:17:38.322473754 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=Screen resolution
Index: gnome-control-center-3.6.3/panels/info/gnome-info-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.6.3.orig/panels/info/gnome-info-panel.desktop.in.in	2014-03-04 19:17:38.334473755 +0100
+++ gnome-control-center-3.6.3/panels/info/gnome-info-panel.desktop.in.in	2014-03-04 19:17:38.326473755 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;X-GNOME-SystemSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=info
Index: gnome-control-center-3.6.3/panels/keyboard/gnome-keyboard-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.6.3.orig/panels/keyboard/gnome-keyboard-panel.desktop.in.in	2014-03-04 19:17:38.334473755 +0100
+++ gnome-control-center-3.6.3/panels/keyboard/gnome-keyboard-panel.desktop.in.in	2014-03-04 19:17:38.326473755 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=keyboard
Index: gnome-control-center-3.6.3/panels/mouse/gnome-mouse-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.6.3.orig/panels/mouse/gnome-mouse-panel.desktop.in.in	2014-03-04 19:17:38.334473755 +0100
+++ gnome-control-center-3.6.3/panels/mouse/gnome-mouse-panel.desktop.in.in	2014-03-04 19:17:38.326473755 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=mouse
Index: gnome-control-center-3.6.3/panels/network/gnome-network-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.6.3.orig/panels/network/gnome-network-panel.desktop.in.in	2014-03-04 19:17:38.334473755 +0100
+++ gnome-control-center-3.6.3/panels/network/gnome-network-panel.desktop.in.in	2014-03-04 19:17:38.326473755 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=network
Index: gnome-control-center-3.6.3/panels/power/gnome-power-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.6.3.orig/panels/power/gnome-power-panel.desktop.in.in	2014-03-04 19:17:38.334473755 +0100
+++ gnome-control-center-3.6.3/panels/power/gnome-power-panel.desktop.in.in	2014-03-04 19:17:38.326473755 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;DesktopSettings;X-GNOME-Settings-Panel;HardwareSettings
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=power
Index: gnome-control-center-3.6.3/panels/screen/gnome-screen-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.6.3.orig/panels/screen/gnome-screen-panel.desktop.in.in	2014-03-04 19:17:38.334473755 +0100
+++ gnome-control-center-3.6.3/panels/screen/gnome-screen-panel.desktop.in.in	2014-03-04 19:17:38.326473755 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;DesktopSettings;X-GNOME-Settings-Panel;X-GNOME-PersonalSettings
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=screen
Index: gnome-control-center-3.6.3/panels/universal-access/gnome-universal-access-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.6.3.orig/panels/universal-access/gnome-universal-access-panel.desktop.in.in	2014-03-04 19:17:38.334473755 +0100
+++ gnome-control-center-3.6.3/panels/universal-access/gnome-universal-access-panel.desktop.in.in	2014-03-04 19:17:38.326473755 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;X-GNOME-SystemSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=Universal Access
Index: gnome-control-center-3.6.3/panels/user-accounts/data/gnome-user-accounts-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.6.3.orig/panels/user-accounts/data/gnome-user-accounts-panel.desktop.in.in	2014-03-04 19:17:38.334473755 +0100
+++ gnome-control-center-3.6.3/panels/user-accounts/data/gnome-user-accounts-panel.desktop.in.in	2014-03-04 19:17:38.326473755 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=System;Settings;X-GNOME-Settings-Panel;X-GNOME-SystemSettings
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=user-accounts
Index: gnome-control-center-3.6.3/panels/wacom/gnome-wacom-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.6.3.orig/panels/wacom/gnome-wacom-panel.desktop.in.in	2014-03-04 19:17:38.334473755 +0100
+++ gnome-control-center-3.6.3/panels/wacom/gnome-wacom-panel.desktop.in.in	2014-03-04 19:17:38.330473755 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=wacom
Index: gnome-control-center-3.6.3/shell/gnome-control-center.desktop.in.in
===================================================================
--- gnome-control-center-3.6.3.orig/shell/gnome-control-center.desktop.in.in	2014-03-04 19:17:38.334473755 +0100
+++ gnome-control-center-3.6.3/shell/gnome-control-center.desktop.in.in	2014-03-04 19:17:38.330473755 +0100
@@ -6,7 +6,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;System;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=shell
Index: gnome-control-center-3.6.3/panels/region/gnome-region-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.6.3.orig/panels/region/gnome-region-panel.desktop.in.in	2012-11-05 09:38:42.000000000 +0100
+++ gnome-control-center-3.6.3/panels/region/gnome-region-panel.desktop.in.in	2014-03-04 19:19:14.274476950 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;DesktopSettings;X-GNOME-Settings-Panel;X-GNOME-PersonalSettings
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=region
Index: gnome-control-center-3.6.3/panels/sound/data/gnome-sound-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.6.3.orig/panels/sound/data/gnome-sound-panel.desktop.in.in	2012-11-14 12:42:44.000000000 +0100
+++ gnome-control-center-3.6.3/panels/sound/data/gnome-sound-panel.desktop.in.in	2014-03-04 19:18:47.826476069 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=sound
