Index: gnome-control-center/panels/bluetooth/gnome-bluetooth-panel.desktop.in.in
===================================================================
--- gnome-control-center.orig/panels/bluetooth/gnome-bluetooth-panel.desktop.in.in
+++ gnome-control-center/panels/bluetooth/gnome-bluetooth-panel.desktop.in.in
@@ -7,7 +7,7 @@ Terminal=false
 Type=Application
 NoDisplay=true
 Categories=GTK;GNOME;Settings;X-GNOME-NetworkSettings;HardwareSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 StartupNotify=true
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-bluetooth
Index: gnome-control-center/panels/color/gnome-color-panel.desktop.in.in
===================================================================
--- gnome-control-center.orig/panels/color/gnome-color-panel.desktop.in.in
+++ gnome-control-center/panels/color/gnome-color-panel.desktop.in.in
@@ -8,7 +8,7 @@ Type=Application
 NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;X-GNOME-Settings-Panel;HardwareSettings;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=color
Index: gnome-control-center/panels/display/gnome-display-panel.desktop.in.in
===================================================================
--- gnome-control-center.orig/panels/display/gnome-display-panel.desktop.in.in
+++ gnome-control-center/panels/display/gnome-display-panel.desktop.in.in
@@ -8,7 +8,7 @@ Type=Application
 NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=Screen resolution
Index: gnome-control-center/panels/info/gnome-info-panel.desktop.in.in
===================================================================
--- gnome-control-center.orig/panels/info/gnome-info-panel.desktop.in.in
+++ gnome-control-center/panels/info/gnome-info-panel.desktop.in.in
@@ -8,7 +8,7 @@ Type=Application
 NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;X-GNOME-SystemSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=info
Index: gnome-control-center/panels/keyboard/gnome-keyboard-panel.desktop.in.in
===================================================================
--- gnome-control-center.orig/panels/keyboard/gnome-keyboard-panel.desktop.in.in
+++ gnome-control-center/panels/keyboard/gnome-keyboard-panel.desktop.in.in
@@ -8,7 +8,7 @@ Type=Application
 NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=keyboard
Index: gnome-control-center/panels/mouse/gnome-mouse-panel.desktop.in.in
===================================================================
--- gnome-control-center.orig/panels/mouse/gnome-mouse-panel.desktop.in.in
+++ gnome-control-center/panels/mouse/gnome-mouse-panel.desktop.in.in
@@ -8,7 +8,7 @@ Type=Application
 NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=mouse
Index: gnome-control-center/panels/network/gnome-network-panel.desktop.in.in
===================================================================
--- gnome-control-center.orig/panels/network/gnome-network-panel.desktop.in.in
+++ gnome-control-center/panels/network/gnome-network-panel.desktop.in.in
@@ -8,7 +8,7 @@ Type=Application
 NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=network
Index: gnome-control-center/panels/power/gnome-power-panel.desktop.in.in
===================================================================
--- gnome-control-center.orig/panels/power/gnome-power-panel.desktop.in.in
+++ gnome-control-center/panels/power/gnome-power-panel.desktop.in.in
@@ -8,7 +8,7 @@ Type=Application
 NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;DesktopSettings;X-GNOME-Settings-Panel;HardwareSettings;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=power
Index: gnome-control-center/panels/universal-access/gnome-universal-access-panel.desktop.in.in
===================================================================
--- gnome-control-center.orig/panels/universal-access/gnome-universal-access-panel.desktop.in.in
+++ gnome-control-center/panels/universal-access/gnome-universal-access-panel.desktop.in.in
@@ -8,7 +8,7 @@ Type=Application
 NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;X-GNOME-SystemSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=Universal Access
Index: gnome-control-center/panels/user-accounts/data/gnome-user-accounts-panel.desktop.in.in
===================================================================
--- gnome-control-center.orig/panels/user-accounts/data/gnome-user-accounts-panel.desktop.in.in
+++ gnome-control-center/panels/user-accounts/data/gnome-user-accounts-panel.desktop.in.in
@@ -8,7 +8,7 @@ Type=Application
 NoDisplay=true
 StartupNotify=true
 Categories=System;Settings;X-GNOME-Settings-Panel;X-GNOME-SystemSettings;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=user-accounts
Index: gnome-control-center/panels/wacom/gnome-wacom-panel.desktop.in.in
===================================================================
--- gnome-control-center.orig/panels/wacom/gnome-wacom-panel.desktop.in.in
+++ gnome-control-center/panels/wacom/gnome-wacom-panel.desktop.in.in
@@ -8,7 +8,7 @@ Type=Application
 NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=wacom
Index: gnome-control-center/shell/gnome-control-center.desktop.in.in
===================================================================
--- gnome-control-center.orig/shell/gnome-control-center.desktop.in.in
+++ gnome-control-center/shell/gnome-control-center.desktop.in.in
@@ -6,7 +6,7 @@ Terminal=false
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;System;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=shell
Index: gnome-control-center/panels/region/gnome-region-panel.desktop.in.in
===================================================================
--- gnome-control-center.orig/panels/region/gnome-region-panel.desktop.in.in
+++ gnome-control-center/panels/region/gnome-region-panel.desktop.in.in
@@ -8,7 +8,7 @@ Type=Application
 NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;DesktopSettings;X-GNOME-Settings-Panel;X-GNOME-PersonalSettings;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=region
Index: gnome-control-center/panels/sound/data/gnome-sound-panel.desktop.in.in
===================================================================
--- gnome-control-center.orig/panels/sound/data/gnome-sound-panel.desktop.in.in
+++ gnome-control-center/panels/sound/data/gnome-sound-panel.desktop.in.in
@@ -8,7 +8,7 @@ Type=Application
 NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=sound
