https://bugzilla.gnome.org/show_bug.cgi?id=693261
---
Index: gnome-control-center-3.6.3/panels/online-accounts/cc-online-accounts-model.c
===================================================================
--- gnome-control-center-3.6.3.orig/panels/online-accounts/cc-online-accounts-model.c	2013-06-14 11:32:21.024067503 +0200
+++ gnome-control-center-3.6.3/panels/online-accounts/cc-online-accounts-model.c	2013-06-14 11:32:21.016067502 +0200
@@ -330,8 +330,8 @@
   icon = g_icon_new_for_string (goa_account_get_provider_icon (account), &error);
   if (icon == NULL)
     {
-      goa_warning ("Error creating GIcon for account: %s (%s, %d)",
-                   error->message, g_quark_to_string (error->domain), error->code);
+      g_warning ("Error creating GIcon for account: %s (%s, %d)",
+                 error->message, g_quark_to_string (error->domain), error->code);
       g_error_free (error);
     }
 
@@ -370,7 +370,7 @@
   GtkTreeIter iter;
   if (!find_iter_for_object (model, object, &iter))
     {
-      goa_warning ("Error removing object %s - not in tree", g_dbus_object_get_object_path (G_DBUS_OBJECT (object)));
+      g_warning ("Error removing object %s - not in tree", g_dbus_object_get_object_path (G_DBUS_OBJECT (object)));
     }
   else
     {
@@ -385,7 +385,7 @@
   GtkTreeIter iter;
   if (!find_iter_for_object (model, object, &iter))
     {
-      goa_warning ("Error updating object %s - not in tree", g_dbus_object_get_object_path (G_DBUS_OBJECT (object)));
+      g_warning ("Error updating object %s - not in tree", g_dbus_object_get_object_path (G_DBUS_OBJECT (object)));
     }
   else
     {
Index: gnome-control-center-3.6.3/panels/online-accounts/cc-online-accounts-panel.c
===================================================================
--- gnome-control-center-3.6.3.orig/panels/online-accounts/cc-online-accounts-panel.c	2013-06-14 11:32:21.024067503 +0200
+++ gnome-control-center-3.6.3/panels/online-accounts/cc-online-accounts-panel.c	2013-06-14 11:32:21.020067502 +0200
@@ -147,8 +147,8 @@
                                  GNOMECC_UI_DIR "/online-accounts.ui",
                                  &error) == 0)
     {
-      goa_warning ("Error loading UI file: %s (%s, %d)",
-                   error->message, g_quark_to_string (error->domain), error->code);
+      g_warning ("Error loading UI file: %s (%s, %d)",
+                 error->message, g_quark_to_string (error->domain), error->code);
       g_error_free (error);
       goto out;
     }
@@ -189,8 +189,8 @@
   panel->client = goa_client_new_sync (NULL /* GCancellable */, &error);
   if (panel->client == NULL)
     {
-      goa_warning ("Error getting a GoaClient: %s (%s, %d)",
-                   error->message, g_quark_to_string (error->domain), error->code);
+      g_warning ("Error getting a GoaClient: %s (%s, %d)",
+                 error->message, g_quark_to_string (error->domain), error->code);
       w = GTK_WIDGET (gtk_builder_get_object (panel->builder, "goa-top-widget"));
       gtk_widget_set_sensitive (w, FALSE);
       g_error_free (error);
