From 9a67cd088493cacb5f23e4afcef151dd3b8de205 Mon Sep 17 00:00:00 2001
From: Ethan Hsieh <ethan.hsieh@canonical.com>
Date: Thu, 2 May 2019 13:28:09 +0800
Subject: [PATCH] power: correct the value of 90 minutes to 5400

Index: gnome-control-center-3.28.2/panels/power/power.ui
===================================================================
--- gnome-control-center-3.28.2.orig/panels/power/power.ui
+++ gnome-control-center-3.28.2/panels/power/power.ui
@@ -39,7 +39,7 @@
       </row>
       <row>
         <col id="0" translatable="yes">90 minutes</col>
-        <col id="1">4800</col>
+        <col id="1">5400</col>
       </row>
       <row>
         <col id="0" translatable="yes">100 minutes</col>
