Index: gnome-control-center-3.6.3/panels/region/gnome-region-panel-input.c
===================================================================
--- gnome-control-center-3.6.3.orig/panels/region/gnome-region-panel-input.c	2013-09-11 10:27:26.979533097 -0400
+++ gnome-control-center-3.6.3/panels/region/gnome-region-panel-input.c	2013-09-11 10:33:09.507518772 -0400
@@ -1104,7 +1104,7 @@
   gtk_widget_set_sensitive (show_button, index >= 0);
   gtk_widget_set_sensitive (up_button, index > 0);
   gtk_widget_set_sensitive (down_button, index >= 0 && index < n_active - 1);
-  gtk_widget_set_sensitive (settings_button, settings_sensitive);
+  gtk_widget_set_visible (settings_button, settings_sensitive);
 }
 
 static void
