From 8776700a73f68ac00bf23d846c9158f8e6c1ce35 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Thu, 26 Jul 2012 15:54:03 +0000
Subject: bluetooth: Update for killswitch API change

---
diff --git a/panels/bluetooth/cc-bluetooth-panel.c b/panels/bluetooth/cc-bluetooth-panel.c
index d6bc29e..de94931 100644
--- a/panels/bluetooth/cc-bluetooth-panel.c
+++ b/panels/bluetooth/cc-bluetooth-panel.c
@@ -356,7 +356,7 @@ power_callback (GObject          *object,
 	state = gtk_switch_get_active (GTK_SWITCH (WID ("switch_bluetooth")));
 	g_debug ("Power switched to %s", state ? "off" : "on");
 	bluetooth_killswitch_set_state (self->priv->killswitch,
-					state ? KILLSWITCH_STATE_UNBLOCKED : KILLSWITCH_STATE_SOFT_BLOCKED);
+					state ? BLUETOOTH_KILLSWITCH_STATE_UNBLOCKED : BLUETOOTH_KILLSWITCH_STATE_SOFT_BLOCKED);
 }
 
 static void
@@ -378,7 +378,7 @@ cc_bluetooth_panel_update_treeview_message (CcBluetoothPanel *self,
 static void
 cc_bluetooth_panel_update_power (CcBluetoothPanel *self)
 {
-	KillswitchState state;
+	BluetoothKillswitchState state;
 	char *path;
 	gboolean powered, sensitive;
 	GtkSwitch *button;
@@ -396,12 +396,12 @@ cc_bluetooth_panel_update_power (CcBluetoothPanel *self)
 
 	if (path == NULL &&
 	    bluetooth_killswitch_has_killswitches (self->priv->killswitch) &&
-	    state != KILLSWITCH_STATE_HARD_BLOCKED) {
+	    state != BLUETOOTH_KILLSWITCH_STATE_HARD_BLOCKED) {
 		g_debug ("Default adapter is unpowered, but should be available");
 		sensitive = TRUE;
 		cc_bluetooth_panel_update_treeview_message (self, _("Bluetooth is disabled"));
 	} else if (path == NULL &&
-		   state == KILLSWITCH_STATE_HARD_BLOCKED) {
+		   state == BLUETOOTH_KILLSWITCH_STATE_HARD_BLOCKED) {
 		g_debug ("Bluetooth is Hard blocked");
 		sensitive = FALSE;
 		cc_bluetooth_panel_update_treeview_message (self, _("Bluetooth is disabled by hardware switch"));
@@ -713,9 +713,9 @@ default_adapter_changed (BluetoothClient  *client,
 }
 
 static void
-killswitch_changed (BluetoothKillswitch *killswitch,
-		    KillswitchState      state,
-		    CcBluetoothPanel    *self)
+killswitch_changed (BluetoothKillswitch      *killswitch,
+		    BluetoothKillswitchState  state,
+		    CcBluetoothPanel         *self)
 {
 	cc_bluetooth_panel_update_state (self);
 	cc_bluetooth_panel_update_power (self);
--
cgit v0.9.0.2

