From c48536c10262c39f9a3cb710a64b09f793cd523d Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Thu, 08 Mar 2012 15:16:54 +0000
Subject: universal-access: Hide zoom options when not in shell

https://bugzilla.gnome.org/show_bug.cgi?id=671386
---
Index: gnome-control-center-3.3.91/panels/universal-access/cc-ua-panel.c
===================================================================
--- gnome-control-center-3.3.91.orig/panels/universal-access/cc-ua-panel.c	2012-03-08 17:20:26.667750516 +0100
+++ gnome-control-center-3.3.91/panels/universal-access/cc-ua-panel.c	2012-03-08 17:20:26.735750513 +0100
@@ -52,6 +52,7 @@
   GSettings *mediakeys_settings;
 
   ZoomOptions *zoom_options;
+  guint shell_watch_id;
 
   GSList *notify_list;
 };
@@ -103,6 +104,12 @@
     }
 
 
+  if (priv->shell_watch_id)
+    {
+      g_bus_unwatch_name (priv->shell_watch_id);
+      priv->shell_watch_id = 0;
+    }
+
   if (priv->builder)
     {
       g_object_unref (priv->builder);
@@ -522,6 +529,27 @@
 }
 
 static void
+shell_vanished_cb (GDBusConnection *connection,
+		   const gchar *name,
+		   CcUaPanel   *self)
+{
+  CcUaPanelPrivate *priv = self->priv;
+
+  gtk_widget_hide (WID (priv->builder, "zoom_frame"));
+}
+
+static void
+shell_appeared_cb (GDBusConnection *connection,
+		   const gchar *name,
+		   const gchar *name_owner,
+		   CcUaPanel   *self)
+{
+  CcUaPanelPrivate *priv = self->priv;
+
+  gtk_widget_show (WID (priv->builder, "zoom_frame"));
+}
+
+static void
 cc_ua_panel_init_seeing (CcUaPanel *self)
 {
   CcUaPanelPrivate *priv = self->priv;
@@ -539,6 +567,13 @@
                    WID (priv->builder, "seeing_enable_toggle_keys_checkbutton"), "active",
                    G_SETTINGS_BIND_DEFAULT);
 
+  priv->shell_watch_id = g_bus_watch_name (G_BUS_TYPE_SESSION,
+					   "org.gnome.Shell",
+					   G_BUS_NAME_WATCHER_FLAGS_NONE,
+					   (GBusNameAppearedCallback) shell_appeared_cb,
+					   (GBusNameVanishedCallback) shell_vanished_cb,
+					   self,
+					   NULL);
   g_signal_connect (WID (priv->builder, "seeing_zoom_preferences_button"),
                     "clicked",
                     G_CALLBACK (zoom_options_launch_cb), self);
Index: gnome-control-center-3.3.91/panels/universal-access/uap.ui
===================================================================
--- gnome-control-center-3.3.91.orig/panels/universal-access/uap.ui	2012-03-05 15:04:55.000000000 +0100
+++ gnome-control-center-3.3.91/panels/universal-access/uap.ui	2012-03-08 17:20:26.743750512 +0100
@@ -428,7 +428,7 @@
                   </packing>
                 </child>
                 <child>
-                  <object class="GtkFrame" id="frame4">
+                  <object class="GtkFrame" id="zoom_frame">
                     <property name="visible">True</property>
                     <property name="label_xalign">0</property>
                     <property name="shadow_type">none</property>
