From be3c71065054b7bc7c2e2325d7ebc03274075497 Mon Sep 17 00:00:00 2001
From: Rodrigo Moya <rodrigo@gnome-db.org>
Date: Mon, 30 Jan 2012 10:34:30 +0000
Subject: shell: Add some keywords to shell's desktop file

https://bugzilla.gnome.org/show_bug.cgi?id=668823
---
diff --git a/shell/gnome-control-center.desktop.in.in b/shell/gnome-control-center.desktop.in.in
index a7160bd..af63827 100644
--- a/shell/gnome-control-center.desktop.in.in
+++ b/shell/gnome-control-center.desktop.in.in
@@ -11,3 +11,4 @@ X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=shell
 X-GNOME-Bugzilla-Version=@VERSION@
+_Keywords=Preferences;Settings;
--
cgit v0.9.0.2

