Index: gnome-control-center-3.1.4/panels/network/cc-network-panel.c
===================================================================
--- gnome-control-center-3.1.4.orig/panels/network/cc-network-panel.c	2011-08-08 11:41:29.528664564 -0400
+++ gnome-control-center-3.1.4/panels/network/cc-network-panel.c	2011-08-08 11:42:01.308691043 -0400
@@ -2657,7 +2657,6 @@
         if (error != NULL) {
                 g_warning ("Getting pretty hostname failed: %s", error->message);
                 g_error_free (error);
-                return NULL;
         }
 
         str = NULL;
