Index: gnome-control-center/panels/background/cc-background-panel.c
===================================================================
--- gnome-control-center.orig/panels/background/cc-background-panel.c
+++ gnome-control-center/panels/background/cc-background-panel.c
@@ -24,6 +24,7 @@
 #include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <glib/gstdio.h>
+#include "pop_desktop_widget.h"
 
 #include <gdesktop-enums.h>
 
@@ -51,6 +52,8 @@ struct _CcBackgroundPanel
 
   GDBusConnection *connection;
 
+  PopDesktopWidget *pop;
+
   GSettings *settings;
   GSettings *lock_settings;
 
@@ -61,10 +64,28 @@ struct _CcBackgroundPanel
   CcBackgroundChooser *background_chooser;
   GtkButton *add_picture_button;
   CcBackgroundPreview *desktop_preview;
+
+  GtkWidget *stack;
+  GtkWidget *stack_switcher;
+  GtkWidget *background_container;
 };
 
 CC_PANEL_REGISTER (CcBackgroundPanel, cc_background_panel)
 
+static void visible_child_changed (GtkWidget *stack, GParamSpec *pspec, CcBackgroundPanel *self) {
+  if (gtk_stack_get_visible_child (GTK_STACK (stack)) == self->background_container) {
+    gtk_widget_show (self->add_picture_button);
+  } else {
+    gtk_widget_hide (self->add_picture_button);
+  }
+}
+
+static GtkWidget *cc_background_panel_get_title_widget (CcPanel *panel) {
+  CcBackgroundPanel *self = CC_BACKGROUND_PANEL (panel);
+
+  return self->stack_switcher;
+}
+
 static void
 update_preview (CcBackgroundPanel *panel)
 {
@@ -230,6 +251,14 @@ cc_background_panel_constructed (GObject
 
   cc_shell_embed_widget_in_header (shell, GTK_WIDGET (self->add_picture_button), GTK_POS_RIGHT);
 
+  g_signal_connect_object (
+    self->stack,
+    "notify::visible-child",
+    G_CALLBACK (visible_child_changed),
+    self,
+    0
+  );
+
   G_OBJECT_CLASS (cc_background_panel_parent_class)->constructed (object);
 }
 
@@ -241,6 +270,7 @@ cc_background_panel_dispose (GObject *ob
   g_clear_object (&panel->settings);
   g_clear_object (&panel->lock_settings);
   g_clear_object (&panel->thumb_factory);
+  g_clear_pointer (&panel->pop, pop_desktop_widget_free);
 
   G_OBJECT_CLASS (cc_background_panel_parent_class)->dispose (object);
 }
@@ -266,6 +296,7 @@ cc_background_panel_class_init (CcBackgr
   g_type_ensure (CC_TYPE_BACKGROUND_PREVIEW);
 
   panel_class->get_help_uri = cc_background_panel_get_help_uri;
+  panel_class->get_title_widget = cc_background_panel_get_title_widget;
 
   object_class->constructed = cc_background_panel_constructed;
   object_class->dispose = cc_background_panel_dispose;
@@ -276,6 +307,9 @@ cc_background_panel_class_init (CcBackgr
   gtk_widget_class_bind_template_child (widget_class, CcBackgroundPanel, add_picture_button);
   gtk_widget_class_bind_template_child (widget_class, CcBackgroundPanel, background_chooser);
   gtk_widget_class_bind_template_child (widget_class, CcBackgroundPanel, desktop_preview);
+  gtk_widget_class_bind_template_child (widget_class, CcBackgroundPanel, stack);
+  gtk_widget_class_bind_template_child (widget_class, CcBackgroundPanel, stack_switcher);
+  gtk_widget_class_bind_template_child (widget_class, CcBackgroundPanel, background_container);
 
   gtk_widget_class_bind_template_callback (widget_class, on_chooser_background_chosen_cb);
   gtk_widget_class_bind_template_callback (widget_class, on_add_picture_button_clicked_cb);
@@ -301,10 +335,12 @@ cc_background_panel_init (CcBackgroundPa
 
   panel->settings = g_settings_new (WP_PATH_ID);
   g_settings_delay (panel->settings);
- 
+
   panel->lock_settings = g_settings_new (WP_LOCK_PATH_ID);
   g_settings_delay (panel->lock_settings);
 
+  panel->pop = pop_desktop_widget_new (GTK_STACK (panel->stack));
+
   /* Load the background */
   reload_current_bg (panel);
   update_preview (panel);
Index: gnome-control-center/panels/background/cc-background-panel.ui
===================================================================
--- gnome-control-center.orig/panels/background/cc-background-panel.ui
+++ gnome-control-center/panels/background/cc-background-panel.ui
@@ -1,49 +1,64 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <interface>
   <!-- interface-requires gtk+ 3.0 -->
+  <object class="GtkStackSwitcher" id="stack_switcher">
+    <property name="visible">True</property>
+    <property name="stack">stack</property>
+  </object>
+
   <template class="CcBackgroundPanel" parent="CcPanel">
     <property name="visible">True</property>
     <property name="can_focus">False</property>
     <child>
-      <object class="GtkBox">
+      <object class="GtkStack" id="stack">
         <property name="visible">True</property>
-        <property name="can_focus">False</property>
-        <property name="orientation">vertical</property>
+        <property name="transition-type">crossfade</property>
 
+        <!-- Background page -->
         <child>
-          <object class="HdyClamp">
-            <property name="visible">1</property>
-            <property name="maximum_size">300</property>
-            <property name="tightening_threshold">200</property>
+          <object class="GtkBox" id="background_container">
+            <property name="visible">True</property>
+            <property name="can_focus">False</property>
+            <property name="orientation">vertical</property>
+
             <child>
-              <object class="GtkBox">
-                <property name="visible">True</property>
-                <property name="can_focus">False</property>
-                <property name="spacing">24</property>
-                <property name="margin-start">12</property>
-                <property name="margin-end">12</property>
-                <property name="margin-top">18</property>
-                <property name="margin-bottom">18</property>
-                <property name="hexpand">True</property>
+              <object class="HdyClamp">
+                <property name="visible">1</property>
                 <child>
-                  <object class="CcBackgroundPreview" id="desktop_preview">
+                  <object class="GtkBox">
                     <property name="visible">True</property>
                     <property name="can_focus">False</property>
-                    <property name="valign">center</property>
+                    <property name="spacing">24</property>
+                    <property name="margin-start">12</property>
+                    <property name="margin-end">12</property>
+                    <property name="margin-top">18</property>
+                    <property name="margin-bottom">18</property>
+                    <property name="hexpand">True</property>
+                    <child>
+                      <object class="CcBackgroundPreview" id="desktop_preview">
+                        <property name="visible">True</property>
+                        <property name="can_focus">False</property>
+                        <property name="valign">center</property>
+                      </object>
+                    </child>
                   </object>
                 </child>
               </object>
             </child>
-          </object>
-        </child>
 
-        <child>
-          <object class="CcBackgroundChooser" id="background_chooser">
-            <property name="visible">True</property>
-            <property name="can_focus">False</property>
-            <property name="expand">True</property>
-            <signal name="background-chosen" handler="on_chooser_background_chosen_cb" object="CcBackgroundPanel" swapped="yes" />
+            <child>
+              <object class="CcBackgroundChooser" id="background_chooser">
+                <property name="visible">True</property>
+                <property name="can_focus">False</property>
+                <property name="expand">True</property>
+                <signal name="background-chosen" handler="on_chooser_background_chosen_cb" object="CcBackgroundPanel" swapped="yes" />
+              </object>
+            </child>
           </object>
+          <packing>
+            <property name="name">background</property>
+            <property name="title" translatable="yes" comments="Change the background in Pop!_OS">Background</property>
+          </packing>
         </child>
       </object>
     </child>
Index: gnome-control-center/panels/background/gnome-background-panel.desktop.in.in
===================================================================
--- gnome-control-center.orig/panels/background/gnome-background-panel.desktop.in.in
+++ gnome-control-center/panels/background/gnome-background-panel.desktop.in.in
@@ -1,6 +1,6 @@
 [Desktop Entry]
-Name=Background
-Comment=Change your background image to a wallpaper or photo
+Name=Desktop
+Comment=Customize desktop features. Change your background image. Toggle dark and light mode.
 Exec=gnome-control-center background
 # Translators: Do NOT translate or transliterate this text (this is an icon file name)!
 Icon=preferences-desktop-wallpaper
@@ -11,4 +11,4 @@ StartupNotify=true
 Categories=GNOME;GTK;Settings;DesktopSettings;X-GNOME-Settings-Panel;X-GNOME-PersonalizationSettings;
 OnlyShowIn=GNOME;
 # Translators: Search terms to find the Background panel. Do NOT translate or localize the semicolons! The list MUST also end with a semicolon!
-Keywords=Wallpaper;Screen;Desktop;
+Keywords=Wallpaper;Screen;Desktop;Dark;Light;Theme;Night;Dock;Panel;Windows;Workspaces;
Index: gnome-control-center/panels/background/meson.build
===================================================================
--- gnome-control-center.orig/panels/background/meson.build
+++ gnome-control-center/panels/background/meson.build
@@ -94,7 +94,8 @@ deps = common_deps + [
   goa_dep,
   libxml_dep,
   dependency('cairo-gobject'),
-  dependency('grilo-0.3', version: '>= 0.3.0')
+  dependency('grilo-0.3', version: '>= 0.3.0'),
+  dependency('pop_desktop_widget')
 ]
 
 cflags += [
