Index: gnome-control-center/panels/sound/cc-sound-panel.ui
===================================================================
--- gnome-control-center.orig/panels/sound/cc-sound-panel.ui
+++ gnome-control-center/panels/sound/cc-sound-panel.ui
@@ -475,12 +475,14 @@
                     </attributes>
                   </object>
                 </child>
+                <!-- Disabled AlertChooser
                 <child>
                   <object class="CcAlertChooser">
                     <property name="visible">True</property>
                     <property name="hexpand">True</property>
                   </object>
                 </child>
+                -->
               </object>
             </child>
           </object>
