Index: gnome-control-center/panels/sound/cc-sound-panel.ui
===================================================================
--- gnome-control-center.orig/panels/sound/cc-sound-panel.ui
+++ gnome-control-center/panels/sound/cc-sound-panel.ui
@@ -464,6 +464,7 @@
                     </child>
                   </object>
                 </child>
+                <!-- Disabled AlertChooser
                 <child>
                   <object class="GtkLabel">
                     <property name="visible">True</property>
@@ -481,6 +482,7 @@
                     <property name="hexpand">True</property>
                   </object>
                 </child>
+                -->
               </object>
             </child>
           </object>
