Index: gnome-control-center/panels/sound/gvc-mixer-dialog.c
===================================================================
--- gnome-control-center.orig/panels/sound/gvc-mixer-dialog.c
+++ gnome-control-center/panels/sound/gvc-mixer-dialog.c
@@ -2034,7 +2034,7 @@ gvc_mixer_dialog_init (GvcMixerDialog *d
         dialog->bars = g_hash_table_new (NULL, NULL);
         dialog->size_group = gtk_size_group_new (GTK_SIZE_GROUP_HORIZONTAL);
 
-        if (strstr (g_getenv("XDG_CURRENT_DESKTOP"), "ubuntu") != NULL) {
+        if ((strstr (g_getenv("XDG_CURRENT_DESKTOP"), "ubuntu") != NULL) || (strstr (g_getenv("XDG_CURRENT_DESKTOP"), "pop") != NULL)) {
                 GSettingsSchema *schema;
                 schema = g_settings_schema_source_lookup (g_settings_schema_source_get_default(),
                                                           AMPLIFIED_SETTINGS, TRUE);
