Index: gnome-control-center-3.3.5/configure.ac
===================================================================
--- gnome-control-center-3.3.5.orig/configure.ac	2012-02-15 23:16:10.491706862 +0100
+++ gnome-control-center-3.3.5/configure.ac	2012-02-15 23:16:15.203707090 +0100
@@ -99,7 +99,7 @@
  gsettings-desktop-schemas >= $DESKTOP_SCHEMAS_REQUIRED_VERSION
  libnotify >= $LIBNOTIFY_REQUIRED_VERSION"
 
-PKG_CHECK_MODULES(LIBGNOME_CONTROL_CENTER, $COMMON_MODULES)
+PKG_CHECK_MODULES(LIBGNOME_CONTROL_CENTER, $COMMON_MODULES gconf-2.0)
 PKG_CHECK_MODULES(LIBLANGUAGE, $COMMON_MODULES gnome-desktop-3.0 fontconfig)
 PKG_CHECK_MODULES(LIBSHORTCUTS, $COMMON_MODULES x11)
 PKG_CHECK_MODULES(SHELL, $COMMON_MODULES libgnome-menu-3.0 gio-unix-2.0)
