Index: gnome-control-center-3.3.91/configure.ac
===================================================================
--- gnome-control-center-3.3.91.orig/configure.ac	2012-03-06 17:52:15.327083023 +0100
+++ gnome-control-center-3.3.91/configure.ac	2012-03-06 17:52:19.075083205 +0100
@@ -96,7 +96,7 @@
  gsettings-desktop-schemas >= $DESKTOP_SCHEMAS_REQUIRED_VERSION
  libnotify >= $LIBNOTIFY_REQUIRED_VERSION"
 
-PKG_CHECK_MODULES(LIBGNOME_CONTROL_CENTER, $COMMON_MODULES)
+PKG_CHECK_MODULES(LIBGNOME_CONTROL_CENTER, $COMMON_MODULES gconf-2.0)
 PKG_CHECK_MODULES(LIBLANGUAGE, $COMMON_MODULES gnome-desktop-3.0 fontconfig)
 PKG_CHECK_MODULES(LIBSHORTCUTS, $COMMON_MODULES x11)
 PKG_CHECK_MODULES(SHELL, $COMMON_MODULES libgnome-menu-3.0 gio-unix-2.0 x11)
