Index: gnome-control-center-3.3.90/configure.ac
===================================================================
--- gnome-control-center-3.3.90.orig/configure.ac	2012-02-21 13:34:43.655097695 +0100
+++ gnome-control-center-3.3.90/configure.ac	2012-02-21 13:34:46.723097812 +0100
@@ -99,7 +99,7 @@
  gsettings-desktop-schemas >= $DESKTOP_SCHEMAS_REQUIRED_VERSION
  libnotify >= $LIBNOTIFY_REQUIRED_VERSION"
 
-PKG_CHECK_MODULES(LIBGNOME_CONTROL_CENTER, $COMMON_MODULES)
+PKG_CHECK_MODULES(LIBGNOME_CONTROL_CENTER, $COMMON_MODULES gconf-2.0)
 PKG_CHECK_MODULES(LIBLANGUAGE, $COMMON_MODULES gnome-desktop-3.0 fontconfig)
 PKG_CHECK_MODULES(LIBSHORTCUTS, $COMMON_MODULES x11)
 PKG_CHECK_MODULES(SHELL, $COMMON_MODULES libgnome-menu-3.0 gio-unix-2.0 x11)
