diff --git a/panels/search/cc-search-locations-dialog.c b/panels/search/cc-search-locations-dialog.c
index 8ec47f2bc6d781d4bd8b31d5801c65f1de21314a..0cd91ed82fec2a3a7f2f5a8df02ef67c55caba20 100644
--- a/panels/search/cc-search-locations-dialog.c
+++ b/panels/search/cc-search-locations-dialog.c
@@ -261,6 +261,7 @@ get_tracker_locations (CcSearchLocationsDialog *self)
       path = path_from_tracker_dir (locations[idx]);
 
       location = g_slice_new0 (Place);
+      location->dialog = self;
       location->location = g_file_new_for_commandline_arg (path);
       location->display_name = g_file_get_basename (location->location);
       location->place_type = PLACE_OTHER;
