Index: gnome-control-center-3.0.2/panels/background/gnome-background-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.0.2.orig/panels/background/gnome-background-panel.desktop.in.in	2011-06-29 17:28:43.778521323 +0100
+++ gnome-control-center-3.0.2/panels/background/gnome-background-panel.desktop.in.in	2011-06-29 17:29:33.286521847 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;DesktopSettings;X-GNOME-Settings-Panel;X-GNOME-PersonalSettings
-OnlyShowIn=GNOME;
+OnlyShowIn=GNOME;Unity;
 X-GNOME-Settings-Panel=background
 # Translators: those are keywords for the background control-center panel
 _X-GNOME-Keywords=Wallpaper;Screen;Desktop;
Index: gnome-control-center-3.0.2/panels/display/gnome-display-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.0.2.orig/panels/display/gnome-display-panel.desktop.in.in	2011-06-29 17:28:52.262521413 +0100
+++ gnome-control-center-3.0.2/panels/display/gnome-display-panel.desktop.in.in	2011-06-29 17:29:41.006521928 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;
+OnlyShowIn=GNOME;Unity;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=Screen resolution
Index: gnome-control-center-3.0.2/panels/info/gnome-info-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.0.2.orig/panels/info/gnome-info-panel.desktop.in.in	2011-06-29 17:28:31.950521197 +0100
+++ gnome-control-center-3.0.2/panels/info/gnome-info-panel.desktop.in.in	2011-06-29 17:29:22.334521731 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;X-GNOME-SystemSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;
+OnlyShowIn=GNOME;Unity;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=info
Index: gnome-control-center-3.0.2/panels/keyboard/gnome-keyboard-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.0.2.orig/panels/keyboard/gnome-keyboard-panel.desktop.in.in	2011-06-29 17:29:03.022521526 +0100
+++ gnome-control-center-3.0.2/panels/keyboard/gnome-keyboard-panel.desktop.in.in	2011-06-29 17:29:50.702522031 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;
+OnlyShowIn=GNOME;Unity;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=keyboard
Index: gnome-control-center-3.0.2/panels/media/gnome-media-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.0.2.orig/panels/media/gnome-media-panel.desktop.in.in	2011-06-29 17:28:56.650521459 +0100
+++ gnome-control-center-3.0.2/panels/media/gnome-media-panel.desktop.in.in	2011-06-29 17:29:45.046521971 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;
+OnlyShowIn=GNOME;Unity;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=Media and autorun
Index: gnome-control-center-3.0.2/panels/mouse/gnome-mouse-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.0.2.orig/panels/mouse/gnome-mouse-panel.desktop.in.in	2011-06-29 17:28:48.966521378 +0100
+++ gnome-control-center-3.0.2/panels/mouse/gnome-mouse-panel.desktop.in.in	2011-06-29 17:29:39.050521908 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;
+OnlyShowIn=GNOME;Unity;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=mouse
Index: gnome-control-center-3.0.2/panels/network/gnome-network-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.0.2.orig/panels/network/gnome-network-panel.desktop.in.in	2011-06-29 17:28:54.462521436 +0100
+++ gnome-control-center-3.0.2/panels/network/gnome-network-panel.desktop.in.in	2011-06-29 17:29:42.866521948 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;
+OnlyShowIn=GNOME;Unity;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=network
Index: gnome-control-center-3.0.2/panels/power/gnome-power-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.0.2.orig/panels/power/gnome-power-panel.desktop.in.in	2011-06-29 17:28:35.086521231 +0100
+++ gnome-control-center-3.0.2/panels/power/gnome-power-panel.desktop.in.in	2011-06-29 17:29:25.478521764 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;DesktopSettings;X-GNOME-Settings-Panel;HardwareSettings
-OnlyShowIn=GNOME;
+OnlyShowIn=GNOME;Unity;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=power
Index: gnome-control-center-3.0.2/panels/region/gnome-region-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.0.2.orig/panels/region/gnome-region-panel.desktop.in.in	2011-06-29 17:28:39.978521283 +0100
+++ gnome-control-center-3.0.2/panels/region/gnome-region-panel.desktop.in.in	2011-06-29 17:29:28.802521799 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;DesktopSettings;X-GNOME-Settings-Panel;X-GNOME-PersonalSettings
-OnlyShowIn=GNOME;
+OnlyShowIn=GNOME;Unity;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=region
Index: gnome-control-center-3.0.2/panels/screen/gnome-screen-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.0.2.orig/panels/screen/gnome-screen-panel.desktop.in.in	2011-06-29 17:28:45.546521341 +0100
+++ gnome-control-center-3.0.2/panels/screen/gnome-screen-panel.desktop.in.in	2011-06-29 17:29:35.306521868 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;DesktopSettings;X-GNOME-Settings-Panel;X-GNOME-PersonalSettings
-OnlyShowIn=GNOME;
+OnlyShowIn=GNOME;Unity;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=screen
Index: gnome-control-center-3.0.2/panels/sound/data/gnome-sound-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.0.2.orig/panels/sound/data/gnome-sound-panel.desktop.in.in	2011-06-29 17:29:06.366521562 +0100
+++ gnome-control-center-3.0.2/panels/sound/data/gnome-sound-panel.desktop.in.in	2011-06-29 17:29:53.178522057 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;
+OnlyShowIn=GNOME;Unity;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=sound
Index: gnome-control-center-3.0.2/panels/universal-access/gnome-universal-access-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.0.2.orig/panels/universal-access/gnome-universal-access-panel.desktop.in.in	2011-06-29 17:28:41.806521302 +0100
+++ gnome-control-center-3.0.2/panels/universal-access/gnome-universal-access-panel.desktop.in.in	2011-06-29 17:29:31.122521824 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;X-GNOME-SystemSettings;X-GNOME-Settings-Panel;
-OnlyShowIn=GNOME;
+OnlyShowIn=GNOME;Unity;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=Universal Access
Index: gnome-control-center-3.0.2/panels/user-accounts/data/gnome-user-accounts-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.0.2.orig/panels/user-accounts/data/gnome-user-accounts-panel.desktop.in.in	2011-06-29 17:28:47.182521359 +0100
+++ gnome-control-center-3.0.2/panels/user-accounts/data/gnome-user-accounts-panel.desktop.in.in	2011-06-29 17:29:37.286521889 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=System;Settings;X-GNOME-Settings-Panel;X-GNOME-SystemSettings
-OnlyShowIn=GNOME;
+OnlyShowIn=GNOME;Unity;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=user-accounts
Index: gnome-control-center-3.0.2/shell/gnome-control-center.desktop.in.in
===================================================================
--- gnome-control-center-3.0.2.orig/shell/gnome-control-center.desktop.in.in	2011-06-29 17:29:08.202521581 +0100
+++ gnome-control-center-3.0.2/shell/gnome-control-center.desktop.in.in	2011-06-29 17:29:56.202522089 +0100
@@ -6,7 +6,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;System;
-OnlyShowIn=GNOME;
+OnlyShowIn=GNOME;Unity;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=shell
Index: gnome-control-center-3.0.2/panels/sound/data/gnome-sound-applet.desktop.in
===================================================================
--- gnome-control-center-3.0.2.orig/panels/sound/data/gnome-sound-applet.desktop.in	2011-06-29 17:30:54.310522704 +0100
+++ gnome-control-center-3.0.2/panels/sound/data/gnome-sound-applet.desktop.in	2011-06-29 17:31:23.858523017 +0100
@@ -7,7 +7,6 @@
 Type=Application
 Categories=
 NoDisplay=true
-OnlyShowIn=GNOME;XFCE;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=sound
@@ -15,4 +14,4 @@
 #X-GNOME-Autostart-Phase=Panel
 X-GNOME-Autostart-Notify=true
 AutostartCondition=GNOME3 if-session gnome-fallback
-OnlyShowIn=GNOME;
+OnlyShowIn=GNOME;Unity;
