# Description: files to translate from ubuntu patches
Index: gnome-control-center-3.6.3/po/POTFILES.in
===================================================================
--- gnome-control-center-3.6.3.orig/po/POTFILES.in	2014-02-26 19:01:53.387678617 +0100
+++ gnome-control-center-3.6.3/po/POTFILES.in	2014-02-26 19:02:22.587679589 +0100
@@ -98,6 +98,9 @@
 panels/sound/gvc-speaker-test.c
 panels/sound/gvc-stream-status-icon.c
 panels/sound/sound-theme-file-utils.c
+panels/sound-nua/data/gnome-sound-nua-panel.desktop.in.in
+panels/sound-nua/gvc-balance-bar.c
+panels/sound-nua/gvc-mixer-dialog.c
 panels/universal-access/cc-ua-panel.c
 panels/universal-access/gnome-universal-access-panel.desktop.in.in
 [type: gettext/glade]panels/universal-access/uap.ui
@@ -131,7 +134,10 @@
 [type: gettext/glade]panels/wacom/gnome-wacom-properties.ui
 panels/wacom/gsd-wacom-device.c
 [type: gettext/glade]panels/wacom/wacom-stylus-page.ui
+shell/cc-shell-nav-bar.c
 shell/control-center.c
 shell/gnomecc.directory.in
 shell/gnome-control-center.desktop.in.in
 [type: gettext/glade]shell/shell.ui
+panels/display/cc-rr-labeler.c
+
