Index: gnome-control-center-3.18.2/panels/keyboard/gnome-keyboard-panel.ui
===================================================================
--- gnome-control-center-3.18.2.orig/panels/keyboard/gnome-keyboard-panel.ui
+++ gnome-control-center-3.18.2/panels/keyboard/gnome-keyboard-panel.ui
@@ -534,23 +534,6 @@
                     <property name="visible">True</property>
                     <property name="can_focus">False</property>
                     <child>
-                      <object class="GtkLinkButton" id="linkbutton">
-                        <property name="label" translatable="yes">Input Sources</property>
-                        <property name="visible">True</property>
-                        <property name="can_focus">True</property>
-                        <property name="receives_default">True</property>
-                        <property name="has_tooltip">False</property>
-                        <property name="use_action_appearance">False</property>
-                        <property name="relief">none</property>
-                        <property name="uri">region panel</property>
-                      </object>
-                      <packing>
-                        <property name="expand">False</property>
-                        <property name="fill">True</property>
-                        <property name="position">0</property>
-                      </packing>
-                    </child>
-                    <child>
                       <placeholder/>
                     </child>
                   </object>
Index: gnome-control-center-3.18.2/panels/keyboard/keyboard-general.c
===================================================================
--- gnome-control-center-3.18.2.orig/panels/keyboard/keyboard-general.c
+++ gnome-control-center-3.18.2/panels/keyboard/keyboard-general.c
@@ -55,7 +55,7 @@ set_rate (const GValue       *value,
 }
 
 static gboolean
-layout_link_clicked (GtkLinkButton *button,
+layout_link_clicked (GtkButton *button,
                      CcPanel       *panel)
 {
   CcShell *shell;
@@ -105,7 +105,18 @@ keyboard_general_init (CcPanel *panel, G
                    gtk_range_get_adjustment (GTK_RANGE (WID ("cursor_blink_time_scale"))), "value",
                    G_SETTINGS_BIND_DEFAULT);
 
-  g_signal_connect (WID ("linkbutton"), "activate-link",
+  GtkWidget *button;
+  CcShell *shell;
+  shell = cc_panel_get_shell (CC_PANEL (panel));
+  button = gtk_button_new_with_label  ("Input Sources");
+  gtk_style_context_add_class (gtk_widget_get_style_context (button),
+                               "text-button");
+  gtk_widget_set_valign (button, GTK_ALIGN_CENTER);
+  gtk_widget_set_visible (button, TRUE);
+
+  cc_shell_embed_widget_in_header (shell, button);
+
+  g_signal_connect (button, "clicked",
                     G_CALLBACK (layout_link_clicked), panel);
 }
 
