Index: gnome-control-center-3.6.3/shell/gnome-control-center.c
===================================================================
--- gnome-control-center-3.6.3.orig/shell/gnome-control-center.c	2014-02-26 19:04:06.407683047 +0100
+++ gnome-control-center-3.6.3/shell/gnome-control-center.c	2014-02-26 19:04:09.000000000 +0100
@@ -1000,6 +1000,10 @@
 
   g_clear_pointer (&priv->current_panel_id, g_free);
 
+  if (!g_strcmp0 (g_getenv ("XDG_CURRENT_DESKTOP"), "Unity") && 
+      !g_strcmp0(start_id, "background"))
+      start_id = "unity-appearance";
+
   /* clear any custom widgets */
   _shell_remove_all_custom_widgets (priv);
 
