diff --git a/meson.build b/meson.build
index de1a73fa5..345fe32d5 100644
--- a/meson.build
+++ b/meson.build
@@ -125,7 +125,7 @@ glib_dep = dependency('glib-2.0', version: '>= 2.56.0')
 gnome_desktop_dep = dependency('gnome-desktop-3.0', version: '>= 3.27.90')
 gnome_settings_dep = dependency('gnome-settings-daemon', version: '>= 3.27.90')
 goa_dep = dependency('goa-1.0', version: goa_req_version)
-gsettings_desktop_dep = dependency('gsettings-desktop-schemas', version: '>= 3.37.1')
+gsettings_desktop_dep = dependency('gsettings-desktop-schemas', version: '>= 3.36.0')
 libxml_dep = dependency('libxml-2.0')
 polkit_gobject_dep = dependency('polkit-gobject-1', version: '>= 0.103')
 pulse_dep = dependency('libpulse', version: pulse_req_version)
