Index: gnome-control-center/panels/applications/meson.build
===================================================================
--- gnome-control-center.orig/panels/applications/meson.build
+++ gnome-control-center/panels/applications/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-       desktop,
          type : 'desktop',
         input : desktop_in,
        output : desktop,
Index: gnome-control-center/panels/background/meson.build
===================================================================
--- gnome-control-center.orig/panels/background/meson.build
+++ gnome-control-center/panels/background/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/bluetooth/meson.build
===================================================================
--- gnome-control-center.orig/panels/bluetooth/meson.build
+++ gnome-control-center/panels/bluetooth/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/camera/meson.build
===================================================================
--- gnome-control-center.orig/panels/camera/meson.build
+++ gnome-control-center/panels/camera/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/color/meson.build
===================================================================
--- gnome-control-center.orig/panels/color/meson.build
+++ gnome-control-center/panels/color/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/connectivity/meson.build
===================================================================
--- gnome-control-center.orig/panels/connectivity/meson.build
+++ gnome-control-center/panels/connectivity/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/datetime/meson.build
===================================================================
--- gnome-control-center.orig/panels/datetime/meson.build
+++ gnome-control-center/panels/datetime/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
@@ -20,7 +19,6 @@ i18n.merge_file(
 polkit = 'org.gnome.controlcenter.@0@.policy'.format(cappletname)
 
 i18n.merge_file(
-  polkit,
   input: polkit + '.in',
   output: polkit,
   po_dir: po_dir,
Index: gnome-control-center/panels/default-apps/meson.build
===================================================================
--- gnome-control-center.orig/panels/default-apps/meson.build
+++ gnome-control-center/panels/default-apps/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/diagnostics/meson.build
===================================================================
--- gnome-control-center.orig/panels/diagnostics/meson.build
+++ gnome-control-center/panels/diagnostics/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/display/meson.build
===================================================================
--- gnome-control-center.orig/panels/display/meson.build
+++ gnome-control-center/panels/display/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/firmware/meson.build
===================================================================
--- gnome-control-center.orig/panels/firmware/meson.build
+++ gnome-control-center/panels/firmware/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-       desktop,
           type : 'desktop',
          input : desktop_in,
         output : desktop,
@@ -36,4 +35,4 @@ panels_libs += static_library(
   include_directories: [ top_inc, common_inc ],
   dependencies: deps,
   c_args: cflags
-)
\ No newline at end of file
+)
Index: gnome-control-center/panels/info-overview/meson.build
===================================================================
--- gnome-control-center.orig/panels/info-overview/meson.build
+++ gnome-control-center/panels/info-overview/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/keyboard/meson.build
===================================================================
--- gnome-control-center.orig/panels/keyboard/meson.build
+++ gnome-control-center/panels/keyboard/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
@@ -45,7 +44,6 @@ xml_files = [
 
 foreach file: xml_files
   i18n.merge_file(
-    file,
     input: file + '.in',
     output: file,
     po_dir: po_dir,
Index: gnome-control-center/panels/location/meson.build
===================================================================
--- gnome-control-center.orig/panels/location/meson.build
+++ gnome-control-center/panels/location/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/lock/meson.build
===================================================================
--- gnome-control-center.orig/panels/lock/meson.build
+++ gnome-control-center/panels/lock/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/microphone/meson.build
===================================================================
--- gnome-control-center.orig/panels/microphone/meson.build
+++ gnome-control-center/panels/microphone/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/mouse/meson.build
===================================================================
--- gnome-control-center.orig/panels/mouse/meson.build
+++ gnome-control-center/panels/mouse/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/network/meson.build
===================================================================
--- gnome-control-center.orig/panels/network/meson.build
+++ gnome-control-center/panels/network/meson.build
@@ -25,7 +25,6 @@ foreach name: panel_names
   )
 
   i18n.merge_file(
-    desktop,
     type: 'desktop',
     input: desktop_in,
     output: desktop,
Index: gnome-control-center/panels/notifications/meson.build
===================================================================
--- gnome-control-center.orig/panels/notifications/meson.build
+++ gnome-control-center/panels/notifications/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/online-accounts/meson.build
===================================================================
--- gnome-control-center.orig/panels/online-accounts/meson.build
+++ gnome-control-center/panels/online-accounts/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/power/meson.build
===================================================================
--- gnome-control-center.orig/panels/power/meson.build
+++ gnome-control-center/panels/power/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/printers/meson.build
===================================================================
--- gnome-control-center.orig/panels/printers/meson.build
+++ gnome-control-center/panels/printers/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
@@ -74,4 +73,3 @@ printers_panel_lib = static_library(
   c_args: cflags + cups_cflags,
 )
 panels_libs += [ printers_panel_lib ]
-
Index: gnome-control-center/panels/region/meson.build
===================================================================
--- gnome-control-center.orig/panels/region/meson.build
+++ gnome-control-center/panels/region/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/removable-media/meson.build
===================================================================
--- gnome-control-center.orig/panels/removable-media/meson.build
+++ gnome-control-center/panels/removable-media/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/search/meson.build
===================================================================
--- gnome-control-center.orig/panels/search/meson.build
+++ gnome-control-center/panels/search/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/sharing/meson.build
===================================================================
--- gnome-control-center.orig/panels/sharing/meson.build
+++ gnome-control-center/panels/sharing/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
@@ -29,7 +28,6 @@ polkit_in = configure_file(
 )
 
 i18n.merge_file(
-  polkit,
   input: polkit_in,
   output: polkit,
   po_dir: po_dir,
Index: gnome-control-center/panels/sound/meson.build
===================================================================
--- gnome-control-center.orig/panels/sound/meson.build
+++ gnome-control-center/panels/sound/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/thunderbolt/meson.build
===================================================================
--- gnome-control-center.orig/panels/thunderbolt/meson.build
+++ gnome-control-center/panels/thunderbolt/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/universal-access/meson.build
===================================================================
--- gnome-control-center.orig/panels/universal-access/meson.build
+++ gnome-control-center/panels/universal-access/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/upgrade/meson.build
===================================================================
--- gnome-control-center.orig/panels/upgrade/meson.build
+++ gnome-control-center/panels/upgrade/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-       desktop,
           type : 'desktop',
          input : desktop_in,
         output : desktop,
Index: gnome-control-center/panels/usage/meson.build
===================================================================
--- gnome-control-center.orig/panels/usage/meson.build
+++ gnome-control-center/panels/usage/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
Index: gnome-control-center/panels/user-accounts/meson.build
===================================================================
--- gnome-control-center.orig/panels/user-accounts/meson.build
+++ gnome-control-center/panels/user-accounts/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
@@ -85,7 +84,6 @@ meson.add_install_script('sh', '-c',
 polkit = 'org.gnome.controlcenter.@0@.policy'.format(cappletname)
 
 i18n.merge_file(
-  polkit,
   input: polkit + '.in',
   output: polkit,
   po_dir: po_dir,
Index: gnome-control-center/panels/wacom/meson.build
===================================================================
--- gnome-control-center.orig/panels/wacom/meson.build
+++ gnome-control-center/panels/wacom/meson.build
@@ -30,7 +30,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-       desktop,
          type : 'desktop',
         input : desktop_in,
        output : desktop,
Index: gnome-control-center/shell/appdata/meson.build
===================================================================
--- gnome-control-center.orig/shell/appdata/meson.build
+++ gnome-control-center/shell/appdata/meson.build
@@ -1,7 +1,6 @@
 appdata = 'gnome-control-center.appdata.xml'
 
 i18n.merge_file(
-  appdata,
   input: appdata + '.in',
   output: appdata,
   po_dir: po_dir,
Index: gnome-control-center/shell/meson.build
===================================================================
--- gnome-control-center.orig/shell/meson.build
+++ gnome-control-center/shell/meson.build
@@ -23,7 +23,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-       desktop,
          type : 'desktop',
         input : desktop_in,
        output : desktop,
Index: gnome-control-center/tests/interactive-panels/applications/meson.build
===================================================================
--- gnome-control-center.orig/tests/interactive-panels/applications/meson.build
+++ gnome-control-center/tests/interactive-panels/applications/meson.build
@@ -12,7 +12,6 @@ desktop_files = [
 
 foreach desktop_file : desktop_files
   i18n.merge_file(
-              desktop,
                 type : 'desktop',
                input : 'gtp-@0@.desktop.in'.format(desktop_file),
               output : 'gnome-@0@-panel.desktop'.format(desktop_file),
Index: gnome-control-center/panels/support/meson.build
===================================================================
--- gnome-control-center.orig/panels/support/meson.build
+++ gnome-control-center/panels/support/meson.build
@@ -8,7 +8,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-       desktop,
           type : 'desktop',
          input : desktop_in,
         output : desktop,
