Index: gnome-control-center/panels/keyboard/01-launchers.xml.in
===================================================================
--- gnome-control-center.orig/panels/keyboard/01-launchers.xml.in
+++ gnome-control-center/panels/keyboard/01-launchers.xml.in
@@ -11,6 +11,8 @@
 
 	<KeyListEntry name="terminal" description="Launch terminal"/>
 
+	<KeyListEntry name="wifi" description="WiFi Settings"/>
+
 	<KeyListEntry name="www" description="Launch web browser"/>
 
 	<KeyListEntry name="home" description="Home folder"/>
