--- a/panels/meson.build
+++ b/panels/meson.build
@@ -18,7 +18,6 @@
   'location',
   'lock',
   'mouse',
-  'multitasking',
   'notifications',
   'online-accounts',
   'power',
--- a/shell/cc-panel-list.c
+++ b/shell/cc-panel-list.c
@@ -402,7 +402,6 @@
   "ubuntu",
   "notifications",
   "search",
-  "multitasking",
   "applications",
   "privacy",
   "online-accounts",
--- a/shell/cc-panel-loader.c
+++ b/shell/cc-panel-loader.c
@@ -45,7 +45,6 @@
 extern GType cc_info_overview_panel_get_type (void);
 extern GType cc_keyboard_panel_get_type (void);
 extern GType cc_mouse_panel_get_type (void);
-extern GType cc_multitasking_panel_get_type (void);
 #ifdef BUILD_NETWORK
 extern GType cc_network_panel_get_type (void);
 extern GType cc_wifi_panel_get_type (void);
@@ -123,7 +122,6 @@
   PANEL_TYPE("lock",             cc_lock_panel_get_type,                 NULL),
   //PANEL_TYPE("microphone",       cc_microphone_panel_get_type,           NULL),
   PANEL_TYPE("mouse",            cc_mouse_panel_get_type,                NULL),
-  PANEL_TYPE("multitasking",     cc_multitasking_panel_get_type,         NULL),
 #ifdef BUILD_NETWORK
   PANEL_TYPE("network",          cc_network_panel_get_type,              NULL),
   PANEL_TYPE("wifi",             cc_wifi_panel_get_type,                 cc_wifi_panel_static_init_func),
